<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    
    <meta name="theme-color" content="#33363b">
    <meta name="msapplication-TileColor" content="#33363b">
    
    
    
    <meta name="keywords" content="python, R, math, bioinformatics">
    
    
    <link rel="apple-touch-icon" sizes="180x180" href="/favicons/apple-touch-icon.png">
    
    
    <link rel="icon" type="image/png" sizes="192x192" href="/favicons/android-chrome-192x192.png">
    
    
    <link rel="icon" type="image/png" sizes="32x32" href="/favicons/favicon-32x32.png">
    
    
    <link rel="icon" type="image/png" sizes="16x16" href="/favicons/favicon-16x16.png">
    
    
    <link rel="mask-icon" href="/favicons/safari-pinned-tab.svg" color="#33363b">
    
    
    <link rel="manifest" href="/favicons/site.webmanifest">
    
    
    <meta name="msapplication-config" content="/favicons/browserconfig.xml">
    
    
    <link rel="alternate" href="/atom.xml" title="mt's blog" type="application/atom+xml" />
    
    <link rel="stylesheet" type="text/css" href="/css/index.css">
    
    <link rel="stylesheet" type="text/css" href="/css/sidebar.css">
    
    
    <link rel="stylesheet" type="text/css" href="/css/custom.css">
    <link rel="stylesheet" type="text/css" href="/css/normalize.css">
    <link rel="stylesheet" type="text/css" href="/css/solarized-light.css">
    <link rel="stylesheet" type="text/css" href="/css/lightgallery.min.css">
    <script type="text/javascript" src="/js/jquery.min.js"></script>
    <script defer type="text/javascript" src="/js/util.js"></script>
    <script defer type="text/javascript" src="/js/scrollspy.js"></script>
    <script defer type="text/javascript" src="/js/fontawesome-all.min.js"></script>
    <script defer type="text/javascript" src="/js/lightgallery.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-fullscreen.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-hash.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-pager.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-thumbnail.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-zoom.min.js"></script>
    
    <script defer src="/js/busuanzi.pure.mini.js"></script>
    
    
    <script defer type="text/javascript" src="/js/search.js"></script>
    <script type="text/javascript">
    $(document).ready(function () {
      var searchPath = "search.xml";
      if (searchPath.length === 0) {
        searchPath = "search.xml";
      }
      var path = "/" + searchPath;
      searchFunc(path, "search-input", "search-result");
    });
    </script>
    
    
    <script defer type="text/javascript" src="/js/index.js"></script>
    
    <script defer type="text/javascript" src="/js/custom.js"></script>
    <title>mt's blog</title>
  </head>
  <body itemscope itemtype="http://schema.org/WebPage" lang=""  data-spy="scroll" data-target=".list-group">
    
<header id="header" class="header">
  <div class="container">
    <div class="header-container">
      <div class="header-title">
        <h1 class="title"><a href="/">mt's blog</a></h1>
        <h2 class="subtitle"></h2>
      </div>
      <div class="logo">
        <img src="/images/ARIA_logo.png" alt="logo">
      </div>
    </div>
    
<nav id="nav" class="nav">
  <a id="nav-toggle" class="nav-toggle"><i class="fas fa-bars"></i></a>
  <ul id="menu">
    
    <li><a href="/">Home</a></li>
    
    <li><a href="/archives/">Archives</a></li>
    
    <li><a href="/categories/">Categories</a></li>
    
    <li><a href="/tags/">Tags</a></li>
    
  </ul>
</nav>


  </div>
</header>


    <main id="main" class="main">
      <div class="container">
        <div class="main-container">
          <div class="content">
            

<section id="index" class="index">
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/15/转录组入门——6/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/15/转录组入门——6/" itemprop="url">转录组入门——6</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-15T20:37:18+08:00">2017-08-15</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/转录组入门/" itemprop="url" rel="index"><span itemprop="name">转录组入门</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/15/转录组入门——6/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/15/转录组入门——6/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="实现reads计数的软件教程"><a href="#实现reads计数的软件教程" class="headerlink" title="实现reads计数的软件教程"></a>实现reads计数的软件教程</h5><blockquote>
<p>map后获得了每条read比对到基因组上的位置，后面需要把基因组的位置对应到基因。这个提供基因位置的注释文件通常以GTF或GFF3格式呈现。基因表达量最直接的分析手段就是计算比对到每个基因的reads有多少条，在转录组测序中，我们通常称这个数字为count<br>对每个基因计数之后得到的count matrix再后续的分析中，要注意标准化的问题。<br>比较同一个样本(within-sample)不同基因之间的表达情况，需要考虑到转录本长度，因为转录本越长，那么检测的片段也会更多。比较不同样本（across sample）同一个基因的表达情况，不必在意转录本长度，但要考虑到测序深度（sequence depth)，测序深度越高，检测到的概率越大。除这两因素外，还需考虑GC%导致的偏差及测序仪器的系统偏差。<br>目前对read count标准化的算法有RPKM（SE）, FPKM（PE），TPM, TMM等。但是，有一些下游分析的软件会要求是输入的count matrix是原始数据，未经标准化，比如说DESeq2，这个时候你需要注意你上一步所用软件会不会进行标准化。<br>计数分为三个水平： gene-level, transcript-level, exon-usage-level<br>标准化方法： FPKM RPKM TMM TPM</p>
</blockquote>
<h5 id="Gene-level"><a href="#Gene-level" class="headerlink" title="Gene-level"></a>Gene-level</h5><blockquote>
<p>HTSeq <a href="https://htseq.readthedocs.io/en/release_0.9.1/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.chenlianfu.com/?p=2438" target="_blank" rel="noopener">中文教程</a><br>featureCounts <a href="http://bioinf.wehi.edu.au/featureCounts/" target="_blank" rel="noopener">官方网址</a> Rsubread <a href="https://bioconductor.org/packages/release/bioc/html/Rsubread.html" target="_blank" rel="noopener">官方网址</a><br>bedtools multicov <a href="https://bedtools.readthedocs.io/en/latest/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.biotrainee.com/thread-153-1-1.html" target="_blank" rel="noopener">中文教程</a><br>Qualimap <a href="http://qualimap.bioinfo.cipf.es/" target="_blank" rel="noopener">官方网址</a> <a href="http://blog.sciencenet.cn/blog-2970729-1070459.html" target="_blank" rel="noopener">中文教程</a><br>GenomicRanges <a href="https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html" target="_blank" rel="noopener">官方网址</a></p>
</blockquote>
<h5 id="Transcript-level"><a href="#Transcript-level" class="headerlink" title="Transcript-level"></a>Transcript-level</h5><blockquote>
<p>cufflink <a href="https://cole-trapnell-lab.github.io/cufflinks/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.chenlianfu.com/?p=623" target="_blank" rel="noopener">中文教程</a><br>stringtie <a href="https://ccb.jhu.edu/software/stringtie/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.bio-info-trainee.com/2073.html" target="_blank" rel="noopener">中文教程</a><br>上述软件都是alignment-based<br>alignment-free软件，如kallisto, silfish, salmon，能够省去比对这一步，直接得到read count，在运行效率上更高。不过最近一篇文献Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis指出这类方法在估计丰度时存在样本特异性和读长偏差<br>salmon</p>
</blockquote>
<h5 id="Exon-usage-level"><a href="#Exon-usage-level" class="headerlink" title="Exon-usage-level"></a>Exon-usage-level</h5><blockquote>
<p>和基因水平的统计类似。但是值得注意的是为了更好的计数，我们需要提供无重叠的外显子区域的gtf文件Detecting differential usage of exons from RNA-seq data。用于分析差异外显子使用的DEXSeq提供了一个Python脚本（dexseq_prepare_annotation.py）执行这个任务。</p>
</blockquote>
<h5 id="基因表达量归一算法"><a href="#基因表达量归一算法" class="headerlink" title="基因表达量归一算法"></a>基因表达量归一算法</h5><p>待续 <a href="https://zhuanlan.zhihu.com/p/26750663" target="_blank" rel="noopener">参考</a></p>
<h5 id="htseq-count"><a href="#htseq-count" class="headerlink" title="htseq-count"></a>htseq-count</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">ls Homo*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $hg19_gff &gt; $&#123;id%_*&#125;.geneCounts 2&gt; $&#123;id%_*&#125;.HTseq.log);done</span><br><span class="line">ls Mus*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $mm10_gff &gt; $&#123;id%_*&#125;.geneCounts 2&gt; $&#123;id%_*&#125;.HTseq.log);done</span><br><span class="line">wc -l Homo*.geneCounts </span><br><span class="line">head -n 4 Homo*.geneCounts</span><br></pre></td></tr></table></figure>
<blockquote>
<p>-f bam/sam： 指定输入文件格式，默认SAM<br>-r name/pos: 你需要利用samtool sort对数据根据read name或者位置进行排序，默认是name<br>-s yes/no/reverse: 数据是否来自于strand-specific assay。DNA是双链的，所以需要判断到底来自于哪条链。如果选择了no， 那么每一条read都会跟正义链和反义链进行比较。默认的yes对于双端测序表示第一个read都在同一个链上，第二个read则在另一条链上。<br>-a 最低质量， 剔除低于阈值的read<br>-m 模式 union（默认）, intersection-strict and intersection-nonempty。一般而言就用默认的，作者也是这样认为的。<br>-i id attribute: 在GTF文件的最后一栏里，会有这个基因的多个命名方式（如下）， RNA-Seq数据分析常用的是gene_id， 当然你可以写一个脚本替换成其他命名方式。</p>
</blockquote>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-6aaf8956467c4fb6.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="homo.genecounts"></p>
<h5 id="编写脚本合并所有的样本为表达矩阵"><a href="#编写脚本合并所有的样本为表达矩阵" class="headerlink" title="编写脚本合并所有的样本为表达矩阵"></a>编写脚本合并所有的样本为表达矩阵</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">sed -i &apos;$d&apos; *geneCounts</span><br><span class="line">perl -lne &apos;if($ARGV=~/Homo_(.*).geneCounts/)&#123;print &quot;$1\t$_&quot;&#125;&apos; *|grep -v Homo_sapiens&gt;hg.count</span><br><span class="line">perl -lne &apos;if($ARGV=~/Mus_(.*).geneCounts/)&#123;print &quot;$1\t$_&quot;&#125;&apos; *|grep -v Mus &gt;mus.count</span><br><span class="line"></span><br><span class="line">hg &lt;- read.csv(file = &quot;hg.count&quot;,header = F,sep = &quot;\t&quot;)</span><br><span class="line">colnames(hg) &lt;- c(&apos;sample&apos;,&apos;gene&apos;,&apos;count&apos;)</span><br><span class="line">library(reshape2)</span><br><span class="line">hg_reads &lt;- dcast(hg,formula = gene ~ sample)</span><br><span class="line">write.table(hg_reads,file = &quot;hg_join.count&quot;,sep = &quot;\t&quot;,quote = FALSE,row.names = FALSE)</span><br><span class="line"></span><br><span class="line">mus &lt;- read.csv(file = &quot;mus.count&quot;,header = F,sep = &quot;\t&quot;)</span><br><span class="line">colnames(mus) &lt;- c(&apos;sample&apos;,&apos;gene&apos;,&apos;count&apos;)</span><br><span class="line">library(reshape2)</span><br><span class="line">mus_reads &lt;- dcast(mus,formula = gene ~ sample)</span><br><span class="line">write.table(mus_reads,file = &quot;mus_join.count&quot;,sep = &quot;\t&quot;,quote = FALSE,row.names = FALSE)</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-bf2396936c8bb516.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="homo_table"></p>
<h5 id="对表达矩阵可进行简单的探索"><a href="#对表达矩阵可进行简单的探索" class="headerlink" title="对表达矩阵可进行简单的探索"></a>对表达矩阵可进行简单的探索</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">summary(hg_reads) </span><br><span class="line">summary(mus_reads) </span><br><span class="line">row.names(hg_reads) &lt;- hg_reads$gene</span><br><span class="line">GAPDH &lt;- hg_reads[rownames(hg_reads)==&quot;GAPDH&quot;,]</span><br><span class="line"></span><br><span class="line">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">biocLite(&quot;dplyr&quot;)</span><br><span class="line">library(dplyr)</span><br><span class="line">sample_mean &lt;- group_by(hg,sample)%&gt;% summarize(mean=mean(count))</span><br><span class="line">gene_mean &lt;- group_by(hg,gene)%&gt;% summarize(mean=mean(count)) </span><br><span class="line">tmp &lt;- full_join(hg,gene_mean,by=&apos;gene&apos;)</span><br><span class="line">write.table(tmp,file = &quot;tmp.count&quot;,sep = &quot;\t&quot;,quote = FALSE,row.names = FALSE)</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-3a0f908f6e1b3a5d.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="summary"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-79ab4164d01877a0.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="image.png"></p>
<h5 id="另一种"><a href="#另一种" class="headerlink" title="另一种"></a>另一种</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line">options(stringsAsFactors = FALSE)</span><br><span class="line">#导入数据</span><br><span class="line">homo_control &lt;- read.table(&quot;Homo_SRR3589956.geneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;homo_control&quot;))</span><br><span class="line">homo_rep1 &lt;- read.table(&quot;Homo_SRR3589957.geneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;homo_rep1&quot;))</span><br><span class="line">homo_rep2 &lt;- read.table(&quot;Homo_SRR3589958.geneCounts&quot;,sep=&quot;\t&quot;,col.names = c(&quot;gene_symbol&quot;,&quot;rep2&quot;))</span><br><span class="line">mus_control1 &lt;- read.table(&quot;Mus_SRR3589959.geneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;mus_control1&quot;))</span><br><span class="line">mus_control2 &lt;- read.table(&quot;Mus_SRR3589960.geneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;mus_control2&quot;)) </span><br><span class="line">mus_rep1 &lt;- read.table(&quot;Mus_SRR3589961.geneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;mus_rep1&quot;)) </span><br><span class="line">mus_rep2 &lt;- read.table(&quot;Mus_SRR3589962.geneCounts&quot;, sep=&quot;\t&quot;,col.names = c(&quot;gene_symbol&quot;,&quot;mus_rep2&quot;))</span><br><span class="line">#去除后五行</span><br><span class="line">homo_control &lt;- homo_control[-58523:-58527,]</span><br><span class="line">homo_rep1 &lt;- homo_rep1[-58523:-58527,]</span><br><span class="line">homo_rep2 &lt;- homo_rep2[-58523:-58527,]</span><br><span class="line">mus_control1 &lt;- mus_control1[-48322:-48326,]</span><br><span class="line">mus_control2 &lt;- mus_control2[-48322:-48326,]</span><br><span class="line">mus_rep1 &lt;- mus_rep1[-48322:-48326,]</span><br><span class="line">mus_rep2 &lt;- mus_rep2[-48322:-48326,]</span><br><span class="line">#整合</span><br><span class="line">homo_raw_count &lt;- merge(merge(homo_control, homo_rep1, by=&quot;gene_symbol&quot;), homo_rep2, by=&quot;gene_symbol&quot;)</span><br><span class="line">mus_raw_count &lt;- merge(merge(mus_control1, mus_control2, by=&quot;gene_symbol&quot;), merge(mus_rep1,mus_rep2, by=&quot;gene_symbol&quot;))</span><br><span class="line">summary(homo_raw_count)</span><br><span class="line">summary(mus_raw_count)</span><br><span class="line">row.names(homo_raw_count) &lt;- homo_raw_count$gene_symbol</span><br><span class="line">row.names(mus_raw_count) &lt;- mus_raw_count$gene_symbol</span><br><span class="line">gapdh &lt;- homo_raw_count[rownames(homo_raw_count)==&quot;GAPDH&quot;,]</span><br><span class="line">gapdh &lt;- mus_raw_count[rownames(mus_raw_count)==&quot;GAPDH&quot;,]</span><br><span class="line"></span><br><span class="line">#因为我们无法在EBI数据库上直接搜索找到ENSMUSG00000024045.5这样的基因，只能是ENSMUSG00000024045的整数，没有小数点，所以需要进一步替换为整数的形式</span><br><span class="line">#第一步将匹配到的.以及后面的数字连续匹配并替换为空，并赋值给ENSEMBL</span><br><span class="line">#ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, raw_count_filt1$gene_id) </span><br><span class="line">#将ENSEMBL重新添加到raw_count_filt1矩阵</span><br><span class="line">#row.names(raw_count_filter) &lt;- ENSEMBL</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-8ee1f5c36ac2faa3.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="gapdh"><br>以上两种都有两个问题：AKAP95未找到 小鼠第一列不是gene name<br>和sort是否为reads name无关，htseq count -i 参数去掉则默认为gene id</p>
<h5 id="看生物学意义特殊的基因表现如何如GAPDH-β-ACTIN"><a href="#看生物学意义特殊的基因表现如何如GAPDH-β-ACTIN" class="headerlink" title="看生物学意义特殊的基因表现如何如GAPDH,β-ACTIN"></a>看生物学意义特殊的基因表现如何如GAPDH,β-ACTIN</h5><p><img src="http://upload-images.jianshu.io/upload_images/2116382-e29d08dec77d3ec7.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="gapdh"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">#! usr/bin/bash</span><br><span class="line">set -u</span><br><span class="line">set -e</span><br><span class="line">set -o pipefail</span><br><span class="line">hg19_ref=~/fastq/hisat2_index/hg19/genome</span><br><span class="line">mm10_ref=~/fastq/hisat2_index/mm10/genome</span><br><span class="line">hg19_gff=~/fastq/refdata/Genecode/gencode.v26lift37.annotation.gff3</span><br><span class="line">mm10_gff=~/fastq/refdata/Genecode/gencode.vM10.annotation.gtf</span><br><span class="line">ls Homo*.sorted.bam|while read id;do(htseq-count -f bam -r pos -s no $id $hg19_gff &gt; $&#123;id%%.*&#125;.sgeneCounts 2&gt; $&#123;id%%.*&#125;.sHTseq.log);done</span><br><span class="line">ls Mus*.sorted.bam|while read id;do(htseq-count -f bam -r pos -s no $id $mm10_gff &gt; $&#123;id%%.*&#125;.sgeneCounts 2&gt; $&#123;id%%.*&#125;.sHTseq.log);done</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">options(stringsAsFactors = FALSE)</span><br><span class="line">homo_control &lt;- read.table(&quot;Homo_SRR3589956.sgeneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;homo_control&quot;))</span><br><span class="line">homo_rep1 &lt;- read.table(&quot;Homo_SRR3589957.sgeneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;homo_rep1&quot;))</span><br><span class="line">homo_rep2 &lt;- read.table(&quot;Homo_SRR3589958.sgeneCounts&quot;,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;rep2&quot;))</span><br><span class="line">mus_control1 &lt;- read.table(&quot;Mus_SRR3589959.sgeneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;mus_control1&quot;))</span><br><span class="line">mus_control2 &lt;- read.table(&quot;Mus_SRR3589960.sgeneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;mus_control2&quot;)) </span><br><span class="line">mus_rep1 &lt;- read.table(&quot;Mus_SRR3589961.sgeneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;mus_rep1&quot;)) </span><br><span class="line">mus_rep2 &lt;- read.table(&quot;Mus_SRR3589962.sgeneCounts&quot;, sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;mus_rep2&quot;))</span><br><span class="line">homo_control &lt;- homo_control[-60472:-60476,]</span><br><span class="line">homo_rep1 &lt;- homo_rep1[-60472:-60476,]</span><br><span class="line">homo_rep2 &lt;- homo_rep2[-60472:-60476,]</span><br><span class="line">mus_control1 &lt;- mus_control1[-48441:-48445,]</span><br><span class="line">mus_control2 &lt;- mus_control2[-48441:-48445,]</span><br><span class="line">mus_rep1 &lt;- mus_rep1[-48441:-48445,]</span><br><span class="line">mus_rep2 &lt;- mus_rep2[-48441:-48445,]</span><br><span class="line">homo_raw_count &lt;- merge(merge(homo_control, homo_rep1, by=&quot;gene_id&quot;), homo_rep2, by=&quot;gene_id&quot;)</span><br><span class="line">mus_raw_count &lt;- merge(merge(mus_control1, mus_control2, by=&quot;gene_id&quot;), merge(mus_rep1,mus_rep2, by=&quot;gene_id&quot;))</span><br><span class="line">ENSEMBL &lt;- gsub(&quot;\\.\\d*_\\d*&quot;, &quot;&quot;, homo_raw_count$gene_id)</span><br><span class="line">row.names(homo_raw_count) &lt;- ENSEMBL # error</span><br><span class="line">ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, mus_raw_count$gene_id) </span><br><span class="line">row.names(mus_raw_count) &lt;- ENSEMBL</span><br><span class="line">summary(homo_raw_count)</span><br><span class="line">summary(mus_raw_count)</span><br><span class="line">#在UniProt数据库找到AKAP95的id</span><br><span class="line">MUS_AKAP95 &lt;- mus_raw_count[rownames(mus_raw_count)==&quot;ENSMUSG00000024045&quot;,] #reads很大</span><br></pre></td></tr></table></figure>
<p>最开始未加 -r pos 有问题 reads数目不同，后续计数没有大问题<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">count.matrix&lt;-function()</span><br><span class="line">&#123;</span><br><span class="line">  .libPaths(&quot;/home/tingm/R/x86_64-pc-linux-gnu-library/3.2&quot;)</span><br><span class="line">  library(limma)</span><br><span class="line">  library(edgeR)</span><br><span class="line">    #change</span><br><span class="line">    path&lt;-&quot;/home/tingm/fastq/homo_result_count/&quot;</span><br><span class="line">    #change</span><br><span class="line">      samples&lt;-read.table(&quot;samples.list.txt&quot;,header=F,sep=&quot;\t&quot;)</span><br><span class="line">      samples&lt;-samples[,1]</span><br><span class="line">      #change</span><br><span class="line">      data.raw&lt;-read.table(&quot;Homo_SRR3589956.sgeneCounts&quot;,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;Homo_SRR3589956.sgeneCounts&quot;))</span><br><span class="line">      for(sample in samples)</span><br><span class="line">      &#123;</span><br><span class="line">        print(sample)</span><br><span class="line">        file&lt;-paste(path,sample,sep=&quot;&quot;)</span><br><span class="line">        data.new&lt;-read.table(file,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,sample))</span><br><span class="line">        data.raw&lt;-merge(data.new,data.raw,by=&quot;gene_id&quot;)</span><br><span class="line">      &#125;</span><br><span class="line">     data.raw&lt;-data.raw[-1:-5,]</span><br><span class="line">     #ENSEMBL &lt;- gsub(&quot;\\.\\d*_\\d*&quot;, &quot;&quot;, data.raw$gene_id)</span><br><span class="line">     #row.names(data.raw) &lt;- ENSEMBL</span><br><span class="line">     # ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, mus_raw_count$gene_id)</span><br><span class="line">     write.table(data.raw,&quot;count.matix.csv&quot;,row.names=F,quote=F)</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure></p>
<blockquote>
<p>参考引用来自以下资料<br><a href="https://mp.weixin.qq.com/s/V32iruEVRKJ_1RubuS_lUQ" target="_blank" rel="noopener">PANDA姐的转录组入门(6)：reads计数</a><br><a href="https://mp.weixin.qq.com/s/RolngZkua6GJboHQMfP52w" target="_blank" rel="noopener">生信媛 转录组入门(6)： reads计数</a><br><a href="http://www.jianshu.com/p/24cf44b610a7" target="_blank" rel="noopener">lxmic转录组入门(6)： reads计数</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/14/转录组入门——5/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/14/转录组入门——5/" itemprop="url">转录组入门——5</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-14T18:21:01+08:00">2017-08-14</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/转录组入门/" itemprop="url" rel="index"><span itemprop="name">转录组入门</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/14/转录组入门——5/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/14/转录组入门——5/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="比对软件简介"><a href="#比对软件简介" class="headerlink" title="比对软件简介"></a>比对软件简介</h5><blockquote>
<p><a href="http://www.jianshu.com/p/681e02e7f9af" target="_blank" rel="noopener">比对软件比较</a><br>Bowtie1 <a href="http://bowtie-bio.sourceforge.net/index.shtml" target="_blank" rel="noopener">官方网址</a><br>Bowtie2 <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" target="_blank" rel="noopener">官方网址</a> <a href="http://blog.leanote.com/post/songtaogui/Bowtie2%E5%AE%89%E8%A3%85%E4%BD%BF%E7%94%A8" target="_blank" rel="noopener">中文教程</a> <a href="http://blog.leanote.com/post/375952584@qq.com/Bowtie2%E7%9A%84%E5%AE%89%E8%A3%85%E4%B8%8E%E4%BD%BF%E7%94%A8" target="_blank" rel="noopener">中文教程</a><br><a href="https://cncbi.github.io/Bowtie2-Manual-CN/" target="_blank" rel="noopener">中文教程</a> <a href="http://fhqdddddd.blog.163.com/blog/static/18699154201422735644352/" target="_blank" rel="noopener">中文教程</a><br>Bowtie1 Bowtie2 <a href="http://blog.csdn.net/soyabean555999/article/details/62235577" target="_blank" rel="noopener">二者区别</a><br><a href="http://www.cnblogs.com/wq242424/p/7025946.html" target="_blank" rel="noopener">Bowtie 安装</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">#下载bowtie2-2.3.2-linux-x86_64.zip解压后有错误 未解决 tbb需要Python2.7</span><br><span class="line">find ./ -name libtbbmalloc_proxy.so.2</span><br><span class="line">echo $LD_LIBRARY_PATH</span><br><span class="line">export LD_LIBRARY_PATH=~/biosoft/tbb2017_20170604oss/lib/:$LD_LIBRARY_PATH</span><br></pre></td></tr></table></figure></p>
<p><a href="http://blog.sina.com.cn/s/blog_83f77c940102wp28.html" target="_blank" rel="noopener">TBB下载地址</a><br><a href="http://www.linuxidc.com/Linux/2014-01/94717.htm" target="_blank" rel="noopener">Linux LD_LIBRARY_PATH</a><br><a href="https://stackoverflow.com/questions/32592956/apt-get-command-not-found" target="_blank" rel="noopener">apt-get 和 yum install</a><br>Tophat <a href="https://ccb.jhu.edu/software/tophat/manual.shtml" target="_blank" rel="noopener">官方网址</a> <a href="http://www.chenlianfu.com/?p=910" target="_blank" rel="noopener">中文教程</a><br>Hisat2 <a href="https://ccb.jhu.edu/software/hisat2/index.shtml" target="_blank" rel="noopener">官方网址</a> <a href="http://blog.biochen.com/archives/337" target="_blank" rel="noopener">中文教程</a><br>STAR <a href="http://www.bio-info-trainee.com/727.html" target="_blank" rel="noopener">中文教程</a> <a href="http://www.cnblogs.com/daimakun/p/5081638.html" target="_blank" rel="noopener">中文教程</a><br>其他比对软件 <a href="http://www.jianshu.com/p/93f96e7538da" target="_blank" rel="noopener">参考</a></p>
</blockquote>
<h5 id="hisat2-下载index"><a href="#hisat2-下载index" class="headerlink" title="hisat2 下载index"></a>hisat2 下载index</h5><blockquote>
<p><a href="http://www.biotrainee.com/thread-26-1-1.html" target="_blank" rel="noopener">bowtie原理</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/hg19.tar.gz</span><br><span class="line">wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/mm10.tar.gz</span><br><span class="line">tar -zxvf *.tar.gz</span><br></pre></td></tr></table></figure></p>
</blockquote>
<h5 id="hiat2-Map，sam转bam，name-sort"><a href="#hiat2-Map，sam转bam，name-sort" class="headerlink" title="hiat2 Map，sam转bam，name-sort"></a>hiat2 Map，sam转bam，name-sort</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">#! usr/bin/bash</span><br><span class="line">set -u</span><br><span class="line">set -e</span><br><span class="line">set -o pipefail</span><br><span class="line">hg19_ref=~/fastq/hisat2_index/hg19/genome</span><br><span class="line">mm10_ref=~/fastq/hisat2_index/mm10/genome</span><br><span class="line">hg19_gff=~/fastq/refdata/Genecode/gencode.v26lift37.annotation.gff3</span><br><span class="line">mm10_gff=~/fastq/refdata/Genecode/gencode.vM10.annotation.gtf</span><br><span class="line">data_path=~/fastq</span><br><span class="line">ls Homo*.fastq.gz|while read id;do(hisat2 -t -p 5 -x $hg19_ref -1 $data_path/$&#123;id%_*&#125;_1.fastq.gz -2 $data_path/$&#123;id%_*&#125;_2.fastq.gz 2&gt;$&#123;id%_*&#125;_map.log | /home/tingm/software/samtools-1.3.1/samtools view -Sb  - &gt;$&#123;id%_*&#125;.bam);done</span><br><span class="line">ls Mus*.fastq.gz|while read id;do(hisat2 -t -p 5 -x $mm10_ref -1 $data_path/$&#123;id%_*&#125;_1.fastq.gz -2 $data_path/$&#123;id%_*&#125;_2.fastq.gz 2&gt;$&#123;id%_*&#125;_map.log | /home/tingm/software/samtools-1.3.1/samtools view -Sb  -  &gt;$&#123;id%_*&#125;.bam);done</span><br><span class="line">ls *.bam|while read id;do(/home/tingm/software/samtools-1.3.1/samtools sort -n -@ 5 $id  -o $&#123;id%%.*&#125;.Nsorted.bam);done</span><br><span class="line">#ls Homo*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $hg19_gff &gt; $&#123;id%%.*&#125;.geneCounts 2&gt; $&#123;id%%.*&#125;.HTseq.log);done</span><br><span class="line">#ls Mus*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $mm10_gff &gt; $&#123;id%%.*&#125;.geneCounts 2&gt; $&#123;id%%.*&#125;.HTseq.log);done</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-614c3461111f4783.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="bam"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-14e7421d4dbb7c5c.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="name-sorted bam"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-ffab566ae8f0b434.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="sorted bam"></p>
<h5 id="BWA"><a href="#BWA" class="headerlink" title="BWA"></a>BWA</h5><p><a href="https://starsyi.github.io/2016/05/24/BWA-%E5%91%BD%E4%BB%A4%E8%AF%A6%E8%A7%A3/" target="_blank" rel="noopener">中文参考</a></p>
<h5 id="Samtools"><a href="#Samtools" class="headerlink" title="Samtools"></a>Samtools</h5><p><a href="http://samtools.sourceforge.net/" target="_blank" rel="noopener">官方网址</a><br><a href="http://www.bio-info-trainee.com/518.html" target="_blank" rel="noopener">下载参考</a><br><a href="http://blog.csdn.net/sinat_38163598/article/details/72910115" target="_blank" rel="noopener">中文教程</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">wget https://github.com/samtools/samtools/releases/download/1.5/samtools-1.5.tar.bz2</span><br><span class="line">tar jxf samtools-1.5.tar.bz2</span><br><span class="line">cd samtools-1.5</span><br><span class="line">make</span><br><span class="line">echo &apos;PATH=$PATH:~/biosoft/samtools-1.5/&apos; &gt;&gt; ~/.bashrc</span><br><span class="line">source ~/.bashrc</span><br></pre></td></tr></table></figure></p>
<h5 id="两种排序方式及索引-默认排序是根据染色体位置-n-reads名称进行排序"><a href="#两种排序方式及索引-默认排序是根据染色体位置-n-reads名称进行排序" class="headerlink" title="两种排序方式及索引(默认排序是根据染色体位置 -n reads名称进行排序)"></a>两种排序方式及索引(默认排序是根据染色体位置 -n reads名称进行排序)</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">ls *.sra.bam|while read id;do(/home/tingm/software/samtools-1.3.1/samtools sort -@ 5 $id -o $&#123;id%%.*&#125;.sorted.bam);done</span><br><span class="line">ls *.sorted.bam|while read id;do(/home/tingm/software/samtools-1.3.1/samtools index $&#123;id%%.*&#125;.sorted.bam;done</span><br><span class="line">#head -1000 Homo_SRR3589956.sra.bam &gt; test.bam</span><br><span class="line">#samtools view test.bam | head</span><br><span class="line">#head -1000 Homo_SRR3589956.Nsorted.bam &gt; test.nsorted.bam</span><br><span class="line">#samtools view test.nsorted.bam | head</span><br><span class="line">#head -1000 Homo_SRR3589956.sorted.bam &gt; test.sorted.bam</span><br><span class="line">#samtools view test.sorted.bam | head</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-1126ea86c945546c.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="bam"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-65746a73521c7f0b.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="reads name"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-f50f20d61159b837.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="chr"></p>
<h5 id="IGV-载入bam文件及其index-如何查看，理解结果？"><a href="#IGV-载入bam文件及其index-如何查看，理解结果？" class="headerlink" title="IGV 载入bam文件及其index  如何查看，理解结果？"></a>IGV 载入bam文件及其index  如何查看，理解结果？</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">igv.sh</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-f0e2a6fa977e4068.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="sort.bam"></p>
<h5 id="质量控制"><a href="#质量控制" class="headerlink" title="质量控制"></a>质量控制</h5><blockquote>
<p>常用工具有<br>Picard <a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener">https://broadinstitute.github.io/picard/</a><br>RSeQC <a href="http://rseqc.sourceforge.net/" target="_blank" rel="noopener">http://rseqc.sourceforge.net/</a><br>Qualimap <a href="http://qualimap.bioinfo.cipf.es/" target="_blank" rel="noopener">http://qualimap.bioinfo.cipf.es/</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"># Python2.7环境下</span><br><span class="line">pip install RSeQC</span><br><span class="line"># 先对bam文件进行统计分析， 从结果上看是符合70~90的比对率要求</span><br><span class="line">bam_stat.py -i Homo_SRR3589956.sorted.bam</span><br><span class="line"># 基因组覆盖率的QC需要提供bed文件，可以直接RSeQC的网站下载，或者可以用gtf转换</span><br><span class="line">wget https://sourceforge.net/projects/rseqc/files/BED/Human_Homo_sapiens/hg19_RefSeq.bed.gz/download -O hg19_RefSeq.bed.gz</span><br><span class="line">gzip -d hg19_RefSeq.bed.gz</span><br><span class="line">read_distribution.py -i Homo_SRR3589956.sorted.bam -r ~/fastq/refdata/hg19_RefSeq.bed</span><br></pre></td></tr></table></figure></p>
</blockquote>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-43015f11229d4941.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="bam文件进行统计分析"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-0ae3b95e94e3037b.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="基因组覆盖率的QC"></p>
<blockquote>
<p>参考资料<br><a href="https://mp.weixin.qq.com/s/C66OKH08mjWvHOmq-s8V1A" target="_blank" rel="noopener">PANDA姐的转录组入门(5)：序列比对</a><br><a href="http://www.jianshu.com/p/681e02e7f9af" target="_blank" rel="noopener">hoptop 转录组入门（5）： 序列比对</a><br><a href="http://www.jianshu.com/p/93f96e7538da" target="_blank" rel="noopener">lxmic转录组入门（5）： 序列比对</a></p>
</blockquote>
<h5 id="Picard"><a href="#Picard" class="headerlink" title="Picard"></a>Picard</h5><p><a href="https://github.com/broadinstitute/picard/releases/tag/2.11.0" target="_blank" rel="noopener">下载参考2</a><br><a href="https://github.com/broadinstitute/picard/releases/tag/1.141" target="_blank" rel="noopener">下载版本较旧 因需java1.8 放弃旧版本</a><br><a href="https://github.com/broadinstitute/picard" target="_blank" rel="noopener">下载参考1 java需1.8</a><br><a href="https://java.com/zh_CN/download/help/linux_x64_install.xml#download" target="_blank" rel="noopener">java8下载</a><br><a href="http://www.cnblogs.com/OA-maque/p/4835040.html" target="_blank" rel="noopener">下载参考</a><br><a href="http://www.bioinfo-scrounger.com/archives/183" target="_blank" rel="noopener">终版下载参考</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">wget https://github.com/broadinstitute/picard/releases/download/2.11.0/picard.jar</span><br><span class="line">java -jar picard.jar -h</span><br></pre></td></tr></table></figure></p>
<p><a href="https://cncbi.github.io/Picard-Manual-CN/" target="_blank" rel="noopener">使用参考</a><br><a href="http://blog.sina.com.cn/s/blog_6721167201018jik.html" target="_blank" rel="noopener">使用参考</a></p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/11/转录组入门——4/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/11/转录组入门——4/" itemprop="url">转录组入门——4</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-11T22:08:50+08:00">2017-08-11</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/转录组入门/" itemprop="url" rel="index"><span itemprop="name">转录组入门</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/11/转录组入门——4/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/11/转录组入门——4/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h4 id="参考基因组-hg19-下载"><a href="#参考基因组-hg19-下载" class="headerlink" title="参考基因组(hg19)下载"></a>参考基因组(hg19)下载</h4><blockquote>
<p>NCBI     UCSC   ENSEMBL数据库 对应关系<br>GRCh36  hg18     ENSEMBL release_52<br>GRCh37  hg19     ENSEMBL release_59/61/64/68/69/75<br>GRCh38  hg38     ENSEMBL  release_76/77/78/80/81/82</p>
</blockquote>
<h5 id="NCBI"><a href="#NCBI" class="headerlink" title="NCBI"></a>NCBI</h5><p><a href="https://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/ARCHIVE/BUILD.37.3/Assembled_chromosomes/seq/" target="_blank" rel="noopener">https://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/ARCHIVE/BUILD.37.3/Assembled_chromosomes/seq/</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">#! /bin/bash</span><br><span class="line">file=$1</span><br><span class="line">chrs=`awk &apos;&#123;print $1&#125;&apos; $file`</span><br><span class="line">for i in $&#123;chrs[@]&#125;;</span><br><span class="line">do echo $i;</span><br><span class="line">wget https://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/ARCHIVE/BUILD.37.3/Assembled_chromosomes/seq/hs_ref_GRCh37.p5_$&#123;i&#125;.fa.gz;</span><br><span class="line">done</span><br><span class="line">gunzip *.gz</span><br><span class="line">for i in $&#123;chrs[@]&#125;;</span><br><span class="line">do cat hs_ref_GRCh37.p5_$&#123;i&#125;.fa &gt;&gt; hg19.fasta;</span><br><span class="line">done</span><br><span class="line">rm -fr *.fa</span><br></pre></td></tr></table></figure></p>
<h5 id="Emsembl-过于大-应该为3G大小？"><a href="#Emsembl-过于大-应该为3G大小？" class="headerlink" title="Emsembl 过于大 应该为3G大小？"></a>Emsembl 过于大 应该为3G大小？</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">nohup wget ftp://ftp.ensembl.org/pub/release-75/fasta/homo_sapiens/dna/Homo_sapiens.GRCh37.75.dna.toplevel.fa.gz &amp;</span><br></pre></td></tr></table></figure>
<h5 id="UCSC"><a href="#UCSC" class="headerlink" title="UCSC"></a>UCSC</h5><p><a href="https://www.jianshu.com/p/f101c9238fc5" target="_blank" rel="noopener">lxmic 转录组入门（4）：了解参考基因组及基因注释</a><br><img src="http://upload-images.jianshu.io/upload_images/2116382-88c008789cd1dd51.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 1 select Downloads —&gt; select genome data"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-2e1ade36ab2e236d.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 2-1 choose hg38"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-d94fcaa75393bf9c.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 2-2 download"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-f4d4898048b1b2a5.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 3-1 choose hg19"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-9ab848763bf29e4f.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 3-2 download"><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">nohup wget http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz &amp;</span><br><span class="line">jobs</span><br><span class="line">fg %1</span><br><span class="line">bg %1</span><br><span class="line">kill %1</span><br><span class="line">tar -zxvf chromFa.tar.gz</span><br><span class="line">cat *.fa &gt; hg19.fa</span><br><span class="line">rm -rf chr*</span><br></pre></td></tr></table></figure></p>
<h4 id="参考基因组注释下载"><a href="#参考基因组注释下载" class="headerlink" title="参考基因组注释下载"></a>参考基因组注释下载</h4><h5 id="NCBI-1"><a href="#NCBI-1" class="headerlink" title="NCBI"></a>NCBI</h5><p><img src="http://upload-images.jianshu.io/upload_images/2116382-02d615eeeb235c1b.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="进入NCBI的Genome数据库"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-30b4ca74609e930f.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Download/FTP"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-75b1fba61ea288b8.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="选择物种：人"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-98122c5892131b89.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="进入下载hg38GTF注释文件"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-40eefc8f0afdd683.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="hg38GTF注释文件"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-eedc1feedef01e9a.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="其他版本"><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">nohup wget https://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/GFF/ref_GRCh38.p7_top_level.gff3.gz &amp;</span><br><span class="line">gzip -d ref_GRCh38.p7_top_level.gff3.gz</span><br></pre></td></tr></table></figure></p>
<h5 id="Ensembl"><a href="#Ensembl" class="headerlink" title="Ensembl"></a>Ensembl</h5><p><a href="http://www.bio-info-trainee.com/1469.html" target="_blank" rel="noopener">生信菜鸟团 基因组各种版本对应关系</a><br><img src="http://upload-images.jianshu.io/upload_images/2116382-78260f48c383e5d9.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Downloads"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-8e63dca8af3c1054.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="select"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-64ee6b0553666966.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="FTP site"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-45b8166f6b848095.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="release-75"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-7b9911c2b8cd58ba.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="GTF"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-27b6ef4681c23204.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Homo sapiens"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-6a2c9ddd309802a9.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="download"><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">nohup wget ftp://ftp.ensembl.org/pub/release-75/gtf/homo_sapiens/Homo_sapiens.GRCh37.75.gtf.gz &amp;</span><br><span class="line">gzip -d Homo_sapiens.GRCh37.75.gtf.gz</span><br></pre></td></tr></table></figure></p>
<h5 id="UCSC-1"><a href="#UCSC-1" class="headerlink" title="UCSC"></a>UCSC</h5><p><img src="http://upload-images.jianshu.io/upload_images/2116382-5cf747fa93b9d9b0.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="choose Table Browser"></p>
<blockquote>
<p>Navigate to <a href="http://genome.ucsc.edu/cgi-bin/hgTables" target="_blank" rel="noopener">http://genome.ucsc.edu/cgi-bin/hgTables</a><br>Select the following options:<br>clade: Mammal<br>genome: Human<br>assembly: Feb. 2009 (GRCh37/hg19)<br>group: Genes and Gene Predictions<br>track: UCSC Genes<br>table: knownGene<br>region: Select “genome” for the entire genome.<br>output format: GTF - gene transfer format<br>output file: enter a file name to save your results to a file, or leave blank to display results in the browser<br>Click ‘get output’.</p>
</blockquote>
<h5 id="Genecode"><a href="#Genecode" class="headerlink" title="Genecode"></a>Genecode</h5><p><a href="http://www.jianshu.com/p/f101c9238fc5" target="_blank" rel="noopener">转录组入门4-了解参考基因组及基因注释</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_26/GRCh37_mapping/gencode.v26lift37.annotation.gtf.gz</span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_26/GRCh37_mapping/gencode.v26lift37.annotation.gff3.gz</span><br><span class="line">gunzip *.gz </span><br><span class="line">rm -rf *.gz</span><br></pre></td></tr></table></figure></p>
<h5 id="IGV"><a href="#IGV" class="headerlink" title="IGV"></a>IGV</h5><p><a href="http://www.biotrainee.com/thread-1998-1-1.html" target="_blank" rel="noopener">转录组作业（四）：来自零基础的小白</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">![软件自带基因组-TP53](http://upload-images.jianshu.io/upload_images/2116382-35124fbd81c8bdab.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240)</span><br><span class="line">![hg19.fa 蓝色带没有变化？？](http://upload-images.jianshu.io/upload_images/2116382-75961165acf84dd2.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240)</span><br><span class="line"></span><br><span class="line">***问题：linux服务器未安装上IGV；自行下载的hg19比对结果和软件自带不一致***</span><br></pre></td></tr></table></figure></p>
<p>sudo apt-get install glib-networking-common<br>```</p>
<blockquote>
<p>其他参考资料<br><a href="http://blog.sina.com.cn/s/blog_3d37a5690100wne7.html" target="_blank" rel="noopener">ncbi中的基因组genome全序列下载 </a><br><a href="http://www.bio-info-trainee.com/1985.html" target="_blank" rel="noopener">参考基因组下载及index建立</a><br><a href="http://www.bio-info-trainee.com/1469.html" target="_blank" rel="noopener">基因组各种版本对应关系</a><br><a href="https://www.yaolibio.com/2016/09/01/retrieve-genome-data-from-ucsc/" target="_blank" rel="noopener">从UCSC获取基因组序列及基因注释</a><br><a href="http://www.jianshu.com/p/f101c9238fc5" target="_blank" rel="noopener">转录组入门4-了解参考基因组及基因注释</a><br><a href="http://www.cnblogs.com/Demo1589/p/6950196.html" target="_blank" rel="noopener">关于基因组注释文件GTF的解释</a><br><a href="http://www.jianshu.com/p/48b5a0972301" target="_blank" rel="noopener">GTF/GFF文件的差异及其看相互转换</a><br><a href="http://mp.weixin.qq.com/s/mq--yW_vmm2C2ZpnYiPq-A" target="_blank" rel="noopener">ensembel及NCBI注释文件下载</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/10/转录组入门——3/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/10/转录组入门——3/" itemprop="url">转录组入门——3</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-10T19:34:02+08:00">2017-08-10</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/转录组入门/" itemprop="url" rel="index"><span itemprop="name">转录组入门</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/10/转录组入门——3/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/10/转录组入门——3/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="SRA-Toolkit-软件"><a href="#SRA-Toolkit-软件" class="headerlink" title="SRA Toolkit 软件"></a>SRA Toolkit 软件</h5><p><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc" target="_blank" rel="noopener">官方网址</a></p>
<h5 id="软件下载（CentOS-6-8）"><a href="#软件下载（CentOS-6-8）" class="headerlink" title="软件下载（CentOS 6.8）"></a>软件下载（CentOS 6.8）</h5><p><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&amp;f=std#s-2" target="_blank" rel="noopener">官方参考</a><br><a href="http://www.jianshu.com/p/6a4855023330" target="_blank" rel="noopener">中文参考</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">#下载软件包</span><br><span class="line">wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.8.2-1/sratoolkit.2.8.2-1-centos_linux64.tar.gz</span><br><span class="line">#解压</span><br><span class="line">tar -zxf sratoolkit.2.8.2-1-centos_linux64.tar.gz </span><br><span class="line">#解压后文件夹移动到biosoft目录下</span><br><span class="line">mv sratoolkit.2.8.2-1-centos_linux64/ biosoft/</span><br><span class="line">#设置绝对路径</span><br><span class="line">echo &apos;PATH=$PATH:~/biosoft/sratoolkit.2.8.2-1-centos_linux64/bin&apos; &gt;&gt; ~/.bashrc</span><br><span class="line">source ~/.bashrc</span><br><span class="line">#测试安装</span><br><span class="line">fastq-dump -h</span><br></pre></td></tr></table></figure></p>
<h5 id="fastq-dump参数"><a href="#fastq-dump参数" class="headerlink" title="fastq-dump参数"></a>fastq-dump参数</h5><p><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&amp;f=fastq-dump" target="_blank" rel="noopener">官方参考</a><br><a href="http://www.biotrainee.com/forum.php?mod=viewthread&amp;tid=1885&amp;fromuid=1685" target="_blank" rel="noopener">中文参考</a><br>–split-3 如果是双端测序数据，则输出两个文件，如果不是则只输出一个文件<br>-O 指定输出路径<br>–gzip 指定输出格式为gzip压缩格式<br>-A –accession 输入sra文件可以是绝对路径</p>
<h5 id="SRA文件转换为fastq文件-仅关注RNA-Seq数据"><a href="#SRA文件转换为fastq文件-仅关注RNA-Seq数据" class="headerlink" title="SRA文件转换为fastq文件(仅关注RNA-Seq数据)"></a>SRA文件转换为fastq文件(仅关注RNA-Seq数据)</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">for i in `seq 56 62`;do fastq-dump --gzip --split-3 -O ~/fastq/ -A ~/ncbi/public/sra/SRR35899$&#123;i&#125;.sra;done</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-f592e2a5c22a432f.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="gz"></p>
<h5 id="生信名词理解"><a href="#生信名词理解" class="headerlink" title="生信名词理解"></a>生信名词理解</h5><p>测序reads：高通量测序平台产生的序列标签就称为reads<br>GC含量：<a href="https://zh.wikipedia.org/wiki/GC%E5%90%AB%E9%87%8F" target="_blank" rel="noopener">中文参考</a><br>质量值：每测一个碱基会有相应的质量值，质量值用于衡量测序准确度<a href="http://www.cnblogs.com/daimakun/p/5092312.html" target="_blank" rel="noopener">中文参考</a><a href="https://en.wikipedia.org/wiki/FASTQ_format" target="_blank" rel="noopener">英文参考</a><br>adapter：接头是一段短的序列已知的核酸链，用于连接序列未知的目标测序片段<br>index：索引序列，是一段很短的寡居核酸链，用于在多个样品混合测序时，标记不同的样品，为区分不同样本而构建<a href="http://www.itwendao.com/article/detail/467548.html" target="_blank" rel="noopener">参考</a></p>
<h5 id="FastQC-软件"><a href="#FastQC-软件" class="headerlink" title="FastQC 软件"></a>FastQC 软件</h5><p><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank" rel="noopener">官方网址</a></p>
<h5 id="软件下载（CentOS-6-8）-1"><a href="#软件下载（CentOS-6-8）-1" class="headerlink" title="软件下载（CentOS 6.8）"></a>软件下载（CentOS 6.8）</h5><p><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/INSTALL.txt" target="_blank" rel="noopener">官方参考</a><br><a href="http://www.jianshu.com/p/6a4855023330" target="_blank" rel="noopener">中文参考</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">java -version</span><br><span class="line">wget http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.5.zip</span><br><span class="line">unzip fastqc_v0.11.5.zip</span><br><span class="line">mv FastQC/ biosoft/</span><br><span class="line">cd ~/biosoft/FastQC</span><br><span class="line">chmod 755 fastqc</span><br><span class="line">echo &apos;PATH=$PATH:~/biosoft/FastQC/&apos; &gt;&gt; ~/.bashrc</span><br><span class="line">source ~/.bashrc</span><br><span class="line">fastqc -h</span><br></pre></td></tr></table></figure></p>
<h5 id="fastqc"><a href="#fastqc" class="headerlink" title="fastqc"></a>fastqc</h5><p>-o 输出目录，需自己创建目录<br>–extract 是否解压输出文件，默认自动解压缩zip文件<br>-f 指定输入文件的类型，支持fastq|bam|sam三种格式的文件，默认自动识别<br>-t 同时处理的文件数目<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fastqc -o fastqc_out --extract *.fastq.gz</span><br></pre></td></tr></table></figure></p>
<h5 id="fastqc质控结果"><a href="#fastqc质控结果" class="headerlink" title="fastqc质控结果"></a>fastqc质控结果</h5><p><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/" target="_blank" rel="noopener">官方参考</a><br><a href="http://www.jianshu.com/p/14fd4de54402" target="_blank" rel="noopener">中文参考</a><br><a href="http://fbb84b26.wiz03.com/share/s/3XK4IC0cm4CL22pU-r1HPcQQ2irG2836uQYm2iZAyh1Zwf3_" target="_blank" rel="noopener">中文参考</a></p>
<h5 id="批量查看MultiQC"><a href="#批量查看MultiQC" class="headerlink" title="批量查看MultiQC"></a>批量查看MultiQC</h5><p><a href="http://multiqc.info/docs/" target="_blank" rel="noopener">官方网址</a><br><a href="http://fbb84b26.wiz03.com/share/s/3XK4IC0cm4CL22pU-r1HPcQQ1iRTvV2GwkwL2AaxYi2fXHP7" target="_blank" rel="noopener">中文参考</a></p>
<h5 id="软件下载及使用"><a href="#软件下载及使用" class="headerlink" title="软件下载及使用"></a>软件下载及使用</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda install -c bioconda multiqc</span><br><span class="line">#进入对应文件，扫描zip文件，忽略html文件</span><br><span class="line">multiqc ~/fastq/fastqc_out/*fastqc.zip --ignore *.html</span><br></pre></td></tr></table></figure>
<h5 id="结果查看-重点关注的部分参考"><a href="#结果查看-重点关注的部分参考" class="headerlink" title="结果查看 重点关注的部分参考"></a>结果查看 重点关注的部分参考</h5><p><a href="https://zhuanlan.zhihu.com/p/27985781" target="_blank" rel="noopener">青山屋主 转录组入门3-测序数据质量检查</a></p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/09/转录组入门——2/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/09/转录组入门——2/" itemprop="url">转录组入门——2</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-09T18:58:42+08:00">2017-08-09</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/转录组入门/" itemprop="url" rel="index"><span itemprop="name">转录组入门</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/09/转录组入门——2/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/09/转录组入门——2/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="CentOS-release-6-5-Final"><a href="#CentOS-release-6-5-Final" class="headerlink" title="CentOS release 6.5 (Final)"></a>CentOS release 6.5 (Final)</h5><h5 id="GEO-SRA数据库的数据存放形式参考"><a href="#GEO-SRA数据库的数据存放形式参考" class="headerlink" title="GEO/SRA数据库的数据存放形式参考"></a>GEO/SRA数据库的数据存放形式参考</h5><p><a href="http://www.biotrainee.com/forum.php?mod=viewthread&amp;tid=1884&amp;fromuid=1685" target="_blank" rel="noopener">青山屋主 转录组入门(2)-作业</a><br><a href="http://www.biotrainee.com/forum.php?mod=viewthread&amp;tid=1829&amp;fromuid=1685" target="_blank" rel="noopener">hoptop 转录组入门（二）读文献下数据</a></p>
<h5 id="软件及参数"><a href="#软件及参数" class="headerlink" title="软件及参数"></a>软件及参数</h5><p>比对：TopHat (v2.0.13) 参考基因组：GRCh37/hg19 GTF文件：GTF version GRCh37.70 去除MQ&gt;30的reads<br>计算平均插入大小和标准差：Picard-tools（v1.126）<br>计算reads count：HTSeq (v0.6.0)<br>差异表达：DESeq (v3.0) 差异外显子：DEXSeq (v3.1)<br>normalize Bigwig files to per million reads：BEDTools (v2.17.0), bedGraphToBigWig tool (v4)<br>统计分析和作图： R (v3.1.1) (<a href="http://www.r-project.org/" target="_blank" rel="noopener">http://www.r-project.org/</a>)<br>GO富集分析：DAVID (<a href="http://david.ncifcrf.gov/" target="_blank" rel="noopener">http://david.ncifcrf.gov/</a>)</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"># download Aspera</span><br><span class="line">wget http://download.asperasoft.com/download/sw/connect/3.6.2/aspera-connect-3.6.2.117442-linux-64.tar.gz</span><br><span class="line">tar zxf aspera-connect-3.6.2.117442-linux-64.tar.gz</span><br><span class="line">sh aspera-connect-3.6.2.117442-linux-64.sh</span><br><span class="line">echo &apos;PATH=$PATH:/home/tma/.aspera/connect/bin&apos; &gt;&gt; ~/.bashrc</span><br><span class="line">source ~/.bashrc</span><br><span class="line">ascp</span><br><span class="line"># download data</span><br><span class="line">for x in $(seq 48 62);do ascp -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp-private.ncbi.nlm.nih.gov --user=anonftp --mode=recv /sra/sra-instant/reads/ByStudy/sra/SRP/SRP075/SRP075747/SRR35899&quot;$x&quot;/SRR35899&quot;$x&quot;.sra . ;done</span><br><span class="line"></span><br><span class="line"># sratoolkit prefetch</span><br><span class="line">for x in $(seq 48 62);do prefetch SRR35899&quot;$x&quot;;done</span><br></pre></td></tr></table></figure>
      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/09/转录组入门——1/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/09/转录组入门——1/" itemprop="url">转录组入门——1</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-09T16:22:38+08:00">2017-08-09</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/转录组入门/" itemprop="url" rel="index"><span itemprop="name">转录组入门</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/09/转录组入门——1/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/09/转录组入门——1/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="系统环境"><a href="#系统环境" class="headerlink" title="系统环境"></a>系统环境</h5><p>CentOS 6.8（虚拟机之前就已安装好，所以不是推荐安装的ubuntu）</p>
<h5 id="软件安装参考"><a href="#软件安装参考" class="headerlink" title="软件安装参考"></a>软件安装参考</h5><p><a href="http://www.biotrainee.com/thread-1742-1-1.html" target="_blank" rel="noopener">Jimmy转录组入门(1)：计算机资源的准备</a><br><a href="http://www.biotrainee.com/thread-1796-1-1.html" target="_blank" rel="noopener">青山屋主转录组入门(1)-作业</a><br><a href="http://www.chenlianfu.com/?p=2438" target="_blank" rel="noopener">HTSeq中文教程</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br></pre></td><td class="code"><pre><span class="line"># download miniconda2</span><br><span class="line">wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh </span><br><span class="line">bash Miniconda2-latest-Linux-x86_64.sh</span><br><span class="line">source ~/.bashrc</span><br><span class="line"></span><br><span class="line"># sratoolkit</span><br><span class="line">conda install -c jfear sratoolkit=2.8.1</span><br><span class="line">fastq-dump</span><br><span class="line"></span><br><span class="line"># fastqc</span><br><span class="line">conda install -c bioconda fastqc=0.11.5</span><br><span class="line">fastqc</span><br><span class="line"></span><br><span class="line"># hisat2</span><br><span class="line">conda install -c bioconda hisat2</span><br><span class="line">hisat2</span><br><span class="line"></span><br><span class="line"># samtools</span><br><span class="line">conda install -c bioconda samtools=1.5 </span><br><span class="line"># error</span><br><span class="line"># samtools: error while loading shared libraries: libbz2.so.1.0: cannot open shared object file: No such file or directory</span><br><span class="line"># 错误解决参考 https://www.biostars.org/p/266111/</span><br><span class="line">conda install -c conda-forge -c bioconda samtools bzip2</span><br><span class="line">samtools</span><br><span class="line"></span><br><span class="line"># htseq-count</span><br><span class="line">conda install -c bioconda htseq=0.7.2</span><br><span class="line">python</span><br><span class="line">&gt;&gt;&gt;import HTSeq</span><br><span class="line"></span><br><span class="line"># R</span><br><span class="line">sudo yum install epel-release</span><br><span class="line">sudo yum install R</span><br><span class="line">R</span><br><span class="line"></span><br><span class="line"># Rstudio</span><br><span class="line">conda install rstudio</span><br><span class="line">rstudio</span><br></pre></td></tr></table></figure></p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  
  <nav class="pagination">
    <a class="extend prev" rel="prev" href="/page/2/"><i class="fas fa-angle-left"></i></a><a class="page-number" href="/">1</a><a class="page-number" href="/page/2/">2</a><span class="page-number current">3</span>
  </nav>
  
  
  

<div class="comments" id="comments">
  
  
  <script type="text/javascript">
  _hcwp = window._hcwp || [];
  _hcwp.push({ "widget": "Bloggerstream", "widget_id": "tingm831@gmail.com", "selector": ".hc-comment-count", "label": "{\%COUNT%\}" });
  
  (function() {
    if ("HC_LOAD_INIT" in window) {
      return;
    }
    HC_LOAD_INIT = true;
    var lang = (navigator.language || navigator.systemLanguage || navigator.userLanguage || "en").substr(0, 2).toLowerCase();
    var hcc = document.createElement("script");
    hcc.type = "text/javascript";
    hcc.defer = true;
    hcc.src = ("https:" === document.location.protocol ? "https" : "http") + "://w.hypercomments.com/widget/hc/tingm831@gmail.com/" + lang + "/widget.js";
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(hcc, s.nextSibling);
  })();
  </script>
  
</div>



  
</section>

          </div>
          
          
          
<aside class="sidebar" id="sidebar">
  
  <div class="search">
    <div class="form-group">
      <i class="fas fa-search"></i> <input type="text" id="search-input" name="q" results="0" placeholder="Search" class="form-control"/>
    </div>
  </div>
  <div class="search-result-box" id="search-result"></div>
  
  
<div class="info sidebar-item" id="info">
  
  <img class="author-avatar" src="/images/avatar.png" alt="mt">
  
  <p class="author-name">mt</p>
  <p class="author-description"> </p>
  <div class="site-count">
    
    <div class="archives-count">
      <p>Archives</p>
      <p><a href="/archives/">26</a></p>
    </div>
    
    
    
    <span class="site-count-divider"> | </span>
    
    <div class="categories-count">
      <p>Categories</p>
      <p><a href="/categories/">3</a></p>
    </div>
    
    
    
    <span class="site-count-divider"> | </span>
    
    <div class="tags-count">
      <p>Tags</p>
      <p><a href="/tags/">2</a></p>
    </div>
    
  </div>
  
  <div class="rss">
    <a class="rss-link button sidebar-item" href="/atom.xml"><i class="fas fa-rss"></i> RSS</a>
  </div>
  
</div>


  <div class="sidebar-sticky">
    
    
    
<hr>
<div class="social-link sidebar-item">
  <p><i class="far fa-address-card"></i> Social Links</p>
  <ul>
    
    <li><i class="fab fa-github"></i> <a href="https://github.com/mt831" target="_blank">GitHub</a></li>
    
  </ul>
</div>


    
    
  </div>
</aside>


          
        </div>
      </div>
    </main>
    
<footer id="footer" class="footer">
  <div class="container">
    <div class="back-to-top">
      <a id="back-to-top"><i class="fas fa-angle-double-up"></i></a>
    </div>
    <div class="footer-container">
      <div class="footer-left">
        <div class="copyright">
          <span class="author">mt</span>
          <span class="year"><i class="far fa-copyright"></i> 2018-4-1 - 2018</span>
          
        </div>
        
        
<div class="busuanzi">
  <span id="busuanzi_container_site_pv"><i class="fas fa-eye"></i> <span id="busuanzi_value_site_pv"></span></span>
  <span id="busuanzi_container_site_uv"><i class="fas fa-user"></i> <span id="busuanzi_value_site_uv"></span></span>
  <span id="busuanzi_container_page_pv"><i class="far fa-file-alt"></i> <span id="busuanzi_value_page_pv"></span></span>
</div>


        
      </div>
      <div class="footer-right">
        <div class="custom-info">
          
        </div>
        <div class="powered-by">
          Proudly Powered by <a href="https://hexo.io/" target="_blank">Hexo</a> | Theme is <a href="https://github.com/AlynxZhou/hexo-theme-aria/" target="_blank">ARIA</a>
        </div>
      </div>
    </div>
  </div>
</footer>


  </body>
</html>
