<!DOCTYPE html><html lang="zh-CN"><head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"><meta content="yes" name="apple-mobile-web-app-capable"><meta content="black-translucent" name="apple-mobile-web-app-status-bar-style"><meta content="telephone=no" name="format-detection"><meta name="description"><title>star.gatk.sh | mt's blog</title><link rel="stylesheet" type="text/css" href="//fonts.neworld.org/css?family=Source+Code+Pro"><link rel="stylesheet" type="text/css" href="/css/style.css?v=2.0.1"><link rel="stylesheet" type="text/css" href="/css/highlight.css?v=2.0.1"><link rel="Shortcut Icon" href="/favicon.ico"><link rel="bookmark" href="/favicon.ico"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="apple-touch-icon-precomposed" href="/apple-touch-icon.png"><link rel="alternate" type="application/atom+xml" href="/atom.xml"></head><body><div class="body_container"><div id="header"><div class="site-name"><h1 class="hidden">star.gatk.sh</h1><a id="logo" href="/.">mt's blog</a><p class="description">学习记录</p></div><div id="nav-menu"><a href="/." class="current"><i class="fa fa-home"> 首页</i></a><a href="/archives/"><i class="fa fa-archive"> 归档</i></a><a href="/about/"><i class="fa fa-user"> 关于</i></a><a href="/atom.xml"><i class="fa fa-rss"> 订阅</i></a></div><div id="search-form"><div id="result-mask" class="hide"></div><label><input id="search-key" type="text" autocomplete="off" placeholder="Arama"></label><div id="result-wrap" class="hide"><div id="search-result"></div></div><div class="hide"><template id="search-tpl"><div class="item"><a href="/{path}" title="{title}"><div class="title">{title}</div><div class="time">{date}</div><div class="tags">{tags}</div></a></div></template></div></div></div><div id="layout" class="layout-g"><div class="layout-l"><div class="content_container"><div class="post"><h1 class="post-title">star.gatk.sh</h1><div class="post-meta"><a href="/2018/06/14/project/star.gatk/#comments" class="comment-count"></a><p><span class="date">Jun 14, 2018</span><span><a href="/categories/project/" class="category">project</a></span><span><i id="busuanzi_container_page_pv"><i id="busuanzi_value_page_pv"></i><i>点击</i></i></span></p></div><div class="post-content"><p>gatk call SNV 为金标准，然而时间较长。GATK目前为4.0版本。<br>目前商业软件<a href="https://www.sentieon.com/products/" target="_blank" rel="noopener">Sentieon</a>准确度匹配GATK，且耗时少。<br><a href="http://www.biotrainee.com/thread-2207-1-1.html" target="_blank" rel="noopener">samtools mpileup + bcftools</a>，用最大似然，比GATK贝叶斯法快不少。<br><a href="http://kaopubear.top/2017-10-16-variantbasic4.html" target="_blank" rel="noopener">variant分析阶段小结4-使用RNA-seq数据</a></p>
<h3 id="star-gatk-sh"><a href="#star-gatk-sh" class="headerlink" title="star.gatk.sh"></a>star.gatk.sh</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br></pre></td><td class="code"><pre><span class="line">#! /bin/bash</span><br><span class="line"></span><br><span class="line">file=$1</span><br><span class="line">samples=`awk &apos;&#123;print $1&#125;&apos; $file`</span><br><span class="line">path_data=/home/tingm/rna_editing/FL-HCC/raw_data</span><br><span class="line">path_results=/home/tingm/rna_editing/FL-HCC/results_star</span><br><span class="line">path_star=/home/yyjiang/software/STAR-2.5.3a/bin/Linux_x86_64</span><br><span class="line">path_gatk=/home/tingm/software</span><br><span class="line">path_snpEff=/home/tingm/software/snpEff</span><br><span class="line"></span><br><span class="line">## download reference fa &amp; gtf file</span><br><span class="line">#cd /home/tingm/my_refdata</span><br><span class="line">#wget http://labshare.cshl.edu/shares/gingeraslab/www-data/dobin/$&#123;path_star&#125;/$&#123;path_star&#125;genomes/GENCODE/GRCh38_Gencode26/GRCh38.primary_assembly.genome.fa</span><br><span class="line">#wget http://labshare.cshl.edu/shares/gingeraslab/www-data/dobin/$&#123;path_star&#125;/$&#123;path_star&#125;genomes/GENCODE/GRCh38_Gencode26/gencode.v26.primary_assembly.annotation.gtf</span><br><span class="line"></span><br><span class="line">## star build index length=50-1</span><br><span class="line">genomeDir=/home/tingm/my_refdata/star_index_49</span><br><span class="line">genomeFastaFiles=/home/tingm/my_refdata/Homo_sapiens_assembly38.fasta</span><br><span class="line">sjdbGTFfile=/home/tingm/my_refdata/gencode.v26.primary_assembly.annotation.gtf</span><br><span class="line">dbsnp_vcf=/home/tingm/my_refdata/dbsnp_146.hg38.vcf</span><br><span class="line">#mkdir $genomeDir</span><br><span class="line">#$&#123;path_star&#125; --runThreadN 6 --runMode genomeGenerate \</span><br><span class="line">#     --genomeDir $genomeDir \</span><br><span class="line">#     --genomeFastaFiles $genomeFastaFiles \</span><br><span class="line">#     --sjdbGTFfile $sjdbGTFfile \</span><br><span class="line">#     --sjdbOverhang 49</span><br><span class="line"></span><br><span class="line">#runDir1=/home/tingm/rna_editing/FL-HCC/results_star/1pass</span><br><span class="line">#mkdir -p $runDir1</span><br><span class="line">runDir2=/home/tingm/rna_editing/FL-HCC/results_star/2pass</span><br><span class="line">#mkdir -p $runDir2</span><br><span class="line">#mkdir -p /home/tingm/rna_editing/FL-HCC/results_star/index_2-pass</span><br><span class="line"></span><br><span class="line">for sample in $&#123;samples[@]&#125;;do</span><br><span class="line"></span><br><span class="line">        xx=$(date +&quot;%y.%m.%d:%H:%M&quot;)</span><br><span class="line">        echo $sample Start calling SNVs in $xx</span><br><span class="line"></span><br><span class="line">        ## Read mapping (1st pass)</span><br><span class="line">#        cd $runDir1</span><br><span class="line">#        $&#123;path_star&#125;/STAR --runThreadN 8 --genomeDir $genomeDir --readFilesCommand zcat --readFilesIn $&#123;path_data&#125;/$&#123;sample&#125;_1.fq.gz $&#123;path_data&#125;/$&#123;sample&#125;_2.fq.gz --outFileNamePrefix ./$&#123;sample&#125;</span><br><span class="line"></span><br><span class="line">        ## Rebuild genome index</span><br><span class="line">#        $&#123;path_star&#125;/STAR --runThreadN 8 --runMode genomeGenerate --genomeDir /home/tingm/rna_editing/FL-HCC/results_star/index_2-pass --genomeFastaFiles $genomeFastaFiles --sjdbGTFfile $sjdbGTFfile --sjdbFileChrStartEnd $runDir1/$&#123;sample&#125;SJ.out.tab --sjdbOverhang 49</span><br><span class="line"></span><br><span class="line">        ## Read mapping (2nd pass)</span><br><span class="line">#        cd $runDir2</span><br><span class="line">#        $&#123;path_star&#125;/STAR --runThreadN 8 --genomeDir /home/tingm/rna_editing/FL-HCC/results_star/index_2-pass --readFilesCommand zcat --readFilesIn $&#123;path_data&#125;/$&#123;sample&#125;_1.fq.gz $&#123;path_data&#125;/$&#123;sample&#125;_2.fq.gz --outFileNamePrefix ./$&#123;sample&#125;</span><br><span class="line">#        rm -rf $&#123;runDir1&#125;/*</span><br><span class="line">#        rm -rf /home/tingm/rna_editing/FL-HCC/results_star/index_2-pass/*</span><br><span class="line"></span><br><span class="line">        ## Adding read group information, sorting, marking duplicates and indexing gatk4.0</span><br><span class="line">        cd $path_results</span><br><span class="line">        gatk --java-options &quot;-Xmx20G&quot; AddOrReplaceReadGroups -I $&#123;runDir2&#125;/$&#123;sample&#125;Aligned.out.sam -O $&#123;path_results&#125;/$&#123;sample&#125;_added_sorted.bam -SO coordinate -ID $&#123;sample&#125; -LB rna -PL illumina -PU HiSeq2500 -SM $&#123;sample&#125;</span><br><span class="line">        gatk --java-options &quot;-Xmx20G&quot; MarkDuplicates -I $&#123;path_results&#125;/$&#123;sample&#125;_added_sorted.bam -O $&#123;path_results&#125;/$&#123;sample&#125;_dedupped.bam --CREATE_INDEX true --VALIDATION_STRINGENCY SILENT -M $&#123;path_results&#125;/$&#123;sample&#125;_dupstats</span><br><span class="line"></span><br><span class="line">        # samtools faidx .fa</span><br><span class="line">        # samtools dict .fa</span><br><span class="line"></span><br><span class="line">        ## Split &apos;N&apos; Trim gatk3.7</span><br><span class="line">        java -Xmx20G -jar $&#123;path_gatk&#125;/GenomeAnalysisTK.jar -T SplitNCigarReads -R $genomeFastaFiles -I $&#123;path_results&#125;/$&#123;sample&#125;_dedupped.bam -o $&#123;path_results&#125;/$&#123;sample&#125;_split.bam -rf ReassignOneMappingQuality -RMQF 255 -RMQT 60 -U ALLOW_N_CIGAR_READS</span><br><span class="line">#        rm -rf $&#123;runDir2&#125;/$&#123;sample&#125;Aligned.out.sam</span><br><span class="line">        rm -rf $&#123;path_results&#125;/$&#123;sample&#125;_added_sorted.bam</span><br><span class="line"></span><br><span class="line">        ## Variant calling gatk3.7 This step takes a long time</span><br><span class="line">        #java -Xmx20G -jar $&#123;path_gatk&#125;/GenomeAnalysisTK.jar -T HaplotypeCaller -R $genomeFastaFiles -I $&#123;path_results&#125;/$&#123;sample&#125;_split.bam -dontUseSoftClippedBases -stand_call_conf 20.0 -D $&#123;dbsnp_vcf&#125; -o $&#123;path_results&#125;/$&#123;sample&#125;.vcf</span><br><span class="line">        java -Xmx20G -jar $&#123;path_gatk&#125;/GenomeAnalysisTK.jar -T HaplotypeCaller -R $genomeFastaFiles -I $&#123;path_results&#125;/$&#123;sample&#125;_split.bam -dontUseSoftClippedBases -stand_call_conf 20.0 -o $&#123;path_results&#125;/$&#123;sample&#125;.vcf</span><br><span class="line">        rm -rf $&#123;path_results&#125;/*.bai</span><br><span class="line"></span><br><span class="line">        ## Variant filtering gatk3.7</span><br><span class="line">        java -Xmx20G -jar $&#123;path_gatk&#125;/GenomeAnalysisTK.jar -T VariantFiltration -R $genomeFastaFiles -V $&#123;path_results&#125;/$&#123;sample&#125;.vcf -window 35 -cluster 3 -filterName FS -filter &quot;FS &gt; 30.0&quot; -filterName QD -filter &quot;QD &lt; 2.0&quot; -o $&#123;path_results&#125;/$&#123;sample&#125;.filt.vcf</span><br><span class="line">        rm -rf $&#123;path_results&#125;/*.idx</span><br><span class="line"></span><br><span class="line">        xx=$(date +&quot;%y.%m.%d:%H:%M&quot;)</span><br><span class="line">        echo $sample End calling SNVs in $xx</span><br><span class="line"></span><br><span class="line">done</span><br><span class="line"></span><br><span class="line">#########################################################################</span><br><span class="line"># reference website</span><br><span class="line">## https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5373595/ Table1.RNA-seq variant calling pipeline used in this work.</span><br><span class="line">## https://gatkforums.broadinstitute.org/gatk/discussion/3891/calling-variants-in-rnaseq 流程官方参考</span><br><span class="line">## http://www.bioinfo-scrounger.com/archives/311 流程中文参考-1</span><br><span class="line">## https://www.wxwenku.com/d/100689018 流程中文参考-2</span><br><span class="line">## http://www.huangshujia.me/2017/09/19/2017-09-19-Begining-WGS-Data-Analysis-The-pipeline.html WGS流程概念理解</span><br><span class="line">## https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_haplotypecaller_HaplotypeCaller.php HaplotypeCaller官方参考</span><br><span class="line">## https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5547479/ HaplotypeCaller参数 非Edits位点参数设置</span><br><span class="line">## http://www.biotrainee.com/thread-1417-1-1.html HaplotypeCaller参数中文参考</span><br><span class="line">## https://zhuanlan.zhihu.com/p/34878471 Variant filtering参数设置</span><br><span class="line">## https://www.biostars.org/p/84686/ zero-base</span><br><span class="line">#########################################################################</span><br><span class="line">## http://www.codexiu.cn/openstack/blog/14713/ annovar</span><br><span class="line"># annovar</span><br><span class="line">#path_annovar=/home/tingm/software/annovar/annovar</span><br><span class="line">#perl $&#123;path_annovar&#125;/convert2annovar.pl -format vcf4 SRR.filt.vcf &gt; SRR.avinput</span><br><span class="line">#perl $&#123;path_annovar&#125;/table_annovar.pl SRR.avinput /home/tingm/software/annovar/annovar/humandb/ -buildver hg19 -out anno -remove -protocol refGene,cytoBand,genomicSuperDups,esp6500siv2_all,1000g2014oct_all,1000g2014oct_afr,1000g2014oct_eas,1000g2014oct_eur,snp138,ljb26_all -operation g,r,r,f,f,f,f,f,f,f -nastring . -csvout</span><br><span class="line">#perl $&#123;path_annovar&#125;/table_annovar.pl SRR.avinput /home/tingm/software/annovar/annovar/humandb38/ -buildver hg38 -out SRR.anno -remove -protocol refGene,avsnp147 -operation g,f -nastring . -csvout</span><br><span class="line">#########################################################################</span><br></pre></td></tr></table></figure>
<h3 id="log-out"><a href="#log-out" class="headerlink" title="log.out"></a>log.out</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br></pre></td><td class="code"><pre><span class="line">SRR Start calling SNVs in 18.06.09:19:32</span><br><span class="line">Jun 09 19:32:40 ..... started STAR run</span><br><span class="line">Jun 09 19:32:40 ..... loading genome</span><br><span class="line">Jun 09 19:37:15 ..... started mapping</span><br><span class="line">Jun 09 19:45:24 ..... finished successfully</span><br><span class="line">Jun 09 19:45:24 ..... started STAR run</span><br><span class="line">Jun 09 19:45:24 ... starting to generate Genome files</span><br><span class="line">Jun 09 19:47:11 ... starting to sort Suffix Array. This may take a long time...</span><br><span class="line">Jun 09 19:47:50 ... sorting Suffix Array chunks and saving them to disk...</span><br><span class="line">Jun 09 20:30:58 ... loading chunks from disk, packing SA...</span><br><span class="line">Jun 09 20:33:11 ... finished generating suffix array</span><br><span class="line">Jun 09 20:33:11 ... generating Suffix Array index</span><br><span class="line">Jun 09 20:38:56 ... completed Suffix Array index</span><br><span class="line">Jun 09 20:38:56 ..... processing annotations GTF</span><br><span class="line">Jun 09 20:39:29 ..... inserting junctions into the genome indices</span><br><span class="line">Jun 09 20:43:35 ... writing Genome to disk ...</span><br><span class="line">Jun 09 20:43:40 ... writing Suffix Array to disk ...</span><br><span class="line">Jun 09 20:44:05 ... writing SAindex to disk</span><br><span class="line">Jun 09 20:44:06 ..... finished successfully</span><br><span class="line">Jun 09 20:44:06 ..... started STAR run</span><br><span class="line">Jun 09 20:44:06 ..... loading genome</span><br><span class="line">Jun 09 20:44:32 ..... started mapping</span><br><span class="line">Jun 09 20:52:27 ..... finished successfully</span><br><span class="line">Tool returned:</span><br><span class="line">0</span><br><span class="line">Tool returned:</span><br><span class="line">0</span><br><span class="line">There were no warn messages.</span><br><span class="line">There were 4 WARN messages, the first 4 are repeated below.</span><br><span class="line">WARN  00:04:21,914 InbreedingCoeff - Annotation will not be calculated. InbreedingCoeff requires at least 10 unrelated samples.</span><br><span class="line">WARN  00:04:24,522 HaplotypeScore - Annotation will not be calculated, must be called from UnifiedGenotyper, not org.broadinstitute.gatk.tools.walkers.haplotypecaller.HaplotypeCaller</span><br><span class="line">WARN  00:19:43,412 HaplotypeCallerGenotypingEngine - location chr1:244056927-244056928: too many alternative alleles found (7) larger than the maximum requested with -maxAltAlleles (6), the following will be dropped: CCTT.</span><br><span class="line">WARN  02:29:18,464 HaplotypeCallerGenotypingEngine - location chr16:2771012-2771013: too many alternative alleles found (8) larger than the maximum requested with -maxAltAlleles (6), the following will be dropped: C, CCTTTT.</span><br><span class="line">There were no warn messages.</span><br><span class="line">SRR End calling SNVs in 18.06.10:03:28</span><br></pre></td></tr></table></figure>
</div><div class="tags"><a href="/tags/bioinformatics/">bioinformatics</a><a href="/tags/gatk/">gatk</a><a href="/tags/star/">star</a></div><div class="post-share"><div class="bdsharebuttonbox"><span style="float:left;line-height: 28px;height: 28px;font-size:16px;font-weight:blod">分享到：</span><a href="#" data-cmd="more" class="bds_more"></a><a href="#" data-cmd="mshare" title="分享到一键分享" class="bds_mshare"></a><a href="#" data-cmd="fbook" title="分享到Facebook" class="bds_fbook"></a><a href="#" data-cmd="twi" title="分享到Twitter" class="bds_twi"></a><a href="#" data-cmd="linkedin" title="分享到linkedin" class="bds_linkedin"></a><a href="#" data-cmd="youdao" title="分享到有道云笔记" class="bds_youdao"></a><a href="#" data-cmd="evernotecn" title="分享到印象笔记" class="bds_evernotecn"></a><a href="#" data-cmd="weixin" title="分享到微信" class="bds_weixin"></a><a href="#" data-cmd="qzone" title="分享到QQ空间" class="bds_qzone"></a><a href="#" data-cmd="tsina" title="分享到新浪微博" class="bds_tsina"></a></div></div><div class="post-nav"><a href="/2018/06/14/project/ann.giremi/" class="pre">ann.giremi.sh</a><a href="/2018/06/14/project/trim.aln.count &amp; find_virus_filt/" class="next">trim.aln.count.sh &amp; find_virus_filt.sh</a></div><div id="comments"></div></div></div></div><div class="layout-r"><div id="sidebar"><div class="search-pla"></div><div id="toc" class="widget"><div class="widget-title"><i class="fa fa-fei">文章目录</i></div><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#star-gatk-sh"><span class="toc-text">star.gatk.sh</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#log-out"><span class="toc-text">log.out</span></a></li></ol></div><div class="widget"><div class="widget-title"><i class="fa fa-xie"> 最新文章</i></div><ul class="post-list"><li class="post-list-item"><a class="post-list-link" href="/2018/06/28/project/nanopolish/">nanopolish</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/20/project/RTG-tools安装/">RTG-tools安装</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/14/project/ann.giremi/">ann.giremi.sh</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/14/project/star.gatk/">star.gatk.sh</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/14/project/trim.aln.count & find_virus_filt/">trim.aln.count.sh & find_virus_filt.sh</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/04/project/giremi及gatk下载/">giremi, gatk下载使用</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/05/16/project/samtools命令/">samtools</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/05/15/Linux/Linux_Tips/">Linux Tips记录</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/05/11/project/安装picard、FastViromeExplorer使用/">安装picard、FastViromeExplorer使用</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/05/11/Linux/linux路径问题/">Linux路径问题</a></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-gui"> 分类</i></div><ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/Beginning-Perl-free-学习笔记/">Beginning Perl (free)学习笔记</a><span class="category-list-count">3</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/Linux基础/">Linux基础</a><span class="category-list-count">2</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/Test/">Test</a><span class="category-list-count">1</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/project/">project</a><span class="category-list-count">8</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/seq实战/">seq实战</a><span class="category-list-count">4</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/数据科学实训营/">数据科学实训营</a><span class="category-list-count">19</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/转录组入门/">转录组入门</a><span class="category-list-count">8</span></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-biao"> 标签</i></div><div class="tagcloud"><a href="/tags/trim-galore/" style="font-size: 15px;">trim_galore</a> <a href="/tags/bioinformatics/" style="font-size: 15px;">bioinformatics</a> <a href="/tags/Linux/" style="font-size: 15px;">Linux</a> <a href="/tags/nanopolish/" style="font-size: 15px;">nanopolish</a> <a href="/tags/samtools/" style="font-size: 15px;">samtools</a> <a href="/tags/giremi/" style="font-size: 15px;">giremi</a> <a href="/tags/gatk/" style="font-size: 15px;">gatk</a> <a href="/tags/bioinformaticsant/" style="font-size: 15px;">bioinformaticsant</a> <a href="/tags/RTG-tools/" style="font-size: 15px;">RTG-tools</a> <a href="/tags/perl/" style="font-size: 15px;">perl</a> <a href="/tags/hisat2/" style="font-size: 15px;">hisat2</a> <a href="/tags/HTSeq/" style="font-size: 15px;">HTSeq</a> <a href="/tags/FastViromeExplorer/" style="font-size: 15px;">FastViromeExplorer</a> <a href="/tags/picard/" style="font-size: 15px;">picard</a> <a href="/tags/git/" style="font-size: 15px;">git</a> <a href="/tags/star/" style="font-size: 15px;">star</a> <a href="/tags/python/" style="font-size: 15px;">python</a> <a href="/tags/SnpEff/" style="font-size: 15px;">SnpEff</a></div></div><div class="widget"><div class="widget-title"><i class="fa fa-archive"> 归档</i></div><ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">六月 2018</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/05/">五月 2018</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/04/">四月 2018</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2017/09/">九月 2017</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2017/08/">八月 2017</a></li></ul></div></div></div></div><a id="totop" href="#top"></a><div id="footer"><div class="footer-info"><p><a href="/baidusitemap.xml">Baidu Site Haritası</a> |  <a href="/atom.xml">订阅</a> |  <a href="/about/">关于</a></p><p>本站总访问量：<i id="busuanzi_container_site_pv"><i id="busuanzi_value_site_pv"></i></i>次</p><p><span> Copyright &copy;<a href="/." rel="nofollow">mt.</a></span><span> Theme by<a rel="nofollow" target="_blank" href="https://github.com/chaooo/hexo-theme-BlueLake"> BlueLake.</a></span><span> Count by<a href="http://busuanzi.ibruce.info/"> busuanzi.</a></span><span> Powered by<a rel="nofollow" target="_blank" href="https://hexo.io"> Hexo.</a></span></p></div></div></div><script src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js" async></script><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?3d768ac3c655ab53fbe8a4ac31a4528a";
  var s = document.getElementsByTagName("script")[0];
  s.parentNode.insertBefore(hm, s);
  })();
</script><script type="text/javascript" src="/js/search.json.js?v=2.0.1"></script><script type="text/javascript" src="/js/toctotop.js?v=2.0.1" async></script><script>window._bd_share_config={"common":{"bdSnsKey":{},"bdText":"","bdMini":"2","bdMiniList":["mshare","weixin","tsina","qzone","linkedin","fbook","twi","print","renren","sqq","evernotecn","bdysc","tqq","tqf","bdxc","kaixin001","tieba","douban","bdhome","thx","ibaidu","meilishuo","mogujie","diandian","huaban","duitang","hx","fx","youdao","sdo","qingbiji","people","xinhua","mail","isohu","yaolan","wealink","ty","iguba","h163","copy"],"bdPic":"","bdStyle":"1","bdSize":"16"},"share":{},"image":{"viewList":["tsina","qzone","weixin","fbook","twi","linkedin","youdao","evernotecn","mshare"],"viewText":"分享到：","viewSize":"16"},"selectShare":{"bdContainerClass":null,"bdSelectMiniList":["tsina","qzone","weixin","fbook","twi","linkedin","youdao","evernotecn","mshare"]}};with(document)0[(getElementsByTagName('head')[0]||head).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
</script></body></html>