<!DOCTYPE html><html lang="zh-CN"><head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"><meta content="yes" name="apple-mobile-web-app-capable"><meta content="black-translucent" name="apple-mobile-web-app-status-bar-style"><meta content="telephone=no" name="format-detection"><meta name="description"><title>mt's blog | 学习记录</title><link rel="stylesheet" type="text/css" href="/css/style.css?v=0.0.0"><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/normalize/8.0.0/normalize.min.css"><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/pure/1.0.0/pure-min.css"><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/pure/1.0.0/grids-responsive-min.css"><link rel="stylesheet" href="//cdn.bootcss.com/font-awesome/4.7.0/css/font-awesome.min.css"><script type="text/javascript" src="//cdn.bootcss.com/jquery/3.3.1/jquery.min.js"></script><link rel="Shortcut Icon" type="image/x-icon" href="/favicon.ico"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="apple-touch-icon-precomposed" href="/apple-touch-icon.png"><link rel="alternate" type="application/atom+xml" href="/atom.xml"></head><body><div class="body_container"><div id="header"><div class="site-name"><h1 class="hidden">mt's blog</h1><a id="logo" href="/.">mt's blog</a><p class="description">学习记录</p></div><div id="nav-menu"><a class="current" href="/."><i class="fa fa-home"> 首页</i></a><a href="/archives/"><i class="fa fa-archive"> 归档</i></a><a href="/about/"><i class="fa fa-user"> 关于</i></a><a href="/atom.xml"><i class="fa fa-rss"> 订阅</i></a></div></div><div class="pure-g" id="layout"><div class="pure-u-1 pure-u-md-3-4"><div class="content_container"><div class="post"><h1 class="post-title"><a href="/2017/08/28/seq实战/一个RNA-seq实战/">一个RNA-seq实战</a></h1><div class="post-meta">2017-08-28</div><div class="post-content"><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">#! usr/bin/bash</span><br><span class="line">set -u</span><br><span class="line">set -e</span><br><span class="line">set -o pipefail</span><br><span class="line">hg19_ref=~/fastq/hisat2_index/hg19/genome</span><br><span class="line">for x in `seq 677 680`;do ascp -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp-private.ncbi.nlm.nih.gov --user=anonftp --mode=recv /sra/sra-instant/reads/ByStudy/sra/SRP/SRP029/SRP029245/SRR957&quot;$x&quot;/SRR957&quot;$x&quot;.sra .;done</span><br><span class="line">##ls *sra |while read id; do fastq-dump --split-3 $id;done</span><br><span class="line">ls *sra |while read id; do fastq-dump --gzip --split-3 -A $id ;done</span><br><span class="line">## ls *fastq |xargs fastqc -t 10</span><br><span class="line">ls *.fastq.gz|while read id;do(hisat2 -t -p 5 -x $hg19_ref -U $id 2&gt;$&#123;id%%.*&#125;.map.log | /home/tingm/software/samtools-1.3.1/samtools view -Sb  - &gt;$&#123;id%%.*&#125;.bam);done</span><br><span class="line">ls *.bam|while read id;do /home/tingm/software/samtools-1.3.1/samtools sort -n -@ 5 $id -o $&#123;id%%.*&#125;.Nsorted.bam;done</span><br><span class="line">ls *.Nsorted.bam |while read id;do htseq-count -f bam -s no -i gene_name $id ~/fastq/refdata/Genecode/gencode.v26lift37.annotation.gff3 1&gt;$&#123;id%%.*&#125;.geneCounts 2&gt;$&#123;id%%.*&#125;.HTseq.log;done</span><br></pre></td></tr></table></figure></div><p class="readmore"><a href="/2017/08/28/seq实战/一个RNA-seq实战/">阅读全文</a></p></div><div class="post"><h1 class="post-title"><a href="/2017/08/23/转录组入门/转录组入门——8/">转录组入门——8</a></h1><div class="post-meta">2017-08-23</div><div class="post-content"><blockquote>
<p>选择p&lt;0.05而且abs(logFC)大于一个与众的基因为显著差异表达基因集，对这个基因集用R包做KEGG/GO超几何分布检验分析。<br>然后把表达矩阵和分组信息分别作出cls和gct文件，导入到GSEA软件分析。<br>基本任务是完成这个分析。</p>
</blockquote></div><p class="readmore"><a href="/2017/08/23/转录组入门/转录组入门——8/">阅读全文</a></p></div><div class="post"><h1 class="post-title"><a href="/2017/08/21/转录组入门/转录组入门——7/">转录组入门——7</a></h1><div class="post-meta">2017-08-21</div><div class="post-content"><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">.libPaths(&quot;/home/tingm/R/x86_64-pc-linux-gnu-library/3.2&quot;)</span><br><span class="line">  library(limma)</span><br><span class="line">  library(edgeR)</span><br><span class="line">  library(DESeq2)</span><br><span class="line">    #change</span><br><span class="line">    path&lt;-&quot;/home/tingm/fastq/mus_result_count/&quot;</span><br><span class="line">    #change</span><br><span class="line">      samples&lt;-read.table(&quot;samples.list.txt&quot;,header=F,sep=&quot;\t&quot;)</span><br><span class="line">      samples&lt;-samples[,1]</span><br><span class="line">      #change</span><br><span class="line">      data.raw&lt;-read.table(&quot;Mus_SRR3589959.sgeneCounts&quot;,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;Mus_SRR3589959.sgeneCounts&quot;))</span><br><span class="line">      for(sample in samples)</span><br><span class="line">      &#123;</span><br><span class="line">        print(sample)</span><br><span class="line">        file&lt;-paste(path,sample,sep=&quot;&quot;)</span><br><span class="line">        data.new&lt;-read.table(file,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,sample))</span><br><span class="line">        data.raw&lt;-merge(data.new,data.raw,by=&quot;gene_id&quot;)</span><br><span class="line">      &#125;</span><br><span class="line">     data.raw&lt;-data.raw[-1:-5,]</span><br><span class="line">        #change</span><br><span class="line">     names(data.raw) &lt;- c(&quot;gene_id&quot;,&quot;Akap95_2&quot;,&quot;control_2&quot;,&quot;Akap95_1&quot;,&quot;control_1&quot;)</span><br><span class="line">     write.table(data.raw,&quot;count.matix.csv&quot;,row.names=F,quote=F)</span><br></pre></td></tr></table></figure></div><p class="readmore"><a href="/2017/08/21/转录组入门/转录组入门——7/">阅读全文</a></p></div><div class="post"><h1 class="post-title"><a href="/2017/08/15/转录组入门/转录组入门——6/">转录组入门——6</a></h1><div class="post-meta">2017-08-15</div><div class="post-content"><h5 id="实现reads计数的软件教程"><a href="#实现reads计数的软件教程" class="headerlink" title="实现reads计数的软件教程"></a>实现reads计数的软件教程</h5><blockquote>
<p>map后获得了每条read比对到基因组上的位置，后面需要把基因组的位置对应到基因。这个提供基因位置的注释文件通常以GTF或GFF3格式呈现。基因表达量最直接的分析手段就是计算比对到每个基因的reads有多少条，在转录组测序中，我们通常称这个数字为count<br>对每个基因计数之后得到的count matrix再后续的分析中，要注意标准化的问题。<br>比较同一个样本(within-sample)不同基因之间的表达情况，需要考虑到转录本长度，因为转录本越长，那么检测的片段也会更多。比较不同样本（across sample）同一个基因的表达情况，不必在意转录本长度，但要考虑到测序深度（sequence depth)，测序深度越高，检测到的概率越大。除这两因素外，还需考虑GC%导致的偏差及测序仪器的系统偏差。<br>目前对read count标准化的算法有RPKM（SE）, FPKM（PE），TPM, TMM等。但是，有一些下游分析的软件会要求是输入的count matrix是原始数据，未经标准化，比如说DESeq2，这个时候你需要注意你上一步所用软件会不会进行标准化。<br>计数分为三个水平： gene-level, transcript-level, exon-usage-level<br>标准化方法： FPKM RPKM TMM TPM</p></div><p class="readmore"><a href="/2017/08/15/转录组入门/转录组入门——6/">阅读全文</a></p></div><div class="post"><h1 class="post-title"><a href="/2017/08/14/转录组入门/转录组入门——5/">转录组入门——5</a></h1><div class="post-meta">2017-08-14</div><div class="post-content"><h5 id="比对软件简介"><a href="#比对软件简介" class="headerlink" title="比对软件简介"></a>比对软件简介</h5><blockquote>
<p><a href="http://www.jianshu.com/p/681e02e7f9af" target="_blank" rel="noopener">比对软件比较</a><br>Bowtie1 <a href="http://bowtie-bio.sourceforge.net/index.shtml" target="_blank" rel="noopener">官方网址</a><br>Bowtie2 <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" target="_blank" rel="noopener">官方网址</a> <a href="http://blog.leanote.com/post/songtaogui/Bowtie2%E5%AE%89%E8%A3%85%E4%BD%BF%E7%94%A8" target="_blank" rel="noopener">中文教程</a> <a href="http://blog.leanote.com/post/375952584@qq.com/Bowtie2%E7%9A%84%E5%AE%89%E8%A3%85%E4%B8%8E%E4%BD%BF%E7%94%A8" target="_blank" rel="noopener">中文教程</a><br><a href="https://cncbi.github.io/Bowtie2-Manual-CN/" target="_blank" rel="noopener">中文教程</a> <a href="http://fhqdddddd.blog.163.com/blog/static/18699154201422735644352/" target="_blank" rel="noopener">中文教程</a><br>Bowtie1 Bowtie2 <a href="http://blog.csdn.net/soyabean555999/article/details/62235577" target="_blank" rel="noopener">二者区别</a><br><a href="http://www.cnblogs.com/wq242424/p/7025946.html" target="_blank" rel="noopener">Bowtie 安装</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">#下载bowtie2-2.3.2-linux-x86_64.zip解压后有错误 未解决 tbb需要Python2.7</span><br><span class="line">find ./ -name libtbbmalloc_proxy.so.2</span><br><span class="line">echo $LD_LIBRARY_PATH</span><br><span class="line">export LD_LIBRARY_PATH=~/biosoft/tbb2017_20170604oss/lib/:$LD_LIBRARY_PATH</span><br></pre></td></tr></table></figure></p></div><p class="readmore"><a href="/2017/08/14/转录组入门/转录组入门——5/">阅读全文</a></p></div><div class="post"><h1 class="post-title"><a href="/2017/08/11/转录组入门/转录组入门——4/">转录组入门——4</a></h1><div class="post-meta">2017-08-11</div><div class="post-content"><h4 id="参考基因组-hg19-下载"><a href="#参考基因组-hg19-下载" class="headerlink" title="参考基因组(hg19)下载"></a>参考基因组(hg19)下载</h4><blockquote>
<p>NCBI     UCSC   ENSEMBL数据库 对应关系<br>GRCh36  hg18     ENSEMBL release_52<br>GRCh37  hg19     ENSEMBL release_59/61/64/68/69/75<br>GRCh38  hg38     ENSEMBL  release_76/77/78/80/81/82</p></div><p class="readmore"><a href="/2017/08/11/转录组入门/转录组入门——4/">阅读全文</a></p></div><div class="post"><h1 class="post-title"><a href="/2017/08/10/转录组入门/转录组入门——3/">转录组入门——3</a></h1><div class="post-meta">2017-08-10</div><div class="post-content"><h5 id="SRA-Toolkit-软件"><a href="#SRA-Toolkit-软件" class="headerlink" title="SRA Toolkit 软件"></a>SRA Toolkit 软件</h5><p><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc" target="_blank" rel="noopener">官方网址</a></p></div><p class="readmore"><a href="/2017/08/10/转录组入门/转录组入门——3/">阅读全文</a></p></div><div class="post"><h1 class="post-title"><a href="/2017/08/09/转录组入门/转录组入门——2/">转录组入门——2</a></h1><div class="post-meta">2017-08-09</div><div class="post-content"><h5 id="CentOS-release-6-5-Final"><a href="#CentOS-release-6-5-Final" class="headerlink" title="CentOS release 6.5 (Final)"></a>CentOS release 6.5 (Final)</h5><h5 id="GEO-SRA数据库的数据存放形式参考"><a href="#GEO-SRA数据库的数据存放形式参考" class="headerlink" title="GEO/SRA数据库的数据存放形式参考"></a>GEO/SRA数据库的数据存放形式参考</h5><p><a href="http://www.biotrainee.com/forum.php?mod=viewthread&amp;tid=1884&amp;fromuid=1685" target="_blank" rel="noopener">青山屋主 转录组入门(2)-作业</a><br><a href="http://www.biotrainee.com/forum.php?mod=viewthread&amp;tid=1829&amp;fromuid=1685" target="_blank" rel="noopener">hoptop 转录组入门（二）读文献下数据</a></p></div><p class="readmore"><a href="/2017/08/09/转录组入门/转录组入门——2/">阅读全文</a></p></div><div class="post"><h1 class="post-title"><a href="/2017/08/09/转录组入门/转录组入门——1/">转录组入门——1</a></h1><div class="post-meta">2017-08-09</div><div class="post-content"><h5 id="系统环境"><a href="#系统环境" class="headerlink" title="系统环境"></a>系统环境</h5><p>CentOS 6.8（虚拟机之前就已安装好，所以不是推荐安装的ubuntu）</p>
<h5 id="软件安装参考"><a href="#软件安装参考" class="headerlink" title="软件安装参考"></a>软件安装参考</h5><p><a href="http://www.biotrainee.com/thread-1742-1-1.html" target="_blank" rel="noopener">Jimmy转录组入门(1)：计算机资源的准备</a><br><a href="http://www.biotrainee.com/thread-1796-1-1.html" target="_blank" rel="noopener">青山屋主转录组入门(1)-作业</a><br><a href="http://www.chenlianfu.com/?p=2438" target="_blank" rel="noopener">HTSeq中文教程</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br></pre></td><td class="code"><pre><span class="line"># download miniconda2</span><br><span class="line">wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh </span><br><span class="line">bash Miniconda2-latest-Linux-x86_64.sh</span><br><span class="line">source ~/.bashrc</span><br><span class="line"></span><br><span class="line"># sratoolkit</span><br><span class="line">conda install -c jfear sratoolkit=2.8.1</span><br><span class="line">fastq-dump</span><br><span class="line"></span><br><span class="line"># fastqc</span><br><span class="line">conda install -c bioconda fastqc=0.11.5</span><br><span class="line">fastqc</span><br><span class="line"></span><br><span class="line"># hisat2</span><br><span class="line">conda install -c bioconda hisat2</span><br><span class="line">hisat2</span><br><span class="line"></span><br><span class="line"># samtools</span><br><span class="line">conda install -c bioconda samtools=1.5 </span><br><span class="line"># error</span><br><span class="line"># samtools: error while loading shared libraries: libbz2.so.1.0: cannot open shared object file: No such file or directory</span><br><span class="line"># 错误解决参考 https://www.biostars.org/p/266111/</span><br><span class="line">conda install -c conda-forge -c bioconda samtools bzip2</span><br><span class="line">samtools</span><br><span class="line"></span><br><span class="line"># htseq-count</span><br><span class="line">conda install -c bioconda htseq=0.7.2</span><br><span class="line">python</span><br><span class="line">&gt;&gt;&gt;import HTSeq</span><br><span class="line"></span><br><span class="line"># R</span><br><span class="line">sudo yum install epel-release</span><br><span class="line">sudo yum install R</span><br><span class="line">R</span><br><span class="line"></span><br><span class="line"># Rstudio</span><br><span class="line">conda install rstudio</span><br><span class="line">rstudio</span><br></pre></td></tr></table></figure></p></div><p class="readmore"><a href="/2017/08/09/转录组入门/转录组入门——1/">阅读全文</a></p></div><nav class="page-navigator"><a class="extend prev" rel="prev" href="/page/4/">上一页</a><a class="page-number" href="/">1</a><span class="space">&hellip;</span><a class="page-number" href="/page/3/">3</a><a class="page-number" href="/page/4/">4</a><span class="page-number current">5</span></nav></div></div><div class="pure-u-1 pure-u-md-1-4"><div id="sidebar"><div class="widget"><div class="search-form"><input id="local-search-input" placeholder="Search" type="text" name="q" results="0"/><div id="local-search-result"></div></div></div><div class="widget"><div class="widget-title"><i class="fa fa-folder-o"> 分类</i></div><ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/Beginning-Perl-free-学习笔记/">Beginning Perl (free)学习笔记</a><span class="category-list-count">3</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/Linux基础/">Linux基础</a><span class="category-list-count">2</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/Test/">Test</a><span class="category-list-count">1</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/mysql学习记录/">mysql学习记录</a><span class="category-list-count">2</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/project/">project</a><span class="category-list-count">8</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/seq实战/">seq实战</a><span class="category-list-count">4</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/数据科学实训营/">数据科学实训营</a><span class="category-list-count">19</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/爬虫/">爬虫</a><span class="category-list-count">1</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/转录组入门/">转录组入门</a><span class="category-list-count">8</span></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-star-o"> 标签</i></div><div class="tagcloud"><a href="/tags/gatk/" style="font-size: 15px;">gatk</a> <a href="/tags/bioinformatics/" style="font-size: 15px;">bioinformatics</a> <a href="/tags/Linux/" style="font-size: 15px;">Linux</a> <a href="/tags/samtools/" style="font-size: 15px;">samtools</a> <a href="/tags/sql/" style="font-size: 15px;">sql</a> <a href="/tags/mysql/" style="font-size: 15px;">mysql</a> <a href="/tags/php/" style="font-size: 15px;">php</a> <a href="/tags/bioinformaticsant/" style="font-size: 15px;">bioinformaticsant</a> <a href="/tags/RTG-tools/" style="font-size: 15px;">RTG-tools</a> <a href="/tags/SnpEff/" style="font-size: 15px;">SnpEff</a> <a href="/tags/giremi/" style="font-size: 15px;">giremi</a> <a href="/tags/perl/" style="font-size: 15px;">perl</a> <a href="/tags/nanopolish/" style="font-size: 15px;">nanopolish</a> <a href="/tags/star/" style="font-size: 15px;">star</a> <a href="/tags/trim-galore/" style="font-size: 15px;">trim_galore</a> <a href="/tags/hisat2/" style="font-size: 15px;">hisat2</a> <a href="/tags/HTSeq/" style="font-size: 15px;">HTSeq</a> <a href="/tags/FastViromeExplorer/" style="font-size: 15px;">FastViromeExplorer</a> <a href="/tags/picard/" style="font-size: 15px;">picard</a> <a href="/tags/git/" style="font-size: 15px;">git</a> <a href="/tags/python/" style="font-size: 15px;">python</a> <a href="/tags/爬虫，python/" style="font-size: 15px;">爬虫，python</a></div></div><div class="widget"><div class="widget-title"><i class="fa fa-file-o"> 最近文章</i></div><ul class="post-list"><li class="post-list-item"><a class="post-list-link" href="/2018/08/07/SQL/sql_2/">mysql学习记录--Part.2</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/08/06/SQL/sql_1/">mysql学习记录--Part.1</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/07/23/爬虫/拉勾网.Part1/">拉勾网.Part-1</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/28/project/nanopolish/">nanopolish</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/20/project/RTG-tools安装/">RTG-tools安装</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/14/project/ann.giremi/">ann.giremi.sh</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/14/project/star.gatk/">star.gatk.sh</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/14/project/trim.aln.count & find_virus_filt/">trim.aln.count.sh & find_virus_filt.sh</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/04/project/giremi及gatk下载/">giremi, gatk下载使用</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/05/16/project/samtools命令/">samtools</a></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-external-link"> 友情链接</i></div><ul></ul><a href="https://github.com/mt831" title="github" target="_blank">github</a></div></div></div><div class="pure-u-1 pure-u-md-3-4"><div id="footer">Copyright © 2018 <a href="/." rel="nofollow">mt's blog.</a> Powered by<a rel="nofollow" target="_blank" href="https://hexo.io"> Hexo.</a><a rel="nofollow" target="_blank" href="https://github.com/tufu9441/maupassant-hexo"> Theme</a> by<a rel="nofollow" target="_blank" href="https://github.com/pagecho"> Cho.</a></div></div></div><a class="show" id="rocket" href="#top"></a><script type="text/javascript" src="/js/totop.js?v=0.0.0" async></script><script type="text/javascript" src="//cdn.bootcss.com/fancybox/3.2.5/jquery.fancybox.min.js" async></script><script type="text/javascript" src="/js/fancybox.js?v=0.0.0" async></script><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/fancybox/3.2.5/jquery.fancybox.min.css"><script type="text/javascript" src="/js/search.js?v=0.0.0"></script><script>var search_path = 'search.xml';
if (search_path.length == 0) {
   search_path = 'search.xml';
}
var path = '/' + search_path;
searchFunc(path, 'local-search-input', 'local-search-result');
</script><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = '//hm.baidu.com/hm.js?' + '3d768ac3c655ab53fbe8a4ac31a4528a';
  var s = document.getElementsByTagName("script")[0];
  s.parentNode.insertBefore(hm, s);
  })();
</script><script type="text/javascript" src="/js/codeblock-resizer.js?v=0.0.0"></script><script type="text/javascript" src="/js/smartresize.js?v=0.0.0"></script></div></body></html>