<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    
    <meta name="theme-color" content="#33363b">
    <meta name="msapplication-TileColor" content="#33363b">
    
    
    
    <meta name="keywords" content="python, R, math, bioinformatics">
    
    
    <link rel="apple-touch-icon" sizes="180x180" href="/favicons/apple-touch-icon.png">
    
    
    <link rel="icon" type="image/png" sizes="192x192" href="/favicons/android-chrome-192x192.png">
    
    
    <link rel="icon" type="image/png" sizes="32x32" href="/favicons/favicon-32x32.png">
    
    
    <link rel="icon" type="image/png" sizes="16x16" href="/favicons/favicon-16x16.png">
    
    
    <link rel="mask-icon" href="/favicons/safari-pinned-tab.svg" color="#33363b">
    
    
    <link rel="manifest" href="/favicons/site.webmanifest">
    
    
    <meta name="msapplication-config" content="/favicons/browserconfig.xml">
    
    
    <link rel="alternate" href="/atom.xml" title="mt's blog" type="application/atom+xml" />
    
    <link rel="stylesheet" type="text/css" href="/css/index.css">
    
    <link rel="stylesheet" type="text/css" href="/css/sidebar.css">
    
    
    <link rel="stylesheet" type="text/css" href="/css/custom.css">
    <link rel="stylesheet" type="text/css" href="/css/normalize.css">
    <link rel="stylesheet" type="text/css" href="/css/solarized-light.css">
    <link rel="stylesheet" type="text/css" href="/css/lightgallery.min.css">
    <script type="text/javascript" src="/js/jquery.min.js"></script>
    <script defer type="text/javascript" src="/js/util.js"></script>
    <script defer type="text/javascript" src="/js/scrollspy.js"></script>
    <script defer type="text/javascript" src="/js/fontawesome-all.min.js"></script>
    <script defer type="text/javascript" src="/js/lightgallery.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-fullscreen.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-hash.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-pager.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-thumbnail.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-zoom.min.js"></script>
    
    <script defer src="/js/busuanzi.pure.mini.js"></script>
    
    
    <script defer type="text/javascript" src="/js/search.js"></script>
    <script type="text/javascript">
    $(document).ready(function () {
      var searchPath = "search.xml";
      if (searchPath.length === 0) {
        searchPath = "search.xml";
      }
      var path = "/" + searchPath;
      searchFunc(path, "search-input", "search-result");
    });
    </script>
    
    
    <script defer type="text/javascript" src="/js/index.js"></script>
    
    <script defer type="text/javascript" src="/js/custom.js"></script>
    <title>mt's blog</title>
  </head>
  <body itemscope itemtype="http://schema.org/WebPage" lang=""  data-spy="scroll" data-target=".list-group">
    
<header id="header" class="header">
  <div class="container">
    <div class="header-container">
      <div class="header-title">
        <h1 class="title"><a href="/">mt's blog</a></h1>
        <h2 class="subtitle"></h2>
      </div>
      <div class="logo">
        <img src="/images/ARIA_logo.png" alt="logo">
      </div>
    </div>
    
<nav id="nav" class="nav">
  <a id="nav-toggle" class="nav-toggle"><i class="fas fa-bars"></i></a>
  <ul id="menu">
    
    <li><a href="/">Home</a></li>
    
    <li><a href="/archives/">Archives</a></li>
    
    <li><a href="/categories/">Categories</a></li>
    
    <li><a href="/tags/">Tags</a></li>
    
  </ul>
</nav>


  </div>
</header>


    <main id="main" class="main">
      <div class="container">
        <div class="main-container">
          <div class="content">
            

<section id="index" class="index">
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2018/04/03/爬虫：模拟登陆和动态网站抓取/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2018/04/03/爬虫：模拟登陆和动态网站抓取/" itemprop="url">爬虫：模拟登陆和动态网站抓取 要点记录</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2018-04-03T19:48:07+08:00">2018-04-03</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/数据科学实训营/" itemprop="url" rel="index"><span itemprop="name">数据科学实训营</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2018/04/03/爬虫：模拟登陆和动态网站抓取/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2018/04/03/爬虫：模拟登陆和动态网站抓取/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="豆瓣模拟登录（google浏览器）"><a href="#豆瓣模拟登录（google浏览器）" class="headerlink" title="豆瓣模拟登录（google浏览器）"></a>豆瓣模拟登录（google浏览器）</h5><p><strong>存在问题？</strong> 登录账号和密码、保持登录状态（跨请求地保持某些参数）、图形验证码<br><strong>操作</strong>：在检查项的Network下查看登录信息，点击login，Header下From data内有表单信息</p>
<blockquote>
<p>source: None<br>redir: <a href="https://www.douban.com/" target="_blank" rel="noopener">https://www.douban.com/</a> #登陆页面<br>form_email: test2 #用户邮箱<br>form_password: 111111 #密码<br>captcha-solution: potato #验证码，豆瓣有验证码图库，比对图片ID号<br>captcha-id: tljqzo0Gv3aYUpIarqXaDBAg:en #验证码ID XPath去取<br>login: 登录</p>
</blockquote>
<p>captcha_img_url = selector.xpath(‘//img[@id=”captcha_image”]//@src’)<br><strong>结果为空，待解决</strong></p>
<h5 id="scrapy-命令行端执行脚本"><a href="#scrapy-命令行端执行脚本" class="headerlink" title="scrapy 命令行端执行脚本"></a>scrapy 命令行端执行脚本</h5><p><strong>待练习</strong><br>[Scrapy简单入门及实例讲解][<a href="https://www.cnblogs.com/kongzhagen/p/6549053.html" target="_blank" rel="noopener">https://www.cnblogs.com/kongzhagen/p/6549053.html</a>]<br>[Scrapy安装、爬虫入门教程、爬虫实例（豆瓣电影爬虫）][<a href="http://www.cnblogs.com/Shirlies/p/4536880.html" target="_blank" rel="noopener">http://www.cnblogs.com/Shirlies/p/4536880.html</a>]</p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/python/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> python</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2018/04/03/L2.Pandas-Series-DataFrame/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2018/04/03/L2.Pandas-Series-DataFrame/" itemprop="url">L2 Pandas Series & DataFrame 关键点记录</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2018-04-03T19:43:53+08:00">2018-04-03</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/数据科学实训营/" itemprop="url" rel="index"><span itemprop="name">数据科学实训营</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2018/04/03/L2.Pandas-Series-DataFrame/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2018/04/03/L2.Pandas-Series-DataFrame/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="Series-一维数组"><a href="#Series-一维数组" class="headerlink" title="Series 一维数组"></a>Series 一维数组</h5><p>构建初始化：可用list构建（dict也可），默认用0-n为Series的index，也可指定<br>位置索引/名称索引：s[[1,2,3]] 内为列表<br>数学运算（广播特性）：list不可（list相加并在一起），基于索引进行计算（list相加并在一起）<br>in判断是否在索引内    .get看值是否有    条件索引取值<br>统计计算：min max mean等    可和numpy结合使用 如np.log()<br>赋值：可和条件索引结合使用<br>数据缺失：notnull() isnull()    expense[expense.isnull()] = expense.mean()</p>
<h5 id="DataFrame-二维数组"><a href="#DataFrame-二维数组" class="headerlink" title="DataFrame 二维数组"></a>DataFrame 二维数组</h5><p>构建：给列名列元素    columns调整列的顺序    index可重定义    可由Series构建<br>取值：下为取列，注意区分    <strong>.loc取行</strong>(传入list或类似切片用法)　<a href="https://blog.csdn.net/hecongqing/article/details/61927615" target="_blank" rel="noopener">.loc用法参考</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">type(df[&apos;apt&apos;])</span><br><span class="line">pandas.core.series.Series</span><br><span class="line">type(df[[&apos;apt&apos;]])</span><br><span class="line">pandas.core.frame.DataFrame</span><br></pre></td></tr></table></figure></p>
<p>df.T 转置    df.shape 、df.info()：行列信息    df.describe():均值分位数等信息<br><a href="https://blog.csdn.net/qq_28219759/article/details/49160073" target="_blank" rel="noopener">.isin参考</a><br>缺失值处理：.fillna</p>
<h5 id="Index-csv文件的读写"><a href="#Index-csv文件的读写" class="headerlink" title="Index csv文件的读写"></a>Index csv文件的读写</h5><p><a href="https://blog.csdn.net/newbiemath/article/details/61432073" target="_blank" rel="noopener">.read_csv参考</a>: index_col parse_dates<br>.to_csv()<br><strong>.index</strong>    index.dayofweek<br>%matplotlib inline 绘图<br>数据选择 | &amp; !</p>
<p><strong><em>以上整理自<a href="www.xiniuedu.com">稀牛学院</a>数据科学实训营 寒小阳老师关于pandas的讲解</em></strong></p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/python/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> python</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2018/04/02/爬虫：静态网站数据获取/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2018/04/02/爬虫：静态网站数据获取/" itemprop="url">爬虫：静态网站数据获取</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2018-04-02T18:29:09+08:00">2018-04-02</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/数据科学实训营/" itemprop="url" rel="index"><span itemprop="name">数据科学实训营</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2018/04/02/爬虫：静态网站数据获取/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2018/04/02/爬虫：静态网站数据获取/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h4 id="XPath"><a href="#XPath" class="headerlink" title="XPath"></a>XPath</h4><h5 id="定义"><a href="#定义" class="headerlink" title="定义"></a>定义</h5><blockquote>
<p>XML 文档中查信息的语言， XML 文档中对元素和属性进行遍历</p>
</blockquote>
<h5 id="七类节点node-XML最小单元"><a href="#七类节点node-XML最小单元" class="headerlink" title="七类节点node XML最小单元"></a>七类节点node XML最小单元</h5><blockquote>
<p>element、attribute、text、namespace、processing-instruction、comment、root</p>
</blockquote>
<h5 id="语法"><a href="#语法" class="headerlink" title="语法"></a>语法</h5><p>选取节点：路径表达式<br><a href="http://www.w3school.com.cn/xpath/index.asp" target="_blank" rel="noopener">XPath教程</a></p>
<blockquote>
<p>nodename 选取此node所有子节点<br>/ 文档根下节点 // 文档里任何位置的节点<br>. 当前节点  .. 当前节点父节点<br>@选属性<br>[] 谓语 找特定节点或包含特定值节点<br>星号 匹配任何元素节点 @* 匹配任何属性节点  node() 匹配任何类型节点<br>| 选取若干路径<br>绝对路径和相对路径：前者以“/”起始</p>
</blockquote>
<h5 id="查看网页XPath路径记录"><a href="#查看网页XPath路径记录" class="headerlink" title="查看网页XPath路径记录"></a>查看网页XPath路径记录</h5><p>右击标题，选检查，elements下查看，右击copy XPath，ctrl+F，ctrl+V，之前粘贴部分被高亮，高亮部分为XPath路径示例，具体语法参照上面<br>//div[@class=”text”]/em/a/text() 取内容<br>div[@class=”text”] 选取所有拥有名为class=”text” 的属性的div 元素<br>a/@href 取属性</p>
<h4 id="requesets"><a href="#requesets" class="headerlink" title="requesets"></a>requesets</h4><p>One.请求网页，<a href="https://blog.csdn.net/shanzhizi/article/details/50903748" target="_blank" rel="noopener">requesets详解</a><br>Two.获取返回状态200和网页编码格式’GB2312’ -&gt; .content显示内容 -&gt; import etree;XPath抓取标题;用text-&gt;传递参数wd和rn -&gt; 超时时间 -&gt;添加代理 -&gt;请求头内容<br><a href="https://www.biaodianfu.com/python-lxml-xpath.html" target="_blank" rel="noopener">etree和XPath</a></p>
<p>豆瓣最热电影评论爬取示例、链家爬取案例练习</p>
<p><strong><em><a href="www.xiniuedu.com">稀牛学院</a>数据科学实训营 助教寒小阳老师爬虫课程 学习整理</em></strong></p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/python/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> python</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2018/04/01/实战1-QQ答疑/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2018/04/01/实战1-QQ答疑/" itemprop="url">实战1.QQ答疑</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2018-04-01T21:13:58+08:00">2018-04-01</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/数据科学实训营/" itemprop="url" rel="index"><span itemprop="name">数据科学实训营</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2018/04/01/实战1-QQ答疑/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2018/04/01/实战1-QQ答疑/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h4 id="实战1-一起来打怪之-python-基础与进阶练习题-QQ答疑"><a href="#实战1-一起来打怪之-python-基础与进阶练习题-QQ答疑" class="headerlink" title="实战1.一起来打怪之 python 基础与进阶练习题 QQ答疑"></a>实战1.一起来打怪之 python 基础与进阶练习题 QQ答疑</h4><h5 id="关于竞赛"><a href="#关于竞赛" class="headerlink" title="关于竞赛"></a>关于竞赛</h5><p>Q：竞赛情况<br>A：竞赛最有名的是Kaggle，国内是阿里天池。获奖情况每个比赛不一样，有的前10，有的前6。一般能到前1%就很不错了。竞赛爬取结果: <a href="https://github.com/iphysresearch/DataSciComp" target="_blank" rel="noopener">土豆的github</a></p>
<h5 id="如何学习"><a href="#如何学习" class="headerlink" title="如何学习"></a>如何学习</h5><p>Q：边听课边在实验室环境里的notebook上做笔记，两个窗口不能同时开着，只能切换运行<br>A：可同时打开notebook同时做笔记。但是建议注意力放在听课上面，用笔和纸速记。可用markdown做笔记。<br>A：问题出现，先google,bing检索，常见社区Stack Overflow</p>
<h5 id="运行平台"><a href="#运行平台" class="headerlink" title="运行平台"></a>运行平台</h5><p>Q：jupyter notebook执行ternimal无法显示<br>A：广告拦截软件拦截，关掉广告拦截软件或者换一个浏览器试试。chrome和firefox都能很好的运行稀牛的实验环境。<br>Q：只要能用anaconda里的jupyter打开发的ipynb的文件就说明配置没错吗<br>A：anaconda应该没有什么需要编辑的，只要装上了，就能启动jupyter<br>Q：jupyter里面有类似上下键翻录历史命令/代码的快捷键？<br>A：shift+enter ctrl+enter<br>Q：在terminal里面输入dir，然后下一条命令还要输入dir，winh和linux下都可以用键盘上下翻的键显示，ipython也可以，notebook里面不知道有没有<br>A：木有<br>Q：将notebook里的cell 转换为markdown格式是什么快捷键<br>A：选中单元格+m<br>Q：本地安装了jupyter ，但是terminal里运行jupyter notebook 每次给出本地地址，需要手动复制到浏览器运行，难道不应该是直接弹出浏览器运行吗？ 怎么设置，是要修改环境变量吗？怎么设置？<br>A：打开失败了，检查一下里面浏览器的设置<br>Q：在jupter里写的markdown可以变成文本吗<br>A：可以点左上角，file——&gt;download as，里面有很多选项<br>Q：shell查看文件<br>A：!head</p>
<h5 id="python基础"><a href="#python基础" class="headerlink" title="python基础"></a>python基础</h5><p>Q：文件读取多语言的文件怎么处理<br>A：建议使用python3，python3对中文支持要好一些，特别爬虫的时候感觉最明显。如果还不行的话建议，在写入文件和读取文件的时候，跟上字符编码。未解决，可能和win10系统相关。<br>Q：input 可以输入字符串列表吗<br>A：input来输入int，raw_input来输入string。可参考：<a href="https://blog.csdn.net/qingquanyingyue/article/details/77964990" target="_blank" rel="noopener">Python输入字符串转列表</a><br>Q：导入了seaborn，为什么画出来的图还是matplotlib默认风格的<br>A：代码用的pylot，sns.set()执行以下，再绘图<br>Q：不定长度参数的函数中 * 号是指针的意思？<br>A：不是，廖学峰的官方网站python3: <a href="https://www.liaoxuefeng.com/wiki/001374738125095c955c1e6d8bb493182103fac9270762a000/001374738449338c8a122a7f2e047899fc162f4a7205ea3000" target="_blank" rel="noopener">参数小结</a></p>
<h5 id="爬虫"><a href="#爬虫" class="headerlink" title="爬虫"></a>爬虫</h5><p>Q：静态数据获取视频中查看元素之后在filter里填了东西是回车就可以找到了吗 xpath表达式<br>A：取东西一般有两种类型。一种是文本，最后需要接一个text()来取文本。另外一种是取属性。可查看：<a href="http://www.w3school.com.cn/xpath/xpath_syntax.asp" target="_blank" rel="noopener">xpath语法</a><br>Q：浏览器打开检查元素，是直接在filter里填xpath就回高亮就会跳到我要找的文本或属性吗？<br>A：是的，只要写对了，就会高亮。直接使用chrome右键copy xpath得到的，可读性不高，可用性不强。<br>Q：怎样的xpath算是可读性比较强的<br>A：简洁、高效。</p>
<p><strong><em>以上来自<a href="www.xiniuedu.com">稀牛学院</a>数据科学实训营 助教易龙老师及学员们的QQ答疑</em></strong></p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/python/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> python</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2018/04/01/L1.来自google的专业python指南/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2018/04/01/L1.来自google的专业python指南/" itemprop="url">来自google的专业python指南</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2018-04-01T18:01:46+08:00">2018-04-01</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/数据科学实训营/" itemprop="url" rel="index"><span itemprop="name">数据科学实训营</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2018/04/01/L1.来自google的专业python指南/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2018/04/01/L1.来自google的专业python指南/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <p>前言：课程结构相对于前四期进行调整。one.爬虫（数据获取）直播切录播。Two.机器学习及其依赖的数据处理课时加重。Three.大数据技能后移，希望教给大家大数据处理和机器学习大数据建模的技能。</p>
<h4 id="python使用原因："><a href="#python使用原因：" class="headerlink" title="python使用原因："></a>python使用原因：</h4><p>数据科学领域top2；<br>深度学习、人工智能领域举例：google: tensorflow、facebook: pytorch + caffe2、Amazon: mnxet、caffe、易上手package：Keras, TFlearn, tensorlayer、python接口；<br>机器学习领域举例：scikit-learn numpy/scipy pandas xgboost/lightGBM；<br>大数据领域举例：spark scala、pyspark、hadoop Map-Reduce、hadoop streaming + python 脚本；<br>数据科学家:数据驱动的解决方案，不花大时间在开发，集中在数据分析建模等问题上</p>
<h4 id="如何寻求帮助"><a href="#如何寻求帮助" class="headerlink" title="如何寻求帮助"></a>如何寻求帮助</h4><p>help() (快捷键win+ctrl+shift+tab)<br>dir()</p>
<h4 id="python运算"><a href="#python运算" class="headerlink" title="python运算"></a>python运算</h4><p>运算符：+ - * / %</p>
<h4 id="基本数据类型、变量"><a href="#基本数据类型、变量" class="headerlink" title="基本数据类型、变量"></a>基本数据类型、变量</h4><p>基本数据类型：int float str bool<br>type() 查看数据类型<br>整数和浮点数在计算机内部存储的方式是不同的，整数运算永远是精确的，而浮点数运算则可能会有四舍五入的误差。</p>
<h4 id="表达式"><a href="#表达式" class="headerlink" title="表达式"></a>表达式</h4><p>+=  -= *= /=</p>
<h4 id="字符串"><a href="#字符串" class="headerlink" title="字符串"></a>字符串</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">x = &quot;hello world!&quot;</span><br><span class="line">abc = &quot;123456&quot; </span><br><span class="line">abc.isdigit()  #Tab可补全</span><br><span class="line">&apos;&apos;&apos; &apos;&apos;&apos; 可内部换行</span><br></pre></td></tr></table></figure>
<p>可参考：<a href="https://blog.csdn.net/woainishifu/article/details/76105667" target="_blank" rel="noopener">三种引号区别</a></p>
<h4 id="字符串切片-splice"><a href="#字符串切片-splice" class="headerlink" title="字符串切片/splice"></a>字符串切片/splice</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">len(tmp_str) </span><br><span class="line">左起0 右起-1 左闭右开</span><br><span class="line">string.lower() </span><br><span class="line">.upper() </span><br><span class="line">.capitalize() </span><br><span class="line">.startswith(&quot;xiao&quot;) </span><br><span class="line">.endswith(&quot;sh&quot;) </span><br><span class="line">.strip() </span><br><span class="line">.find() #-1未找到  </span><br><span class="line">.split()</span><br></pre></td></tr></table></figure>
<h4 id="列表-list（存储一连串数据）"><a href="#列表-list（存储一连串数据）" class="headerlink" title="列表/list（存储一连串数据）"></a>列表/list（存储一连串数据）</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">names=[&apos;Hanmei&apos;,&apos;Lilei&apos;,&apos;Hanxiao&apos;,&apos;Xi&apos;,&apos;Bob&apos;,&apos;Davie&apos;]</span><br><span class="line">len(names)</span><br><span class="line">mixed = [&apos;Han&apos;,2,3.14,[&apos;Li&apos;,&apos;Hanmei&apos;]]</span><br><span class="line">len(mixed)</span><br><span class="line">mixed[1]</span><br><span class="line">mixed[-2]</span><br><span class="line">mixed[-1][-1]</span><br><span class="line">&apos;\n&apos;.join(names)</span><br><span class="line">names.append(&quot;xiaohong&quot;)#尾部添加</span><br><span class="line">names.remove([&quot;xiaohong&quot;])</span><br><span class="line">names.extend([&apos;xiao&apos;,&apos;xiaoxiao&apos;,&apos;Bao&apos;])</span><br><span class="line">names.reverse()</span><br><span class="line">help(list.pop)</span><br><span class="line">help(list.insert)</span><br></pre></td></tr></table></figure>
<h4 id="判断循环"><a href="#判断循环" class="headerlink" title="判断循环"></a>判断循环</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">age = 66</span><br><span class="line">if age&gt;80:</span><br><span class="line">	print(&apos;very old&apos;)</span><br><span class="line">elif age&gt;30:</span><br><span class="line">	print(&apos;old&apos;)</span><br><span class="line">else:</span><br><span class="line">	print(&apos;young&apos;)</span><br><span class="line">	</span><br><span class="line">for index,stu in enumerate(names):</span><br><span class="line">	print(index,stu)</span><br><span class="line">	</span><br><span class="line">i = 0</span><br><span class="line">while i&lt;10:</span><br><span class="line"> print(i)</span><br><span class="line"> i +=1</span><br><span class="line"> </span><br><span class="line">注意continue break 区分</span><br><span class="line">enumerate()函数将一个可遍历的数据对象(如list、元组或字符串)组合为一个索引序列，同时列出数据和数据下标，一般用在for循环当中</span><br></pre></td></tr></table></figure>
<h5 id="列表推导式"><a href="#列表推导式" class="headerlink" title="列表推导式"></a>列表推导式</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">[&quot;my name is: &quot; +name for name in names]</span><br><span class="line">[num + 5 for num in num_list]</span><br><span class="line">[num**3 for num in num_list if (num%2==1 and num &lt; 7)]</span><br><span class="line">#与或非and or not</span><br></pre></td></tr></table></figure>
<h4 id="集合Set"><a href="#集合Set" class="headerlink" title="集合Set"></a>集合Set</h4><p>y = set([‘h’,’a’,’m’])<br>与列表和元组不同，集合是无序的，也无法通过数字进行索引。此外，集合中的元素不能重复</p>
<h4 id="字典"><a href="#字典" class="headerlink" title="字典"></a>字典</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">legs = &#123;&apos;spider&apos;:8,&apos;pig&apos;:4,&apos;ducl&apos;:2&#125;</span><br><span class="line">type(legs)</span><br><span class="line">legs.values()</span><br><span class="line">legs.keys()</span><br><span class="line">&apos;bird&apos; in legs</span><br><span class="line">for animal, leg_num in legs.items():</span><br><span class="line">	print(animal,leg_num)</span><br></pre></td></tr></table></figure>
<h5 id="字典推导式"><a href="#字典推导式" class="headerlink" title="字典推导式"></a>字典推导式</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">my_list = [1,3,5,7,9,2,4]</span><br><span class="line">dic = &#123;&#125;</span><br><span class="line">for num in my_list:</span><br><span class="line">	dic[num] = num **3</span><br><span class="line">	</span><br><span class="line">&#123;num:num**3 for num in my_list&#125;</span><br></pre></td></tr></table></figure>
<h5 id="高级排序"><a href="#高级排序" class="headerlink" title="高级排序"></a>高级排序</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">num_list1.sort()#成员函数，对自身做调整，num_list1排序结果为返回值返回</span><br><span class="line">sorted(num_list2) #python中函数，num_list2未更改</span><br><span class="line">sorted(str, reverse=True)#字母反序</span><br><span class="line">sorted(str,key=len)#根据key函数即长度排序，长度一致asic排序</span><br><span class="line">sorted(str,key=str.lower)#小写后排序，key function给函数名，无需括号</span><br></pre></td></tr></table></figure>
<h4 id="函数"><a href="#函数" class="headerlink" title="函数"></a>函数</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">def get_first(my_list):</span><br><span class="line">	return my_list[0]</span><br><span class="line">get_first([&apos;Han&apos;,&apos;Li&apos;])</span><br><span class="line">classed = [&apos;Han&apos;,&apos;Li&apos;],[&apos;Xiao&apos;,&apos;Ming&apos;],[&apos;Wang&apos;,&apos;Xiao&apos;]</span><br><span class="line">sorted(classed,key=get_first)</span><br><span class="line">#不定长度的参数</span><br><span class="line">def print_all(*args):</span><br><span class="line">	print(type(args))</span><br><span class="line">	print(args)</span><br></pre></td></tr></table></figure>
<h4 id="文件读写"><a href="#文件读写" class="headerlink" title="文件读写"></a>文件读写</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">head -2 test.txt #shell</span><br><span class="line">f = open(&quot;test.txt&quot;, &apos;r&apos;)</span><br><span class="line">contents = f.readlines()</span><br><span class="line">f.close()</span><br><span class="line"></span><br><span class="line">for line in open(&apos;test.txt&apos;,&apos;r&apos;,encoding=&apos;UTF-8&apos;):</span><br><span class="line">	print(line.strip())</span><br><span class="line">	print(&quot;\n&quot;)</span><br></pre></td></tr></table></figure>
<h5 id="统计文件中的词频"><a href="#统计文件中的词频" class="headerlink" title="统计文件中的词频"></a>统计文件中的词频</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">def my_word_count(in_file,out_file):</span><br><span class="line">	word_count = &#123;&#125;</span><br><span class="line">	for line in open(in_file,&apos;r&apos;,encoding=&apos;utf-8&apos;):</span><br><span class="line">		words = line.strip().split(&quot; &quot;)</span><br><span class="line">		for word in words:</span><br><span class="line">			if word.lower() in word_count:</span><br><span class="line">				word_count[word.lower()] += 1 </span><br><span class="line">			else:</span><br><span class="line">				word_count[word.lower()] = 1</span><br><span class="line">	out = open(out_file,&apos;w&apos;,encoding=&apos;utf-8&apos;)</span><br><span class="line">	for word in word_count:</span><br><span class="line">		out.write(word+&quot;:&quot;+str(word_count[word])+&quot;\n&quot;)</span><br><span class="line">	print(&quot;词频统计完成&quot;)</span><br><span class="line">	out.close()</span><br><span class="line">	</span><br><span class="line">in_file = &apos;test.txt&apos;</span><br><span class="line">out_file = &apos;out.txt&apos;</span><br><span class="line">my_word_count(in_file,out_file)</span><br><span class="line"># += 间无空格 严格缩进</span><br></pre></td></tr></table></figure>
<h4 id="正则表达式"><a href="#正则表达式" class="headerlink" title="正则表达式"></a>正则表达式</h4><blockquote>
<p>\d匹配一个数字<br>\w匹配一个字母或数字<br>\s匹配一个空格（也包括Tab等空白符）<br>.匹配任意字符<br><em>表示任意个字符（包括0个）<br>+表示至少一个字符<br>?表示0个或1个字符<br>{n}表示n个字符<br>{n,m}表示n-m个字符<br>[0-9a-zA-Z_]匹配一个数字、字母或者下划线<br>[0-9a-zA-Z_]+可以匹配至少由一个数字、字母或者下划线组成的字符串，比如’a100’，’0<em>Z’，’Py3000’等等；<br>[a-zA-Z\</em>][0-9a-zA-Z_]</em>可以匹配由字母或下划线开头，后接任意个由一个数字、字母或者下划线组成的字符串，也就是Python合法的变量；<br>[a-zA-Z_][0-9a-zA-Z_]{0, 19}更精确地限制了变量的长度是1-20个字符（前面1个字符+后面最多19个字符）。<br>A|B可以匹配A或B，所以(P|p)ython可以匹配’Python’或者’python’。<br>^表示行的开头，^\d表示必须以数字开头。<br>\d$表示必须以数字结束</p>
</blockquote>
<p>参考资料: <a href="https://www.liaoxuefeng.com/wiki/0014316089557264a6b348958f449949df42a6d3a2e542c000/00143193331387014ccd1040c814dee8b2164bb4f064cff000" target="_blank" rel="noopener">廖雪峰的Python教程</a></p>
<h5 id="有意思的拓展链接整理中"><a href="#有意思的拓展链接整理中" class="headerlink" title="有意思的拓展链接整理中"></a>有意思的拓展链接整理中</h5><p><a href="https://mp.weixin.qq.com/s/7Ephzy4NuwuHJwu8M5ur4Q" target="_blank" rel="noopener">八道python练习题</a><br><a href="http://aosabook.org/blog/" target="_blank" rel="noopener">http://aosabook.org/blog/</a></p>
<p><strong><em>以上来自<a href="www.xiniuedu.com">稀牛学院</a>数据科学实训营 寒小阳老师关于python基础的讲解<br>由于课程版权问题，后续会减去课程具体内容，更换为个人学习体验和拓展学习内容</em></strong></p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/python/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> python</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/09/05/一个MeDIP-seq实战/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/09/05/一个MeDIP-seq实战/" itemprop="url">一个MeDIP-seq实战</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-09-05T19:10:14+08:00">2017-09-05</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/seq实战/" itemprop="url" rel="index"><span itemprop="name">seq实战</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/09/05/一个MeDIP-seq实战/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/09/05/一个MeDIP-seq实战/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP018/SRP018845/SRR764931/SRR764931.sra</span><br><span class="line">wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP018/SRP018845/SRR764932/SRR764932.sra</span><br><span class="line">ls *sra |while read id; do fastq-dump --gzip --split-3 -A $id;done</span><br><span class="line">bowtie2 -p 6 -x /home/tingm/refdata/biotree_refdata/bowtie2_index/mm10/mm10 -U SRR764931.sra.fastq.gz | /home/tingm/software/samtools-1.3.1/samtools sort -O bam -o shGFP.bam</span><br><span class="line">bowtie2 -p 6 -x /home/tingm/refdata/biotree_refdata/bowtie2_index/mm10/mm10 -U SRR764932.sra.fastq.gz | /home/tingm/software/samtools-1.3.1/samtools sort -O bam -o shDnmt3L.bam</span><br><span class="line"></span><br><span class="line">/home/tingm/software/samtools-1.3.1/samtools index shDnmt3L.bam</span><br><span class="line">/home/tingm/software/samtools-1.3.1/samtools index shGFP.bam</span><br><span class="line"></span><br><span class="line">macs2 callpeak -t shDnmt3L.bam -m 10 30 -p 1e-5 -f BAM -g mm -n shDnmt3L 2&gt;shDnmt3L.masc2.log</span><br><span class="line">macs2 callpeak -t shGFP.bam -m 10 30 -p 1e-5 -f BAM -g mm -n shGFP 2&gt;shGFP.masc2.log</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">~/.local/bin/bamCoverage -b shDnmt3L.bam -o shDnmt3L.bw #-p 10 --normalizeUsingRPKM</span><br><span class="line">~/.local/bin/bamCoverage -b shGFP.bam -o shGFP.bw #-p 10 --normalizeUsingRPKM</span><br><span class="line"></span><br><span class="line">~/.local/bin/computeMatrix reference-point --referencePoint TSS -b 10000 -a 10000 -R /home/tingm/refdata/biotree_refdata/bowtie2_index/mm10_refSeq.bed -S shDnmt3L.bw --skipZeros -o matrix1_shDnmt3L_TSS.gz</span><br><span class="line">~/.local/bin/plotHeatmap -m matrix1_shDnmt3L_TSS.gz -out shDnmt3L.png</span><br><span class="line"></span><br><span class="line">~/.local/bin/computeMatrix reference-point --referencePoint TSS -b 10000 -a 10000 -R /home/tingm/refdata/biotree_refdata/bowtie2_index/mm10_refSeq.bed -S shGFP.bw --skipZeros -o matrix1_shGFP_TSS.gz</span><br><span class="line">~/.local/bin/plotHeatmap -m matrix1_shGFP_TSS.gz -out shGFP.png</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-411a344663d438d9.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="all files"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-6a55ccb7f1d36457.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="shDnmt3L.png"><br>不知道结果对不对</p>
<blockquote>
<p>MeDIP-seq 跟ChIP-seq的分析手段是一模一样的，同理hMeDIP-seq，caMeDIP-seq等等，都没有本质上的区别，只是用的抗体不一样而已<br><a href="http://www.bio-info-trainee.com/2352.html" target="_blank" rel="noopener">一个MeDIP-seq实战</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/09/02/一个ChIP-seq实战/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/09/02/一个ChIP-seq实战/" itemprop="url">一个ChIP-seq实战</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-09-02T20:10:00+08:00">2017-09-02</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/seq实战/" itemprop="url" rel="index"><span itemprop="name">seq实战</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/09/02/一个ChIP-seq实战/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/09/02/一个ChIP-seq实战/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <p>###软件准备<br>已有sratoolkit,fastqc,bowtie2,samtools,htseq-count,bedtools,R,Rstudio<br>下载macs2,deeptools</p>
<blockquote>
<p><a href="https://bedtools.readthedocs.io/en/latest/content/installation.html" target="_blank" rel="noopener">bedtools官方下载参考</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">#download bedtools</span><br><span class="line">wget https://github.com/arq5x/bedtools2/releases/download/v2.25.0/bedtools-2.25.0.tar.gz</span><br><span class="line">tar -zxvf bedtools-2.25.0.tar.gz</span><br><span class="line">cd bedtools2</span><br><span class="line">make</span><br><span class="line">echo &apos;PATH=$PATH:/home/tingm/practise/biosoft/bedtools2/bin&apos; &gt;&gt; ~/.bashrc</span><br><span class="line">source ~/.bashrc</span><br><span class="line">su yum install BEDTools #root</span><br></pre></td></tr></table></figure></p>
</blockquote>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">#download pyBigWig</span><br><span class="line">pip install pyBigWig --user</span><br></pre></td></tr></table></figure>
<blockquote>
<p><a href="https://github.com/fidelram/deepTools" target="_blank" rel="noopener">deeptools官方网址</a><br><a href="https://deeptools.readthedocs.io/en/latest/content/example_usage.html" target="_blank" rel="noopener">deeptools官方使用教程</a><br><a href="http://www.bio-info-trainee.com/2136.html" target="_blank" rel="noopener">中文教程</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">#download deeptools </span><br><span class="line">conda install deeptools #conda</span><br><span class="line">git clone https://github.com/fidelram/deepTools</span><br><span class="line">cd deepTools</span><br><span class="line">python setup.py install --user</span><br><span class="line">~/.local/bin/deeptools</span><br></pre></td></tr></table></figure></p>
<p>centos本身python为2.6版本，且无pip 进行以下操作<br>参考 <a href="https://my.oschina.net/vaero/blog/210486" target="_blank" rel="noopener">https://my.oschina.net/vaero/blog/210486</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">#download pip 先安装python2.7并设置软连接，指定yum</span><br><span class="line">python</span><br><span class="line">#安装setuptools</span><br><span class="line">wget https://pypi.python.org/packages/26/d1/dc7fe14ce4a3ff3faebf1ac11350de4104ea2d2a80c98393b55c84362b0c/setuptools-32.1.0.tar.gz</span><br><span class="line">tar zxvf setuptools-32.1.0.tar.gz</span><br><span class="line">cd setuptools-32.1.0/</span><br><span class="line">python setup.py build</span><br><span class="line">sudo python setup.py install</span><br><span class="line">#安装pip 参考https://my.oschina.net/vaero/blog/210486</span><br><span class="line">wget --no-check-certificate  https://pypi.python.org/packages/11/b6/abcb525026a4be042b486df43905d6893fb04f05aac21c32c638e939e447/pip-9.0.1.tar.gz</span><br><span class="line">tar zxvf pip-9.0.1.tar.gz</span><br><span class="line">cd pip-9.0.1</span><br><span class="line">python setup.py install(error)</span><br><span class="line">yum install openssl openssl-devel -y</span><br><span class="line">make  # Python-2.7.6目录</span><br><span class="line">make install</span><br><span class="line">easy_install -i http://pypi.douban.com/simple pip</span><br><span class="line">sudo python setup.py install</span><br><span class="line">pip</span><br></pre></td></tr></table></figure></p>
<p><a href="http://liulab.dfci.harvard.edu/MACS/" target="_blank" rel="noopener">MACS2官方网址</a><br><a href="https://pypi.python.org/pypi/MACS2/2.1.1.20160309" target="_blank" rel="noopener">https://pypi.python.org/pypi/MACS2/2.1.1.20160309</a><br><a href="http://www.bio-info-trainee.com/680.html" target="_blank" rel="noopener">MACS2中文教程</a><br>centos python 2.6 上未安装成功<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">pip install MACS2 #error</span><br><span class="line">wget https://pypi.python.org/packages/9f/99/a8ac96b357f6b0a6f559fe0f5a81bcae12b98579551620ce07c5183aee2c/MACS2-2.1.1.20160309.tar.gz</span><br><span class="line">tar zxvf MACS2-2.1.1.20160309.tar.gz</span><br><span class="line">cd MACS2-2.1.1.20160309</span><br><span class="line">python setup.py install --user </span><br><span class="line">#error CRITICAL:Numpy must be installed!</span><br><span class="line">pip install numpy # python 2.7存在问题</span><br><span class="line">~/.local/bin/macs2 --help #未安装成功</span><br></pre></td></tr></table></figure></p>
</blockquote>
<p>###bed下载<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">curl &apos;https://genome.ucsc.edu/cgi-bin/hgTables?hgsid=603641571_MKLxB78UAcjaI3oj5YiFCA0Ms86o&amp;boolshad.hgta_printCustomTrackHeaders=0&amp;hgta_ctName=tb_refGene&amp;hgta_ctDesc=table+browser+query+on+refGene&amp;hgta_ctVis=pack&amp;hgta_ctUrl=&amp;fbQual=whole&amp;fbUpBases=200&amp;fbExonBases=0&amp;fbIntronBases=0&amp;fbDownBases=200&amp;hgta_doGetBed=get+BED&apos; &gt; mm10_refSeq.bed</span><br></pre></td></tr></table></figure></p>
<p>###分析过程 bowtie索引下载要注意，重新下载解压后未报错<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">for x in $(seq 4 9);do ascp -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp-private.ncbi.nlm.nih.gov --user=anonftp --mode=recv /sra/sra-instant/reads/ByStudy/sra/SRP/SRP017/SRP017311/SRR62020&quot;$x&quot;/SRR62020&quot;$x&quot;.sra . ;done</span><br><span class="line">for i in `seq 4 9`;do fastq-dump --gzip --split-3 -A SRR62020$&#123;i&#125;.sra;done</span><br><span class="line">mkdir fastqc_out</span><br><span class="line">fastqc -o fastqc_out --extract *.fastq.gz</span><br><span class="line">ls *.fastq.gz |while read id;do (bowtie2 -p 6 -3 5 --local -x /home/tingm/refdata/biotree_refdata/bowtie2_index/mm10/mm10 -U $&#123;id&#125;| /home/tingm/software/samtools-1.3.1/samtools sort -O bam -o $&#123;id%%.*&#125;.bam);done</span><br></pre></td></tr></table></figure></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-799435398bc98a86.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="sra"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-1a1409af497bbe9f.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="fastq.gz"></p>
<p>###peaking calling<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">macs2 callpeak -c IgGold.bam -t suz12.bam -m 10 30 -p 1e-5 -f BAM -g mm -n suz12 2&gt;suz12.masc2.log</span><br><span class="line">macs2 callpeak -c IgGold.bam -t ring1B.bam -m 10 30 -p 1e-5 -f BAM -g mm -n ring1B 2&gt;ring1B.masc2.log</span><br><span class="line">macs2 callpeak -c IgG.bam -t cbx7.bam -m 10 30 -p 1e-5 -f BAM -g mm -n cbx7 2&gt;cbx7.masc2.log</span><br><span class="line">macs2 callpeak -c IgG.bam -t RYBP.bam -m 10 30 -p 1e-5 -f BAM -g mm -n RYBP 2&gt;RYBP.masc2.log</span><br></pre></td></tr></table></figure></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-653ad38751fdd141.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="bed 比参考资料大很多原因？"></p>
<p>####bam转换成bw文件<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | while read id</span><br><span class="line">do</span><br><span class="line">file=$(basename $id )</span><br><span class="line">sample=$&#123;file%%.*&#125;</span><br><span class="line">echo $sample</span><br><span class="line">/home/tingm/software/samtools-1.3.1/samtools index $id $id.bai</span><br><span class="line">~/.local/bin/bamCoverage -b $id -o $sample.bw ## 这里有个参数，-p 10 --normalizeUsingRPKM</span><br><span class="line">~/.local/bin/computeMatrix reference-point --referencePoint TSS -b 10000 -a 10000 -R /home/tingm/refdata/biotree_refdata/bowtie2_index/mm10_refSeq.bed -S $sample.bw --skipZeros -o matrix1_$&#123;sample&#125;_TSS.gz --outFileSortedRegions regions1_$&#123;sample&#125;_genes.bed</span><br><span class="line">~/.local/bin/plotHeatmap -m matrix1_$&#123;sample&#125;_TSS.gz -out $&#123;sample&#125;.png</span><br><span class="line">done</span><br></pre></td></tr></table></figure></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-d600963c06989821.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="bw"></p>
<p>####画基因的TSS附近的和genebody附近的profile和heatmap图<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">~/.local/bin/computeMatrix reference-point -p 10 --referencePoint TSS -b 2000 -a 2000 -S *bw -R /home/tingm/refdata/biotree_refdata/bowtie2_index/mm10_refSeq.bed --skipZeros -o tmp4.mat.gz</span><br><span class="line">~/.local/bin/plotHeatmap -m tmp4.mat.gz -out tmp4.merge.png</span><br><span class="line">~/.local/bin/plotProfile --dpi 720 -m tmp4.mat.gz -out tmp4.profile.pdf --plotFileFormat pdf --perGroup</span><br><span class="line">~/.local/bin/plotHeatmap --dpi 720 -m tmp4.mat.gz -out tmp4.merge.pdf --plotFileFormat pdf</span><br><span class="line">修改了bed文件地址</span><br><span class="line">~/.local/bin/computeMatrix scale-regions -p 6 -S *bw -R /home/tingm/my_refdata/biotree_refdata/bowtie2_index/mm10_refSeq.bed -b 3000 -a 3000 -m 5000 --skipZeros -o tmp5.mat.gz</span><br><span class="line">~/.local/bin/plotHeatmap -m tmp5.mat.gz -out tmp5.merge.png</span><br><span class="line">~/.local/bin/plotProfile --dpi 720 -m tmp5.mat.gz -out tmp5.profile.pdf --plotFileFormat pdf --perGroup</span><br><span class="line">~/.local/bin/plotHeatmap --dpi 720 -m tmp5.mat.gz -out tmp5.merge.pdf --plotFileFormat pdf</span><br></pre></td></tr></table></figure></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-31b32871a5651511.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="profile"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-197a1c9d806d58ae.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="tmp4.merge.png"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-cac3a547fc9a9b63.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="profile"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-002424d613ddd9a1.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="tmp5.merge.png"></p>
<blockquote>
<p><a href="https://my.oschina.net/vaero/blog/210486" target="_blank" rel="noopener">CentOS升级Python2.7及安装pip</a><br><a href="https://mp.weixin.qq.com/s/iHnogzM6QSu8tlImE26_Wg" target="_blank" rel="noopener">一文学会ChIP-Seq数据分析（想想也不可能）</a><br><a href="https://mp.weixin.qq.com/s/_A0rHldzEgVk7bgwt457qQ" target="_blank" rel="noopener">ChIP-seq实战分析</a><br><a href="https://mp.weixin.qq.com/s/SK6GVqSTAwZCirMP1GlRHQ" target="_blank" rel="noopener">生物信息学常见1000个软件的安装代码！</a><br><a href="http://www.biotrainee.com/thread-2013-1-1.html" target="_blank" rel="noopener">ChIP-seq基础入门传送门</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/28/一个RNA-seq实战/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/28/一个RNA-seq实战/" itemprop="url">一个RNA-seq实战</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-28T20:09:43+08:00">2017-08-28</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/seq实战/" itemprop="url" rel="index"><span itemprop="name">seq实战</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/28/一个RNA-seq实战/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/28/一个RNA-seq实战/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">#! usr/bin/bash</span><br><span class="line">set -u</span><br><span class="line">set -e</span><br><span class="line">set -o pipefail</span><br><span class="line">hg19_ref=~/fastq/hisat2_index/hg19/genome</span><br><span class="line">for x in `seq 677 680`;do ascp -T -l 200M -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh --host=ftp-private.ncbi.nlm.nih.gov --user=anonftp --mode=recv /sra/sra-instant/reads/ByStudy/sra/SRP/SRP029/SRP029245/SRR957&quot;$x&quot;/SRR957&quot;$x&quot;.sra .;done</span><br><span class="line">##ls *sra |while read id; do fastq-dump --split-3 $id;done</span><br><span class="line">ls *sra |while read id; do fastq-dump --gzip --split-3 -A $id ;done</span><br><span class="line">## ls *fastq |xargs fastqc -t 10</span><br><span class="line">ls *.fastq.gz|while read id;do(hisat2 -t -p 5 -x $hg19_ref -U $id 2&gt;$&#123;id%%.*&#125;.map.log | /home/tingm/software/samtools-1.3.1/samtools view -Sb  - &gt;$&#123;id%%.*&#125;.bam);done</span><br><span class="line">ls *.bam|while read id;do /home/tingm/software/samtools-1.3.1/samtools sort -n -@ 5 $id -o $&#123;id%%.*&#125;.Nsorted.bam;done</span><br><span class="line">ls *.Nsorted.bam |while read id;do htseq-count -f bam -s no -i gene_name $id ~/fastq/refdata/Genecode/gencode.v26lift37.annotation.gff3 1&gt;$&#123;id%%.*&#125;.geneCounts 2&gt;$&#123;id%%.*&#125;.HTseq.log;done</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-ef5cc558fa035ea6.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="sra"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-ae125a7bba77e247.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="gz"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-4686ec7c0e85f6d2.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="map.log"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-f6963429b31bd0dd.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="bam Nsort.bam"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-db82a63d605483a7.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="genecounts"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-bcf87ea38bdef212.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="genecounts"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br></pre></td><td class="code"><pre><span class="line">#download deseq</span><br><span class="line">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">biocLite(&quot;DESeq&quot;)</span><br><span class="line">#载入R示范数据</span><br><span class="line">biocLite(&quot;pasilla&quot;)</span><br><span class="line">#查看r下载位置</span><br><span class="line">.libPaths()</span><br><span class="line">#转到pasilla下载位置</span><br><span class="line">setwd(&quot;/home/tingm/R/x86_64-pc-linux-gnu-library/3.2/pasilla/&quot;)</span><br><span class="line">#设置路径</span><br><span class="line">datafile =system.file(&quot;extdata/pasilla_gene_counts.tsv&quot;,package=&quot;pasilla&quot;)                                              </span><br><span class="line">#读入数据</span><br><span class="line">pasillaCounttable=read.table(datafile,header=TRUE,row.names=1)  </span><br><span class="line">dim(pasillaCounttable)      </span><br><span class="line">#差异分析</span><br><span class="line">pasillaDesign = data.frame(row.names = colnames(pasillaCounttable),condition = c(&quot;untreated&quot;,&quot;untreated&quot;,&quot;untreated&quot;,&quot;untreated&quot;,&quot;treated&quot;,&quot;treated&quot;,&quot;treated&quot;),libType = c(&quot;single-end&quot;,&quot;single-end&quot;,&quot;paired-end&quot;,&quot;paired-end&quot;,&quot;single-end&quot;,&quot;paired-end&quot;,&quot;paired-end&quot;)) </span><br><span class="line">de = newCountDataSet(pasillaCounttable,pasillaDesign[,1]) </span><br><span class="line">de=estimateSizeFactors(de)</span><br><span class="line">de=estimateDispersions(de)</span><br><span class="line">#res是表达量检测结果</span><br><span class="line">res=nbinomTest(de,&quot;untreated&quot;,&quot;treated&quot;)</span><br><span class="line">uniq=na.omit(res)</span><br><span class="line">#输出差异显著值</span><br><span class="line">sum(na.omit(res$padj&lt;0.05))</span><br><span class="line">#输出表格</span><br><span class="line">write.table(res,file=&quot;file.csv&quot;,sep=&quot;,&quot;)</span><br><span class="line"></span><br><span class="line">#数据分析代码</span><br><span class="line">library(DESeq)</span><br><span class="line">C1=read.table(&quot;control_1.geneCounts&quot;,row.names=1)</span><br><span class="line">C2=read.table(&quot;control_2.geneCounts&quot;,row.names=1)</span><br><span class="line">S1=read.table(&quot;siSUZ12_1.geneCounts&quot;,row.names=1)</span><br><span class="line">S2=read.table(&quot;siSUZ12_2.geneCounts&quot;,row.names=1)</span><br><span class="line">data=cbind(C1,C2,S1,S2)</span><br><span class="line">n=nrow(data)</span><br><span class="line">#删除后四行</span><br><span class="line">data&lt;-data[-c(58523:58527),]</span><br><span class="line">#差异分析</span><br><span class="line">colnames(data)=c(&quot;C1&quot;,&quot;C2&quot;,&quot;S1&quot;,&quot;S2&quot;)</span><br><span class="line">type=rep(c(&quot;C&quot;,&quot;S&quot;),c(2,2))</span><br><span class="line">de=newCountDataSet(data,type)</span><br><span class="line">de=estimateSizeFactors(de)#locfunc = genefilter :: shorth数据集中0太多</span><br><span class="line">de=estimateDispersions(de)</span><br><span class="line">res=nbinomTest(de,&quot;C&quot;,&quot;S&quot;)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"># DESeq2 </span><br><span class="line">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">biocLite(&quot;DESeq2&quot;)</span><br><span class="line">library(DESeq2)</span><br><span class="line">biocLite(&quot;limma&quot;)</span><br><span class="line">library(limma)</span><br><span class="line">library(pasilla)</span><br><span class="line">data(pasillaGenes)</span><br><span class="line">exprSet=counts(pasillaGenes)  ##做好表达矩阵</span><br><span class="line">group_list=pasillaGenes$condition##做好分组因子即可</span><br><span class="line">colData &lt;- data.frame(row.names=colnames(exprSet), group_list=group_list)</span><br><span class="line">dds &lt;- DESeqDataSetFromMatrix(countData = exprSet,</span><br><span class="line">colData = colData,design = ~ group_list)</span><br><span class="line">dds2 &lt;- DESeq(dds)  ##第二步，直接用DESeq函数即可</span><br><span class="line">resultsNames(dds2)</span><br><span class="line">res &lt;-  results(dds2, contrast=c(&quot;group_list&quot;,&quot;treated&quot;,&quot;untreated&quot;))</span><br><span class="line">## 提取你想要的差异分析结果，我们这里是treated组对untreated组进行比较</span><br><span class="line">resOrdered &lt;- res[order(res$padj),]</span><br><span class="line">resOrdered=as.data.frame(resOrdered)</span><br><span class="line"></span><br><span class="line">#实战数据分析</span><br><span class="line">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">biocLite(&quot;org.Hs.eg.db&quot;)</span><br><span class="line">library(org.Hs.eg.db)</span><br><span class="line">keytypes(org.Hs.eg.db)</span><br><span class="line">tmp=select(org.Hs.eg.db,keys=res$id,columns=c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;,&quot;GENENAME&quot;,&quot;ENSEMBL&quot;), keytype=&quot;SYMBOL&quot;)</span><br><span class="line">#合并res和tmp</span><br><span class="line">res=merge(tmp,res,by.x=&quot;SYMBOL&quot;,by.y=&quot;id&quot;,all=TRUE)</span><br><span class="line">#go</span><br><span class="line">tmp=select(org.Hs.eg.db, keys=res$ENSEMBL, columns=&quot;GO&quot;, keytype=&quot;ENSEMBL&quot;)</span><br><span class="line">ensembl_go=unlist(tapply(tmp[,2],as.factor(tmp[,1]),function(x) paste(x,collapse =&quot;|&quot;),simplify =F))</span><br><span class="line">#为res加入go注释，</span><br><span class="line">res$go=ensembl_go[res$ENSEMBL]#为res加入一列go</span><br><span class="line">#写入all——data</span><br><span class="line">all_res=res</span><br><span class="line">write.csv(res,file=&quot;all_data.csv&quot;,row.names =F)</span><br><span class="line">uniq=na.omit(res)#删除无效基因</span><br><span class="line">sort_uniq=uniq[order(uniq$padj),]#按照矫正p值排序</span><br><span class="line">#写入排序后的all_data</span><br><span class="line">write.csv(res,file=&quot;all_data.csv&quot;,row.names =F)</span><br><span class="line">#标记上下调基因</span><br><span class="line">sort_uniq$up_down=ifelse(sort_uniq$baseMeanA&gt;sort_uniq$baseMeanB,&quot;up&quot;,&quot;down&quot;)</span><br><span class="line">#交换上下调基因列位置</span><br><span class="line">final_res=sort_uniq[,c(12,1:11)]</span><br><span class="line">#写出最后数据</span><br><span class="line">write.csv(final_res,file=&quot;final_annotation_gene_bedtools_sort_uniq.csv&quot;,row.names =F)</span><br><span class="line">#然后挑选出padj值小于0.05的数据来做富集</span><br><span class="line">tmp=select(org.Hs.eg.db, keys=sort_uniq[sort_uniq$padj&lt;0.05,1], columns=&quot;ENTREZID&quot;, keytype=&quot;SYMBOL&quot;)</span><br><span class="line">diff_ENTREZID=tmp$ENTREZID</span><br><span class="line">diff_SYMBOL=tmp$SYMBOL</span><br><span class="line">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">biocLite(&quot;DOSE&quot;)</span><br><span class="line">biocLite(&quot;clusterProfiler&quot;)</span><br><span class="line">require(DOSE)</span><br><span class="line">require(clusterProfiler)</span><br><span class="line">diff_ENTREZID=na.omit(diff_ENTREZID)</span><br><span class="line">diff_SYMBOL=na.omit(diff_SYMBOL)</span><br><span class="line">ego &lt;-enrichGO(gene=diff_ENTREZID,OrgDb=org.Hs.eg.db,ont=&quot;CC&quot;,pvalueCutoff=0.05)  ##error</span><br><span class="line">ekk &lt;- enrichKEGG(gene=diff_ENTREZID,organism = &apos;hsa&apos;,pvalueCutoff = 0.05)</span><br><span class="line">write.csv(summary(ekk),&quot;KEGG-enrich.csv&quot;,row.names =F)</span><br><span class="line">write.csv(summary(ego),&quot;GO-enrich.csv&quot;,row.names =F)</span><br></pre></td></tr></table></figure>
<p>###goenrich有错误 后续差异分析如何进行得到的结果？</p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-5b18f8068dc254cb.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt=""></p>
<blockquote>
<p>R语言DESeq找差异基因 <a href="http://www.bio-info-trainee.com/741.html" target="_blank" rel="noopener">http://www.bio-info-trainee.com/741.html</a><br>用DESeq进行差异分析的源代码 <a href="http://www.bio-info-trainee.com/867.html" target="_blank" rel="noopener">http://www.bio-info-trainee.com/867.html</a><br>用R语言的DESeq2包来对RNA-seq数据做差异分析 <a href="http://www.bio-info-trainee.com/1533.html#more-1533" target="_blank" rel="noopener">http://www.bio-info-trainee.com/1533.html#more-1533</a><br>Bioconductor annotation packages <a href="https://davetang.org/muse/2013/12/16/bioconductor-annotation-packages/" target="_blank" rel="noopener">https://davetang.org/muse/2013/12/16/bioconductor-annotation-packages/</a><br>clusterProfiler <a href="https://bioconductor.org/packages/release/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html" target="_blank" rel="noopener">https://bioconductor.org/packages/release/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/23/转录组入门——8/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/23/转录组入门——8/" itemprop="url">转录组入门——8</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-23T18:02:51+08:00">2017-08-23</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/转录组入门/" itemprop="url" rel="index"><span itemprop="name">转录组入门</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/23/转录组入门——8/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/23/转录组入门——8/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <blockquote>
<p>选择p&lt;0.05而且abs(logFC)大于一个与众的基因为显著差异表达基因集，对这个基因集用R包做KEGG/GO超几何分布检验分析。<br>然后把表达矩阵和分组信息分别作出cls和gct文件，导入到GSEA软件分析。<br>基本任务是完成这个分析。</p>
</blockquote>
<h5 id="GO-KEGG分析-clusterProfiler"><a href="#GO-KEGG分析-clusterProfiler" class="headerlink" title="GO/KEGG分析 clusterProfiler"></a>GO/KEGG分析 clusterProfiler</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">biocLite(&quot;DOSE&quot;)</span><br><span class="line">require(DOSE)</span><br><span class="line">require(clusterProfiler)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line"></span><br><span class="line">##去除ID的版本号</span><br><span class="line">diff_gene_ENSEMBL &lt;- unlist(lapply(diff_gene, function(x)&#123;strsplit(x, &quot;\\.&quot;)[[1]][1]&#125;))</span><br><span class="line">##GOid mapping + GO富集</span><br><span class="line">## R版本 在服务器上R 无法使用</span><br><span class="line">ego &lt;- enrichGO(gene = diff_gene_ENSEMBL, OrgDb = org.Mm.eg.db,</span><br><span class="line">                                keytype = &quot;ENSEMBL&quot;, ont = &quot;BP&quot;, pAdjustMethod = &quot;BH&quot;,</span><br><span class="line">                                pvalueCutoff = 0.01, qvalueCutoff = 0.05)</span><br><span class="line">##查看富集结果数据</span><br><span class="line">enrich_go &lt;- as.data.frame(ego)</span><br><span class="line">##作图</span><br><span class="line">barplot(ego, showCategory=10)</span><br><span class="line">dotplot(ego)</span><br><span class="line">enrichMap(ego)</span><br><span class="line">plotGOgraph(ego)</span><br><span class="line"></span><br><span class="line">##ID转化</span><br><span class="line">ids &lt;- bitr(diff_gene_ENSEMBL, fromType = &quot;ENSEMBL&quot;, toType = &quot;ENTREZID&quot;, OrgDb = &quot;org.Mm.eg.db&quot;)</span><br><span class="line">kk &lt;- enrichKEGG(gene = ids[,2], organism = &quot;mmu&quot;, keyType = &quot;kegg&quot;,pvalueCutoff = 0.05, pAdjustMethod = &quot;BH&quot;, qvalueCutoff = 0.1)</span><br><span class="line">##查看富集结果数据</span><br><span class="line">enrich_kegg &lt;- as.data.frame(kk)</span><br><span class="line">##作图</span><br><span class="line">dotplot(kk)</span><br></pre></td></tr></table></figure>
<h5 id="GSEA"><a href="#GSEA" class="headerlink" title="GSEA"></a>GSEA</h5><blockquote>
<p>基因集富集分析 (Gene Set Enrichment Analysis, GSEA) 的基本思想是使用预定义的基因集（通常来自功能注释或先前实验的结果），将基因按照在两类样本中的差异表达程度排序，然后检验预先设定的基因集合是否在这个排序表的顶端或者底端富集。基因集合富集分析检测基因集合而不是单个基因的表达变化，因此可以包含这些细微的表达变化，预期得到更为理想的结果。</p>
</blockquote>
<h5 id="Gene-Set-Enrichment-Analysis（GSEA）"><a href="#Gene-Set-Enrichment-Analysis（GSEA）" class="headerlink" title="Gene Set Enrichment Analysis（GSEA）"></a>Gene Set Enrichment Analysis（GSEA）</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">diff_gene_DESeq2 &lt;- subset(res, padj &lt; 0.05 &amp; (log2FoldChange &gt; 1 | log2FoldChange &lt; -1))</span><br><span class="line">diff_gene_ENSEMBL &lt;- unlist(lapply(rownames(diff_gene_DESeq2), function(x)&#123;strsplit(x, &quot;\\.&quot;)[[1]][1]&#125;))</span><br><span class="line">rownames(diff_gene_DESeq2) &lt;- diff_gene_ENSEMBL</span><br><span class="line">diff_gene_ENTREZID &lt;- select(org.Mm.eg.db,keys = diff_gene_ENSEMBL,columns =&quot;ENTREZID&quot;,keytype = &quot;ENSEMBL&quot;)</span><br><span class="line">diff_gene_DESeq2$ENTREZID &lt;- diff_gene_ENTREZID[,2]</span><br><span class="line"># 获取按照log2FC大小来排序的基因列表</span><br><span class="line">genelist &lt;- diff_gene_DESeq2$log2FoldChange</span><br><span class="line">names(genelist) &lt;- diff_gene_DESeq2$ENTREZID</span><br><span class="line">genelist &lt;- sort(genelist, decreasing = TRUE)</span><br><span class="line"># GSEA分析（具体参数参考：https://mp.weixin.qq.com/s/p-n5jq5Rx2TqDBStS2nzoQ）</span><br><span class="line">gsemf &lt;- gseGO(genelist,OrgDb = org.Mm.eg.db,ont=&quot;MF&quot;)</span><br><span class="line"># 查看大致信息</span><br><span class="line">head(gsemf)</span><br><span class="line"># 画出GSEA图</span><br><span class="line">gseaplot(gsemf, geneSetID=&quot;GO:0000977&quot;)</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-d045f82c5453691f.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="go"></p>
<blockquote>
<p><a href="http://www.biotrainee.com/thread-2018-1-1.html" target="_blank" rel="noopener">转录组入门（7-8）从差异基因到富集分析</a><br><a href="https://mp.weixin.qq.com/s/p-n5jq5Rx2TqDBStS2nzoQ" target="_blank" rel="noopener">从零开始学转录组（8）：富集分析</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/21/转录组入门——7/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/21/转录组入门——7/" itemprop="url">转录组入门——7</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-21T16:02:38+08:00">2017-08-21</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/转录组入门/" itemprop="url" rel="index"><span itemprop="name">转录组入门</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/21/转录组入门——7/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/21/转录组入门——7/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">.libPaths(&quot;/home/tingm/R/x86_64-pc-linux-gnu-library/3.2&quot;)</span><br><span class="line">  library(limma)</span><br><span class="line">  library(edgeR)</span><br><span class="line">  library(DESeq2)</span><br><span class="line">    #change</span><br><span class="line">    path&lt;-&quot;/home/tingm/fastq/mus_result_count/&quot;</span><br><span class="line">    #change</span><br><span class="line">      samples&lt;-read.table(&quot;samples.list.txt&quot;,header=F,sep=&quot;\t&quot;)</span><br><span class="line">      samples&lt;-samples[,1]</span><br><span class="line">      #change</span><br><span class="line">      data.raw&lt;-read.table(&quot;Mus_SRR3589959.sgeneCounts&quot;,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;Mus_SRR3589959.sgeneCounts&quot;))</span><br><span class="line">      for(sample in samples)</span><br><span class="line">      &#123;</span><br><span class="line">        print(sample)</span><br><span class="line">        file&lt;-paste(path,sample,sep=&quot;&quot;)</span><br><span class="line">        data.new&lt;-read.table(file,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,sample))</span><br><span class="line">        data.raw&lt;-merge(data.new,data.raw,by=&quot;gene_id&quot;)</span><br><span class="line">      &#125;</span><br><span class="line">     data.raw&lt;-data.raw[-1:-5,]</span><br><span class="line">        #change</span><br><span class="line">     names(data.raw) &lt;- c(&quot;gene_id&quot;,&quot;Akap95_2&quot;,&quot;control_2&quot;,&quot;Akap95_1&quot;,&quot;control_1&quot;)</span><br><span class="line">     write.table(data.raw,&quot;count.matix.csv&quot;,row.names=F,quote=F)</span><br></pre></td></tr></table></figure>
<h5 id="使用DESeq2进行差异基因分析-筛选出基因数目不一样"><a href="#使用DESeq2进行差异基因分析-筛选出基因数目不一样" class="headerlink" title="使用DESeq2进行差异基因分析 筛选出基因数目不一样"></a>使用DESeq2进行差异基因分析 筛选出基因数目不一样</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">##数据预处理</span><br><span class="line">database &lt;- read.table(file = &quot;count.matix.csv&quot;, sep = &quot; &quot;, header = T,row.names = 1)</span><br><span class="line">database &lt;- round(as.matrix(database))</span><br><span class="line">##设置分组信息并构建dds对象</span><br><span class="line">condition &lt;- factor(c(&quot;Akap95&quot;,&quot;control&quot;,&quot;Akap95&quot;,&quot;control&quot;), levels = c(&quot;control&quot;, &quot;Akap95&quot;))</span><br><span class="line">coldata &lt;- data.frame(row.names = colnames(database), condition)</span><br><span class="line">dds &lt;- DESeqDataSetFromMatrix(countData=database, colData=coldata, design=~condition)</span><br><span class="line">dds &lt;- dds[ rowSums(counts(dds)) &gt; 1, ]</span><br><span class="line">##使用DESeq函数估计离散度，然后差异分析获得res对象</span><br><span class="line">dds &lt;- DESeq(dds)</span><br><span class="line">res &lt;- results(dds)</span><br><span class="line">#summary(res)</span><br><span class="line">#最后设定阈值，筛选差异基因，导出数据(全部数据。包括标准化后的count数)</span><br><span class="line">res &lt;- res[order(res$padj),]</span><br><span class="line">diff_gene_DESeq2 &lt;- subset(res, padj &lt; 0.05 &amp; (log2FoldChange &gt; 1 | log2FoldChange &lt; -1))</span><br><span class="line">diff_gene_DESeq2 &lt;- row.names(diff_gene_DESeq2)</span><br><span class="line">resdata &lt;- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)),by=&quot;row.names&quot;,sort=FALSE)</span><br><span class="line">write.csv(resdata,file = &quot;control_vs_Akap95.csv&quot;,row.names = F)</span><br></pre></td></tr></table></figure>
<h5 id="使用edgeR进行差异基因分析"><a href="#使用edgeR进行差异基因分析" class="headerlink" title="使用edgeR进行差异基因分析"></a>使用edgeR进行差异基因分析</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">##跟DESeq2一样，导入数据，预处理（用了cpm函数）</span><br><span class="line">exprSet&lt;- read.table(file = &quot;count.matix.csv&quot;, sep = &quot; &quot;, header = TRUE, row.names = 1, stringsAsFactors = FALSE)</span><br><span class="line">group_list &lt;- factor(c(&quot;Akap95&quot;,&quot;control&quot;,&quot;Akap95&quot;,&quot;control&quot;), levels = c(&quot;control&quot;, &quot;Akap95&quot;))</span><br><span class="line">exprSet &lt;- exprSet[rowSums(cpm(exprSet) &gt; 1) &gt;= 2,]</span><br><span class="line">##设置分组信息，并做TMM标准化</span><br><span class="line">exprSet &lt;- DGEList(counts = exprSet, group = group_list)</span><br><span class="line">exprSet &lt;- calcNormFactors(exprSet)</span><br><span class="line">##使用qCML（quantile-adjusted conditional maximum likelihood）估计离散度（只针对单因素实验设计）</span><br><span class="line">exprSet &lt;- estimateCommonDisp(exprSet)</span><br><span class="line">exprSet &lt;- estimateTagwiseDisp(exprSet)</span><br><span class="line">##寻找差异gene(这里的exactTest函数还是基于qCML并且只针对单因素实验设计)，然后按照阈值进行筛选即可</span><br><span class="line">et &lt;- exactTest(exprSet)</span><br><span class="line">tTag &lt;- topTags(et, n=nrow(exprSet))</span><br><span class="line">diff_gene_edgeR &lt;- subset(tTag$table, FDR &lt; 0.05 &amp; (logFC &gt; 1 | logFC &lt; -1))</span><br><span class="line">diff_gene_edgeR &lt;- row.names(diff_gene_edgeR)</span><br><span class="line">write.csv(tTag$table,file = &quot;control_vs_Akap95_edgeR.csv&quot;)</span><br></pre></td></tr></table></figure>
<h5 id="使用limma进行差异分析-limma的voom流程"><a href="#使用limma进行差异分析-limma的voom流程" class="headerlink" title="使用limma进行差异分析   limma的voom流程"></a>使用limma进行差异分析   limma的voom流程</h5><p>待续</p>
<h5 id="不同软件包分析结果比较"><a href="#不同软件包分析结果比较" class="headerlink" title="不同软件包分析结果比较"></a>不同软件包分析结果比较</h5><p>待续</p>
<blockquote>
<p><a href="http://www.jianshu.com/p/3bfb21d24b74" target="_blank" rel="noopener">转录组入门（7）：差异基因分析</a><br><a href="https://mp.weixin.qq.com/s/1mda7NpOxShbOxZKSJar7Q" target="_blank" rel="noopener">从零开始学转录组（7）：差异基因表达分析</a><br><a href="http://www.biotrainee.com/thread-2018-1-1.html" target="_blank" rel="noopener">转录组入门（7-8）从差异基因到富集分析</a><br><a href="http://www.agctu.org/question/179" target="_blank" rel="noopener">为什么RNA-seq的read数服从泊松分布？</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  
  <nav class="pagination">
    <a class="extend prev" rel="prev" href="/"><i class="fas fa-angle-left"></i></a><a class="page-number" href="/">1</a><span class="page-number current">2</span><a class="page-number" href="/page/3/">3</a><a class="extend next" rel="next" href="/page/3/"><i class="fas fa-angle-right"></i></a>
  </nav>
  
  
  

<div class="comments" id="comments">
  
  
  <script type="text/javascript">
  _hcwp = window._hcwp || [];
  _hcwp.push({ "widget": "Bloggerstream", "widget_id": "tingm831@gmail.com", "selector": ".hc-comment-count", "label": "{\%COUNT%\}" });
  
  (function() {
    if ("HC_LOAD_INIT" in window) {
      return;
    }
    HC_LOAD_INIT = true;
    var lang = (navigator.language || navigator.systemLanguage || navigator.userLanguage || "en").substr(0, 2).toLowerCase();
    var hcc = document.createElement("script");
    hcc.type = "text/javascript";
    hcc.defer = true;
    hcc.src = ("https:" === document.location.protocol ? "https" : "http") + "://w.hypercomments.com/widget/hc/tingm831@gmail.com/" + lang + "/widget.js";
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(hcc, s.nextSibling);
  })();
  </script>
  
</div>



  
</section>

          </div>
          
          
          
<aside class="sidebar" id="sidebar">
  
  <div class="search">
    <div class="form-group">
      <i class="fas fa-search"></i> <input type="text" id="search-input" name="q" results="0" placeholder="Search" class="form-control"/>
    </div>
  </div>
  <div class="search-result-box" id="search-result"></div>
  
  
<div class="info sidebar-item" id="info">
  
  <img class="author-avatar" src="/images/avatar.png" alt="mt">
  
  <p class="author-name">mt</p>
  <p class="author-description"> </p>
  <div class="site-count">
    
    <div class="archives-count">
      <p>Archives</p>
      <p><a href="/archives/">26</a></p>
    </div>
    
    
    
    <span class="site-count-divider"> | </span>
    
    <div class="categories-count">
      <p>Categories</p>
      <p><a href="/categories/">3</a></p>
    </div>
    
    
    
    <span class="site-count-divider"> | </span>
    
    <div class="tags-count">
      <p>Tags</p>
      <p><a href="/tags/">2</a></p>
    </div>
    
  </div>
  
  <div class="rss">
    <a class="rss-link button sidebar-item" href="/atom.xml"><i class="fas fa-rss"></i> RSS</a>
  </div>
  
</div>


  <div class="sidebar-sticky">
    
    
    
<hr>
<div class="social-link sidebar-item">
  <p><i class="far fa-address-card"></i> Social Links</p>
  <ul>
    
    <li><i class="fab fa-github"></i> <a href="https://github.com/mt831" target="_blank">GitHub</a></li>
    
  </ul>
</div>


    
    
  </div>
</aside>


          
        </div>
      </div>
    </main>
    
<footer id="footer" class="footer">
  <div class="container">
    <div class="back-to-top">
      <a id="back-to-top"><i class="fas fa-angle-double-up"></i></a>
    </div>
    <div class="footer-container">
      <div class="footer-left">
        <div class="copyright">
          <span class="author">mt</span>
          <span class="year"><i class="far fa-copyright"></i> 2018-4-1 - 2018</span>
          
        </div>
        
        
<div class="busuanzi">
  <span id="busuanzi_container_site_pv"><i class="fas fa-eye"></i> <span id="busuanzi_value_site_pv"></span></span>
  <span id="busuanzi_container_site_uv"><i class="fas fa-user"></i> <span id="busuanzi_value_site_uv"></span></span>
  <span id="busuanzi_container_page_pv"><i class="far fa-file-alt"></i> <span id="busuanzi_value_page_pv"></span></span>
</div>


        
      </div>
      <div class="footer-right">
        <div class="custom-info">
          
        </div>
        <div class="powered-by">
          Proudly Powered by <a href="https://hexo.io/" target="_blank">Hexo</a> | Theme is <a href="https://github.com/AlynxZhou/hexo-theme-aria/" target="_blank">ARIA</a>
        </div>
      </div>
    </div>
  </div>
</footer>


  </body>
</html>
