<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    
    <meta name="theme-color" content="#33363b">
    <meta name="msapplication-TileColor" content="#33363b">
    
    
    
    <meta name="keywords" content="python, R, math, bioinformatics">
    
    
    <link rel="apple-touch-icon" sizes="180x180" href="/favicons/apple-touch-icon.png">
    
    
    <link rel="icon" type="image/png" sizes="192x192" href="/favicons/android-chrome-192x192.png">
    
    
    <link rel="icon" type="image/png" sizes="32x32" href="/favicons/favicon-32x32.png">
    
    
    <link rel="icon" type="image/png" sizes="16x16" href="/favicons/favicon-16x16.png">
    
    
    <link rel="mask-icon" href="/favicons/safari-pinned-tab.svg" color="#33363b">
    
    
    <link rel="manifest" href="/favicons/site.webmanifest">
    
    
    <meta name="msapplication-config" content="/favicons/browserconfig.xml">
    
    
    <link rel="alternate" href="/atom.xml" title="Hexo" type="application/atom+xml" />
    
    <link rel="stylesheet" type="text/css" href="/css/index.css">
    
    <link rel="stylesheet" type="text/css" href="/css/sidebar.css">
    
    
    <link rel="stylesheet" type="text/css" href="/css/custom.css">
    <link rel="stylesheet" type="text/css" href="/css/normalize.css">
    <link rel="stylesheet" type="text/css" href="/css/atom-one-dark.css">
    <link rel="stylesheet" type="text/css" href="/css/lightgallery.min.css">
    <script type="text/javascript" src="/js/jquery.min.js"></script>
    <script defer type="text/javascript" src="/js/util.js"></script>
    <script defer type="text/javascript" src="/js/scrollspy.js"></script>
    <script defer type="text/javascript" src="/js/fontawesome-all.min.js"></script>
    <script defer type="text/javascript" src="/js/lightgallery.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-fullscreen.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-hash.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-pager.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-thumbnail.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-zoom.min.js"></script>
    
    <script defer src="/js/busuanzi.pure.mini.js"></script>
    
    
    <script defer type="text/javascript" src="/js/search.js"></script>
    <script type="text/javascript">
    $(document).ready(function () {
      var searchPath = "search.xml";
      if (searchPath.length === 0) {
        searchPath = "search.xml";
      }
      var path = "/" + searchPath;
      searchFunc(path, "search-input", "search-result");
    });
    </script>
    
    
    <script defer type="text/javascript" src="/js/index.js"></script>
    
    <script defer type="text/javascript" src="/js/custom.js"></script>
    <title>Hexo</title>
  </head>
  <body itemscope itemtype="http://schema.org/WebPage" lang=""  data-spy="scroll" data-target=".list-group">
    
<header id="header" class="header">
  <div class="container">
    <div class="header-container">
      <div class="header-title">
        <h1 class="title"><a href="/">Hexo</a></h1>
        <h2 class="subtitle"></h2>
      </div>
      <div class="logo">
        <img src="/images/ARIA_logo.png" alt="logo">
      </div>
    </div>
    
<nav id="nav" class="nav">
  <a id="nav-toggle" class="nav-toggle"><i class="fas fa-bars"></i></a>
  <ul id="menu">
    
    <li><a href="/">Home</a></li>
    
    <li><a href="/archives/">Archives</a></li>
    
    <li><a href="/categories/">Categories</a></li>
    
    <li><a href="/tags/">Tags</a></li>
    
    <li><a href="/about/">About</a></li>
    
  </ul>
</nav>


  </div>
</header>


    <main id="main" class="main">
      <div class="container">
        <div class="main-container">
          <div class="content">
            

<section id="index" class="index">
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2018/04/03/L2.Pandas-Series-DataFrame/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2018/04/03/L2.Pandas-Series-DataFrame/" itemprop="url">L2 Pandas Series & DataFrame 关键点记录</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2018-04-03T19:43:53+08:00">2018-04-03</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2018/04/03/L2.Pandas-Series-DataFrame/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2018/04/03/L2.Pandas-Series-DataFrame/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="Series-一维数组"><a href="#Series-一维数组" class="headerlink" title="Series 一维数组"></a>Series 一维数组</h5><p>构建初始化：可用list构建（dict也可），默认用0-n为Series的index，也可指定<br>位置索引/名称索引：s[[1,2,3]] 内为列表<br>数学运算（广播特性）：list不可（list相加并在一起），基于索引进行计算（list相加并在一起）<br>in判断是否在索引内    .get看值是否有    条件索引取值<br>统计计算：min max mean等    可和numpy结合使用 如np.log()<br>赋值：可和条件索引结合使用<br>数据缺失：notnull() isnull()    expense[expense.isnull()] = expense.mean()</p>
<h5 id="DataFrame-二维数组"><a href="#DataFrame-二维数组" class="headerlink" title="DataFrame 二维数组"></a>DataFrame 二维数组</h5><p>构建：给列名列元素    columns调整列的顺序    index可重定义    可由Series构建<br>取值：下为取列，注意区分    <strong>.loc取行</strong>(传入list或类似切片用法)　<a href="https://blog.csdn.net/hecongqing/article/details/61927615" target="_blank" rel="noopener">.loc用法参考</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">type(df[&apos;apt&apos;])</span><br><span class="line">pandas.core.series.Series</span><br><span class="line">type(df[[&apos;apt&apos;]])</span><br><span class="line">pandas.core.frame.DataFrame</span><br></pre></td></tr></table></figure></p>
<p>df.T 转置    df.shape 、df.info()：行列信息    df.describe():均值分位数等信息<br><a href="https://blog.csdn.net/qq_28219759/article/details/49160073" target="_blank" rel="noopener">.isin参考</a><br>缺失值处理：.fillna</p>
<h5 id="Index-csv文件的读写"><a href="#Index-csv文件的读写" class="headerlink" title="Index csv文件的读写"></a>Index csv文件的读写</h5><p><a href="https://blog.csdn.net/newbiemath/article/details/61432073" target="_blank" rel="noopener">.read_csv参考</a>: index_col parse_dates<br>.to_csv()<br><strong>.index</strong>    index.dayofweek<br>%matplotlib inline 绘图<br>数据选择 | &amp; !</p>
<p><strong><em>以上整理自<a href="www.xiniuedu.com">稀牛学院</a>数据科学实训营 寒小阳老师关于pandas的讲解</em></strong></p>

      
    </main>
    <footer class="post-footer">
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2018/04/02/爬虫：静态网站数据获取/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2018/04/02/爬虫：静态网站数据获取/" itemprop="url">爬虫：静态网站数据获取</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2018-04-02T18:29:09+08:00">2018-04-02</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2018/04/02/爬虫：静态网站数据获取/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2018/04/02/爬虫：静态网站数据获取/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h4 id="XPath"><a href="#XPath" class="headerlink" title="XPath"></a>XPath</h4><h5 id="定义"><a href="#定义" class="headerlink" title="定义"></a>定义</h5><blockquote>
<p>XML 文档中查信息的语言， XML 文档中对元素和属性进行遍历</p>
</blockquote>
<h5 id="七类节点node-XML最小单元"><a href="#七类节点node-XML最小单元" class="headerlink" title="七类节点node XML最小单元"></a>七类节点node XML最小单元</h5><blockquote>
<p>element、attribute、text、namespace、processing-instruction、comment、root</p>
</blockquote>
<h5 id="语法"><a href="#语法" class="headerlink" title="语法"></a>语法</h5><p>选取节点：路径表达式<br><a href="http://www.w3school.com.cn/xpath/index.asp" target="_blank" rel="noopener">XPath教程</a></p>
<blockquote>
<p>nodename 选取此node所有子节点<br>/ 文档根下节点 // 文档里任何位置的节点<br>. 当前节点  .. 当前节点父节点<br>@选属性<br>[] 谓语 找特定节点或包含特定值节点<br>星号 匹配任何元素节点 @* 匹配任何属性节点  node() 匹配任何类型节点<br>| 选取若干路径<br>绝对路径和相对路径：前者以“/”起始</p>
</blockquote>
<h5 id="查看网页XPath路径记录"><a href="#查看网页XPath路径记录" class="headerlink" title="查看网页XPath路径记录"></a>查看网页XPath路径记录</h5><p>右击标题，选检查，elements下查看，右击copy XPath，ctrl+F，ctrl+V，之前粘贴部分被高亮，高亮部分为XPath路径示例，具体语法参照上面<br>//div[@class=”text”]/em/a/text() 取内容<br>div[@class=”text”] 选取所有拥有名为class=”text” 的属性的div 元素<br>a/@href 取属性</p>
<h4 id="requesets"><a href="#requesets" class="headerlink" title="requesets"></a>requesets</h4><p>One.请求网页，<a href="https://blog.csdn.net/shanzhizi/article/details/50903748" target="_blank" rel="noopener">requesets详解</a><br>Two.获取返回状态200和网页编码格式’GB2312’ -&gt; .content显示内容 -&gt; import etree;XPath抓取标题;用text-&gt;传递参数wd和rn -&gt; 超时时间 -&gt;添加代理 -&gt;请求头内容<br><a href="https://www.biaodianfu.com/python-lxml-xpath.html" target="_blank" rel="noopener">etree和XPath</a></p>
<p>豆瓣最热电影评论爬取示例、链家爬取案例练习</p>
<p><strong><em><a href="www.xiniuedu.com">稀牛学院</a>数据科学实训营 助教寒小阳老师爬虫课程 学习整理</em></strong></p>

      
    </main>
    <footer class="post-footer">
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2018/04/01/实战1-QQ答疑/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2018/04/01/实战1-QQ答疑/" itemprop="url">实战1.QQ答疑</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2018-04-01T21:13:58+08:00">2018-04-01</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2018/04/01/实战1-QQ答疑/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2018/04/01/实战1-QQ答疑/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h4 id="实战1-一起来打怪之-python-基础与进阶练习题-QQ答疑"><a href="#实战1-一起来打怪之-python-基础与进阶练习题-QQ答疑" class="headerlink" title="实战1.一起来打怪之 python 基础与进阶练习题 QQ答疑"></a>实战1.一起来打怪之 python 基础与进阶练习题 QQ答疑</h4><h5 id="关于竞赛"><a href="#关于竞赛" class="headerlink" title="关于竞赛"></a>关于竞赛</h5><p>Q：竞赛情况<br>A：竞赛最有名的是Kaggle，国内是阿里天池。获奖情况每个比赛不一样，有的前10，有的前6。一般能到前1%就很不错了。竞赛爬取结果: <a href="https://github.com/iphysresearch/DataSciComp" target="_blank" rel="noopener">土豆的github</a></p>
<h5 id="如何学习"><a href="#如何学习" class="headerlink" title="如何学习"></a>如何学习</h5><p>Q：边听课边在实验室环境里的notebook上做笔记，两个窗口不能同时开着，只能切换运行<br>A：可同时打开notebook同时做笔记。但是建议注意力放在听课上面，用笔和纸速记。可用markdown做笔记。<br>A：问题出现，先google,bing检索，常见社区Stack Overflow</p>
<h5 id="运行平台"><a href="#运行平台" class="headerlink" title="运行平台"></a>运行平台</h5><p>Q：jupyter notebook执行ternimal无法显示<br>A：广告拦截软件拦截，关掉广告拦截软件或者换一个浏览器试试。chrome和firefox都能很好的运行稀牛的实验环境。<br>Q：只要能用anaconda里的jupyter打开发的ipynb的文件就说明配置没错吗<br>A：anaconda应该没有什么需要编辑的，只要装上了，就能启动jupyter<br>Q：jupyter里面有类似上下键翻录历史命令/代码的快捷键？<br>A：shift+enter ctrl+enter<br>Q：在terminal里面输入dir，然后下一条命令还要输入dir，winh和linux下都可以用键盘上下翻的键显示，ipython也可以，notebook里面不知道有没有<br>A：木有<br>Q：将notebook里的cell 转换为markdown格式是什么快捷键<br>A：选中单元格+m<br>Q：本地安装了jupyter ，但是terminal里运行jupyter notebook 每次给出本地地址，需要手动复制到浏览器运行，难道不应该是直接弹出浏览器运行吗？ 怎么设置，是要修改环境变量吗？怎么设置？<br>A：打开失败了，检查一下里面浏览器的设置<br>Q：在jupter里写的markdown可以变成文本吗<br>A：可以点左上角，file——&gt;download as，里面有很多选项<br>Q：shell查看文件<br>A：!head</p>
<h5 id="python基础"><a href="#python基础" class="headerlink" title="python基础"></a>python基础</h5><p>Q：文件读取多语言的文件怎么处理<br>A：建议使用python3，python3对中文支持要好一些，特别爬虫的时候感觉最明显。如果还不行的话建议，在写入文件和读取文件的时候，跟上字符编码。未解决，可能和win10系统相关。<br>Q：input 可以输入字符串列表吗<br>A：input来输入int，raw_input来输入string。可参考：<a href="https://blog.csdn.net/qingquanyingyue/article/details/77964990" target="_blank" rel="noopener">Python输入字符串转列表</a><br>Q：导入了seaborn，为什么画出来的图还是matplotlib默认风格的<br>A：代码用的pylot，sns.set()执行以下，再绘图<br>Q：不定长度参数的函数中 * 号是指针的意思？<br>A：不是，廖学峰的官方网站python3: <a href="https://www.liaoxuefeng.com/wiki/001374738125095c955c1e6d8bb493182103fac9270762a000/001374738449338c8a122a7f2e047899fc162f4a7205ea3000" target="_blank" rel="noopener">参数小结</a></p>
<h5 id="爬虫"><a href="#爬虫" class="headerlink" title="爬虫"></a>爬虫</h5><p>Q：静态数据获取视频中查看元素之后在filter里填了东西是回车就可以找到了吗 xpath表达式<br>A：取东西一般有两种类型。一种是文本，最后需要接一个text()来取文本。另外一种是取属性。可查看：<a href="http://www.w3school.com.cn/xpath/xpath_syntax.asp" target="_blank" rel="noopener">xpath语法</a><br>Q：浏览器打开检查元素，是直接在filter里填xpath就回高亮就会跳到我要找的文本或属性吗？<br>A：是的，只要写对了，就会高亮。直接使用chrome右键copy xpath得到的，可读性不高，可用性不强。<br>Q：怎样的xpath算是可读性比较强的<br>A：简洁、高效。</p>
<p><strong><em>以上来自<a href="www.xiniuedu.com">稀牛学院</a>数据科学实训营 助教易龙老师及学员们的QQ答疑</em></strong></p>

      
    </main>
    <footer class="post-footer">
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2018/04/01/L1.来自google的专业python指南/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2018/04/01/L1.来自google的专业python指南/" itemprop="url">来自google的专业python指南</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2018-04-01T18:01:46+08:00">2018-04-01</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2018/04/01/L1.来自google的专业python指南/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2018/04/01/L1.来自google的专业python指南/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <p>前言：课程结构相对于前四期进行调整。one.爬虫（数据获取）直播切录播。Two.机器学习及其依赖的数据处理课时加重。Three.大数据技能后移，希望教给大家大数据处理和机器学习大数据建模的技能。</p>
<h4 id="python使用原因："><a href="#python使用原因：" class="headerlink" title="python使用原因："></a>python使用原因：</h4><p>数据科学领域top2；<br>深度学习、人工智能领域举例：google: tensorflow、facebook: pytorch + caffe2、Amazon: mnxet、caffe、易上手package：Keras, TFlearn, tensorlayer、python接口；<br>机器学习领域举例：scikit-learn numpy/scipy pandas xgboost/lightGBM；<br>大数据领域举例：spark scala、pyspark、hadoop Map-Reduce、hadoop streaming + python 脚本；<br>数据科学家:数据驱动的解决方案，不花大时间在开发，集中在数据分析建模等问题上</p>
<h4 id="如何寻求帮助"><a href="#如何寻求帮助" class="headerlink" title="如何寻求帮助"></a>如何寻求帮助</h4><p>help() (快捷键win+ctrl+shift+tab)<br>dir()</p>
<h4 id="python运算"><a href="#python运算" class="headerlink" title="python运算"></a>python运算</h4><p>运算符：+ - * / %</p>
<h4 id="基本数据类型、变量"><a href="#基本数据类型、变量" class="headerlink" title="基本数据类型、变量"></a>基本数据类型、变量</h4><p>基本数据类型：int float str bool<br>type() 查看数据类型<br>整数和浮点数在计算机内部存储的方式是不同的，整数运算永远是精确的，而浮点数运算则可能会有四舍五入的误差。</p>
<h4 id="表达式"><a href="#表达式" class="headerlink" title="表达式"></a>表达式</h4><p>+=  -= *= /=</p>
<h4 id="字符串"><a href="#字符串" class="headerlink" title="字符串"></a>字符串</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">x = &quot;hello world!&quot;</span><br><span class="line">abc = &quot;123456&quot; </span><br><span class="line">abc.isdigit()  #Tab可补全</span><br><span class="line">&apos;&apos;&apos; &apos;&apos;&apos; 可内部换行</span><br></pre></td></tr></table></figure>
<p>可参考：<a href="https://blog.csdn.net/woainishifu/article/details/76105667" target="_blank" rel="noopener">三种引号区别</a></p>
<h4 id="字符串切片-splice"><a href="#字符串切片-splice" class="headerlink" title="字符串切片/splice"></a>字符串切片/splice</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">len(tmp_str) </span><br><span class="line">左起0 右起-1 左闭右开</span><br><span class="line">string.lower() </span><br><span class="line">.upper() </span><br><span class="line">.capitalize() </span><br><span class="line">.startswith(&quot;xiao&quot;) </span><br><span class="line">.endswith(&quot;sh&quot;) </span><br><span class="line">.strip() </span><br><span class="line">.find() #-1未找到  </span><br><span class="line">.split()</span><br></pre></td></tr></table></figure>
<h4 id="列表-list（存储一连串数据）"><a href="#列表-list（存储一连串数据）" class="headerlink" title="列表/list（存储一连串数据）"></a>列表/list（存储一连串数据）</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">names=[&apos;Hanmei&apos;,&apos;Lilei&apos;,&apos;Hanxiao&apos;,&apos;Xi&apos;,&apos;Bob&apos;,&apos;Davie&apos;]</span><br><span class="line">len(names)</span><br><span class="line">mixed = [&apos;Han&apos;,2,3.14,[&apos;Li&apos;,&apos;Hanmei&apos;]]</span><br><span class="line">len(mixed)</span><br><span class="line">mixed[1]</span><br><span class="line">mixed[-2]</span><br><span class="line">mixed[-1][-1]</span><br><span class="line">&apos;\n&apos;.join(names)</span><br><span class="line">names.append(&quot;xiaohong&quot;)#尾部添加</span><br><span class="line">names.remove([&quot;xiaohong&quot;])</span><br><span class="line">names.extend([&apos;xiao&apos;,&apos;xiaoxiao&apos;,&apos;Bao&apos;])</span><br><span class="line">names.reverse()</span><br><span class="line">help(list.pop)</span><br><span class="line">help(list.insert)</span><br></pre></td></tr></table></figure>
<h4 id="判断循环"><a href="#判断循环" class="headerlink" title="判断循环"></a>判断循环</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">age = 66</span><br><span class="line">if age&gt;80:</span><br><span class="line">	print(&apos;very old&apos;)</span><br><span class="line">elif age&gt;30:</span><br><span class="line">	print(&apos;old&apos;)</span><br><span class="line">else:</span><br><span class="line">	print(&apos;young&apos;)</span><br><span class="line">	</span><br><span class="line">for index,stu in enumerate(names):</span><br><span class="line">	print(index,stu)</span><br><span class="line">	</span><br><span class="line">i = 0</span><br><span class="line">while i&lt;10:</span><br><span class="line"> print(i)</span><br><span class="line"> i +=1</span><br><span class="line"> </span><br><span class="line">注意continue break 区分</span><br><span class="line">enumerate()函数将一个可遍历的数据对象(如list、元组或字符串)组合为一个索引序列，同时列出数据和数据下标，一般用在for循环当中</span><br></pre></td></tr></table></figure>
<h5 id="列表推导式"><a href="#列表推导式" class="headerlink" title="列表推导式"></a>列表推导式</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">[&quot;my name is: &quot; +name for name in names]</span><br><span class="line">[num + 5 for num in num_list]</span><br><span class="line">[num**3 for num in num_list if (num%2==1 and num &lt; 7)]</span><br><span class="line">#与或非and or not</span><br></pre></td></tr></table></figure>
<h4 id="集合Set"><a href="#集合Set" class="headerlink" title="集合Set"></a>集合Set</h4><p>y = set([‘h’,’a’,’m’])<br>与列表和元组不同，集合是无序的，也无法通过数字进行索引。此外，集合中的元素不能重复</p>
<h4 id="字典"><a href="#字典" class="headerlink" title="字典"></a>字典</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">legs = &#123;&apos;spider&apos;:8,&apos;pig&apos;:4,&apos;ducl&apos;:2&#125;</span><br><span class="line">type(legs)</span><br><span class="line">legs.values()</span><br><span class="line">legs.keys()</span><br><span class="line">&apos;bird&apos; in legs</span><br><span class="line">for animal, leg_num in legs.items():</span><br><span class="line">	print(animal,leg_num)</span><br></pre></td></tr></table></figure>
<h5 id="字典推导式"><a href="#字典推导式" class="headerlink" title="字典推导式"></a>字典推导式</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">my_list = [1,3,5,7,9,2,4]</span><br><span class="line">dic = &#123;&#125;</span><br><span class="line">for num in my_list:</span><br><span class="line">	dic[num] = num **3</span><br><span class="line">	</span><br><span class="line">&#123;num:num**3 for num in my_list&#125;</span><br></pre></td></tr></table></figure>
<h5 id="高级排序"><a href="#高级排序" class="headerlink" title="高级排序"></a>高级排序</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">num_list1.sort()#成员函数，对自身做调整，num_list1排序结果为返回值返回</span><br><span class="line">sorted(num_list2) #python中函数，num_list2未更改</span><br><span class="line">sorted(str, reverse=True)#字母反序</span><br><span class="line">sorted(str,key=len)#根据key函数即长度排序，长度一致asic排序</span><br><span class="line">sorted(str,key=str.lower)#小写后排序，key function给函数名，无需括号</span><br></pre></td></tr></table></figure>
<h4 id="函数"><a href="#函数" class="headerlink" title="函数"></a>函数</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">def get_first(my_list):</span><br><span class="line">	return my_list[0]</span><br><span class="line">get_first([&apos;Han&apos;,&apos;Li&apos;])</span><br><span class="line">classed = [&apos;Han&apos;,&apos;Li&apos;],[&apos;Xiao&apos;,&apos;Ming&apos;],[&apos;Wang&apos;,&apos;Xiao&apos;]</span><br><span class="line">sorted(classed,key=get_first)</span><br><span class="line">#不定长度的参数</span><br><span class="line">def print_all(*args):</span><br><span class="line">	print(type(args))</span><br><span class="line">	print(args)</span><br></pre></td></tr></table></figure>
<h4 id="文件读写"><a href="#文件读写" class="headerlink" title="文件读写"></a>文件读写</h4><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">head -2 test.txt #shell</span><br><span class="line">f = open(&quot;test.txt&quot;, &apos;r&apos;)</span><br><span class="line">contents = f.readlines()</span><br><span class="line">f.close()</span><br><span class="line"></span><br><span class="line">for line in open(&apos;test.txt&apos;,&apos;r&apos;,encoding=&apos;UTF-8&apos;):</span><br><span class="line">	print(line.strip())</span><br><span class="line">	print(&quot;\n&quot;)</span><br></pre></td></tr></table></figure>
<h5 id="统计文件中的词频"><a href="#统计文件中的词频" class="headerlink" title="统计文件中的词频"></a>统计文件中的词频</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">def my_word_count(in_file,out_file):</span><br><span class="line">	word_count = &#123;&#125;</span><br><span class="line">	for line in open(in_file,&apos;r&apos;,encoding=&apos;utf-8&apos;):</span><br><span class="line">		words = line.strip().split(&quot; &quot;)</span><br><span class="line">		for word in words:</span><br><span class="line">			if word.lower() in word_count:</span><br><span class="line">				word_count[word.lower()] += 1 </span><br><span class="line">			else:</span><br><span class="line">				word_count[word.lower()] = 1</span><br><span class="line">	out = open(out_file,&apos;w&apos;,encoding=&apos;utf-8&apos;)</span><br><span class="line">	for word in word_count:</span><br><span class="line">		out.write(word+&quot;:&quot;+str(word_count[word])+&quot;\n&quot;)</span><br><span class="line">	print(&quot;词频统计完成&quot;)</span><br><span class="line">	out.close()</span><br><span class="line">	</span><br><span class="line">in_file = &apos;test.txt&apos;</span><br><span class="line">out_file = &apos;out.txt&apos;</span><br><span class="line">my_word_count(in_file,out_file)</span><br><span class="line"># += 间无空格 严格缩进</span><br></pre></td></tr></table></figure>
<h4 id="正则表达式"><a href="#正则表达式" class="headerlink" title="正则表达式"></a>正则表达式</h4><blockquote>
<p>\d匹配一个数字<br>\w匹配一个字母或数字<br>\s匹配一个空格（也包括Tab等空白符）<br>.匹配任意字符<br><em>表示任意个字符（包括0个）<br>+表示至少一个字符<br>?表示0个或1个字符<br>{n}表示n个字符<br>{n,m}表示n-m个字符<br>[0-9a-zA-Z_]匹配一个数字、字母或者下划线<br>[0-9a-zA-Z_]+可以匹配至少由一个数字、字母或者下划线组成的字符串，比如’a100’，’0<em>Z’，’Py3000’等等；<br>[a-zA-Z\</em>][0-9a-zA-Z_]</em>可以匹配由字母或下划线开头，后接任意个由一个数字、字母或者下划线组成的字符串，也就是Python合法的变量；<br>[a-zA-Z_][0-9a-zA-Z_]{0, 19}更精确地限制了变量的长度是1-20个字符（前面1个字符+后面最多19个字符）。<br>A|B可以匹配A或B，所以(P|p)ython可以匹配’Python’或者’python’。<br>^表示行的开头，^\d表示必须以数字开头。<br>\d$表示必须以数字结束</p>
</blockquote>
<p>参考资料: <a href="https://www.liaoxuefeng.com/wiki/0014316089557264a6b348958f449949df42a6d3a2e542c000/00143193331387014ccd1040c814dee8b2164bb4f064cff000" target="_blank" rel="noopener">廖雪峰的Python教程</a></p>
<h5 id="有意思的拓展链接整理中"><a href="#有意思的拓展链接整理中" class="headerlink" title="有意思的拓展链接整理中"></a>有意思的拓展链接整理中</h5><p><a href="https://mp.weixin.qq.com/s/7Ephzy4NuwuHJwu8M5ur4Q" target="_blank" rel="noopener">八道python练习题</a><br><a href="http://aosabook.org/blog/" target="_blank" rel="noopener">http://aosabook.org/blog/</a></p>
<p><strong><em>以上来自<a href="www.xiniuedu.com">稀牛学院</a>数据科学实训营 寒小阳老师关于python基础的讲解<br>由于课程版权问题，后续会减去课程具体内容，更换为个人学习体验和拓展学习内容</em></strong></p>

      
    </main>
    <footer class="post-footer">
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/23/转录组入门——8/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/23/转录组入门——8/" itemprop="url">转录组入门——8</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-23T18:02:51+08:00">2017-08-23</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/23/转录组入门——8/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/23/转录组入门——8/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <blockquote>
<p>选择p&lt;0.05而且abs(logFC)大于一个与众的基因为显著差异表达基因集，对这个基因集用R包做KEGG/GO超几何分布检验分析。<br>然后把表达矩阵和分组信息分别作出cls和gct文件，导入到GSEA软件分析。<br>基本任务是完成这个分析。</p>
</blockquote>
<h5 id="GO-KEGG分析-clusterProfiler"><a href="#GO-KEGG分析-clusterProfiler" class="headerlink" title="GO/KEGG分析 clusterProfiler"></a>GO/KEGG分析 clusterProfiler</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">biocLite(&quot;DOSE&quot;)</span><br><span class="line">require(DOSE)</span><br><span class="line">require(clusterProfiler)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line"></span><br><span class="line">##去除ID的版本号</span><br><span class="line">diff_gene_ENSEMBL &lt;- unlist(lapply(diff_gene, function(x)&#123;strsplit(x, &quot;\\.&quot;)[[1]][1]&#125;))</span><br><span class="line">##GOid mapping + GO富集</span><br><span class="line">## R版本 在服务器上R 无法使用</span><br><span class="line">ego &lt;- enrichGO(gene = diff_gene_ENSEMBL, OrgDb = org.Mm.eg.db,</span><br><span class="line">                                keytype = &quot;ENSEMBL&quot;, ont = &quot;BP&quot;, pAdjustMethod = &quot;BH&quot;,</span><br><span class="line">                                pvalueCutoff = 0.01, qvalueCutoff = 0.05)</span><br><span class="line">##查看富集结果数据</span><br><span class="line">enrich_go &lt;- as.data.frame(ego)</span><br><span class="line">##作图</span><br><span class="line">barplot(ego, showCategory=10)</span><br><span class="line">dotplot(ego)</span><br><span class="line">enrichMap(ego)</span><br><span class="line">plotGOgraph(ego)</span><br><span class="line"></span><br><span class="line">##ID转化</span><br><span class="line">ids &lt;- bitr(diff_gene_ENSEMBL, fromType = &quot;ENSEMBL&quot;, toType = &quot;ENTREZID&quot;, OrgDb = &quot;org.Mm.eg.db&quot;)</span><br><span class="line">kk &lt;- enrichKEGG(gene = ids[,2], organism = &quot;mmu&quot;, keyType = &quot;kegg&quot;,pvalueCutoff = 0.05, pAdjustMethod = &quot;BH&quot;, qvalueCutoff = 0.1)</span><br><span class="line">##查看富集结果数据</span><br><span class="line">enrich_kegg &lt;- as.data.frame(kk)</span><br><span class="line">##作图</span><br><span class="line">dotplot(kk)</span><br></pre></td></tr></table></figure>
<h5 id="GSEA"><a href="#GSEA" class="headerlink" title="GSEA"></a>GSEA</h5><blockquote>
<p>基因集富集分析 (Gene Set Enrichment Analysis, GSEA) 的基本思想是使用预定义的基因集（通常来自功能注释或先前实验的结果），将基因按照在两类样本中的差异表达程度排序，然后检验预先设定的基因集合是否在这个排序表的顶端或者底端富集。基因集合富集分析检测基因集合而不是单个基因的表达变化，因此可以包含这些细微的表达变化，预期得到更为理想的结果。</p>
</blockquote>
<h5 id="Gene-Set-Enrichment-Analysis（GSEA）"><a href="#Gene-Set-Enrichment-Analysis（GSEA）" class="headerlink" title="Gene Set Enrichment Analysis（GSEA）"></a>Gene Set Enrichment Analysis（GSEA）</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">diff_gene_DESeq2 &lt;- subset(res, padj &lt; 0.05 &amp; (log2FoldChange &gt; 1 | log2FoldChange &lt; -1))</span><br><span class="line">diff_gene_ENSEMBL &lt;- unlist(lapply(rownames(diff_gene_DESeq2), function(x)&#123;strsplit(x, &quot;\\.&quot;)[[1]][1]&#125;))</span><br><span class="line">rownames(diff_gene_DESeq2) &lt;- diff_gene_ENSEMBL</span><br><span class="line">diff_gene_ENTREZID &lt;- select(org.Mm.eg.db,keys = diff_gene_ENSEMBL,columns =&quot;ENTREZID&quot;,keytype = &quot;ENSEMBL&quot;)</span><br><span class="line">diff_gene_DESeq2$ENTREZID &lt;- diff_gene_ENTREZID[,2]</span><br><span class="line"># 获取按照log2FC大小来排序的基因列表</span><br><span class="line">genelist &lt;- diff_gene_DESeq2$log2FoldChange</span><br><span class="line">names(genelist) &lt;- diff_gene_DESeq2$ENTREZID</span><br><span class="line">genelist &lt;- sort(genelist, decreasing = TRUE)</span><br><span class="line"># GSEA分析（具体参数参考：https://mp.weixin.qq.com/s/p-n5jq5Rx2TqDBStS2nzoQ）</span><br><span class="line">gsemf &lt;- gseGO(genelist,OrgDb = org.Mm.eg.db,ont=&quot;MF&quot;)</span><br><span class="line"># 查看大致信息</span><br><span class="line">head(gsemf)</span><br><span class="line"># 画出GSEA图</span><br><span class="line">gseaplot(gsemf, geneSetID=&quot;GO:0000977&quot;)</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-d045f82c5453691f.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="go"></p>
<blockquote>
<p><a href="http://www.biotrainee.com/thread-2018-1-1.html" target="_blank" rel="noopener">转录组入门（7-8）从差异基因到富集分析</a><br><a href="https://mp.weixin.qq.com/s/p-n5jq5Rx2TqDBStS2nzoQ" target="_blank" rel="noopener">从零开始学转录组（8）：富集分析</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/21/转录组入门——7/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/21/转录组入门——7/" itemprop="url">转录组入门——7</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-21T16:02:38+08:00">2017-08-21</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/21/转录组入门——7/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/21/转录组入门——7/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">.libPaths(&quot;/home/tingm/R/x86_64-pc-linux-gnu-library/3.2&quot;)</span><br><span class="line">  library(limma)</span><br><span class="line">  library(edgeR)</span><br><span class="line">  library(DESeq2)</span><br><span class="line">    #change</span><br><span class="line">    path&lt;-&quot;/home/tingm/fastq/mus_result_count/&quot;</span><br><span class="line">    #change</span><br><span class="line">      samples&lt;-read.table(&quot;samples.list.txt&quot;,header=F,sep=&quot;\t&quot;)</span><br><span class="line">      samples&lt;-samples[,1]</span><br><span class="line">      #change</span><br><span class="line">      data.raw&lt;-read.table(&quot;Mus_SRR3589959.sgeneCounts&quot;,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;Mus_SRR3589959.sgeneCounts&quot;))</span><br><span class="line">      for(sample in samples)</span><br><span class="line">      &#123;</span><br><span class="line">        print(sample)</span><br><span class="line">        file&lt;-paste(path,sample,sep=&quot;&quot;)</span><br><span class="line">        data.new&lt;-read.table(file,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,sample))</span><br><span class="line">        data.raw&lt;-merge(data.new,data.raw,by=&quot;gene_id&quot;)</span><br><span class="line">      &#125;</span><br><span class="line">     data.raw&lt;-data.raw[-1:-5,]</span><br><span class="line">        #change</span><br><span class="line">     names(data.raw) &lt;- c(&quot;gene_id&quot;,&quot;Akap95_2&quot;,&quot;control_2&quot;,&quot;Akap95_1&quot;,&quot;control_1&quot;)</span><br><span class="line">     write.table(data.raw,&quot;count.matix.csv&quot;,row.names=F,quote=F)</span><br></pre></td></tr></table></figure>
<h5 id="使用DESeq2进行差异基因分析-筛选出基因数目不一样"><a href="#使用DESeq2进行差异基因分析-筛选出基因数目不一样" class="headerlink" title="使用DESeq2进行差异基因分析 筛选出基因数目不一样"></a>使用DESeq2进行差异基因分析 筛选出基因数目不一样</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">##数据预处理</span><br><span class="line">database &lt;- read.table(file = &quot;count.matix.csv&quot;, sep = &quot; &quot;, header = T,row.names = 1)</span><br><span class="line">database &lt;- round(as.matrix(database))</span><br><span class="line">##设置分组信息并构建dds对象</span><br><span class="line">condition &lt;- factor(c(&quot;Akap95&quot;,&quot;control&quot;,&quot;Akap95&quot;,&quot;control&quot;), levels = c(&quot;control&quot;, &quot;Akap95&quot;))</span><br><span class="line">coldata &lt;- data.frame(row.names = colnames(database), condition)</span><br><span class="line">dds &lt;- DESeqDataSetFromMatrix(countData=database, colData=coldata, design=~condition)</span><br><span class="line">dds &lt;- dds[ rowSums(counts(dds)) &gt; 1, ]</span><br><span class="line">##使用DESeq函数估计离散度，然后差异分析获得res对象</span><br><span class="line">dds &lt;- DESeq(dds)</span><br><span class="line">res &lt;- results(dds)</span><br><span class="line">#summary(res)</span><br><span class="line">#最后设定阈值，筛选差异基因，导出数据(全部数据。包括标准化后的count数)</span><br><span class="line">res &lt;- res[order(res$padj),]</span><br><span class="line">diff_gene_DESeq2 &lt;- subset(res, padj &lt; 0.05 &amp; (log2FoldChange &gt; 1 | log2FoldChange &lt; -1))</span><br><span class="line">diff_gene_DESeq2 &lt;- row.names(diff_gene_DESeq2)</span><br><span class="line">resdata &lt;- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)),by=&quot;row.names&quot;,sort=FALSE)</span><br><span class="line">write.csv(resdata,file = &quot;control_vs_Akap95.csv&quot;,row.names = F)</span><br></pre></td></tr></table></figure>
<h5 id="使用edgeR进行差异基因分析"><a href="#使用edgeR进行差异基因分析" class="headerlink" title="使用edgeR进行差异基因分析"></a>使用edgeR进行差异基因分析</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">##跟DESeq2一样，导入数据，预处理（用了cpm函数）</span><br><span class="line">exprSet&lt;- read.table(file = &quot;count.matix.csv&quot;, sep = &quot; &quot;, header = TRUE, row.names = 1, stringsAsFactors = FALSE)</span><br><span class="line">group_list &lt;- factor(c(&quot;Akap95&quot;,&quot;control&quot;,&quot;Akap95&quot;,&quot;control&quot;), levels = c(&quot;control&quot;, &quot;Akap95&quot;))</span><br><span class="line">exprSet &lt;- exprSet[rowSums(cpm(exprSet) &gt; 1) &gt;= 2,]</span><br><span class="line">##设置分组信息，并做TMM标准化</span><br><span class="line">exprSet &lt;- DGEList(counts = exprSet, group = group_list)</span><br><span class="line">exprSet &lt;- calcNormFactors(exprSet)</span><br><span class="line">##使用qCML（quantile-adjusted conditional maximum likelihood）估计离散度（只针对单因素实验设计）</span><br><span class="line">exprSet &lt;- estimateCommonDisp(exprSet)</span><br><span class="line">exprSet &lt;- estimateTagwiseDisp(exprSet)</span><br><span class="line">##寻找差异gene(这里的exactTest函数还是基于qCML并且只针对单因素实验设计)，然后按照阈值进行筛选即可</span><br><span class="line">et &lt;- exactTest(exprSet)</span><br><span class="line">tTag &lt;- topTags(et, n=nrow(exprSet))</span><br><span class="line">diff_gene_edgeR &lt;- subset(tTag$table, FDR &lt; 0.05 &amp; (logFC &gt; 1 | logFC &lt; -1))</span><br><span class="line">diff_gene_edgeR &lt;- row.names(diff_gene_edgeR)</span><br><span class="line">write.csv(tTag$table,file = &quot;control_vs_Akap95_edgeR.csv&quot;)</span><br></pre></td></tr></table></figure>
<h5 id="使用limma进行差异分析-limma的voom流程"><a href="#使用limma进行差异分析-limma的voom流程" class="headerlink" title="使用limma进行差异分析   limma的voom流程"></a>使用limma进行差异分析   limma的voom流程</h5><p>待续</p>
<h5 id="不同软件包分析结果比较"><a href="#不同软件包分析结果比较" class="headerlink" title="不同软件包分析结果比较"></a>不同软件包分析结果比较</h5><p>待续</p>
<blockquote>
<p><a href="http://www.jianshu.com/p/3bfb21d24b74" target="_blank" rel="noopener">转录组入门（7）：差异基因分析</a><br><a href="https://mp.weixin.qq.com/s/1mda7NpOxShbOxZKSJar7Q" target="_blank" rel="noopener">从零开始学转录组（7）：差异基因表达分析</a><br><a href="http://www.biotrainee.com/thread-2018-1-1.html" target="_blank" rel="noopener">转录组入门（7-8）从差异基因到富集分析</a><br><a href="http://www.agctu.org/question/179" target="_blank" rel="noopener">为什么RNA-seq的read数服从泊松分布？</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/15/转录组入门——6/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/15/转录组入门——6/" itemprop="url">转录组入门——6</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-15T20:37:18+08:00">2017-08-15</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/15/转录组入门——6/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/15/转录组入门——6/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="实现reads计数的软件教程"><a href="#实现reads计数的软件教程" class="headerlink" title="实现reads计数的软件教程"></a>实现reads计数的软件教程</h5><blockquote>
<p>map后获得了每条read比对到基因组上的位置，后面需要把基因组的位置对应到基因。这个提供基因位置的注释文件通常以GTF或GFF3格式呈现。基因表达量最直接的分析手段就是计算比对到每个基因的reads有多少条，在转录组测序中，我们通常称这个数字为count<br>对每个基因计数之后得到的count matrix再后续的分析中，要注意标准化的问题。<br>比较同一个样本(within-sample)不同基因之间的表达情况，需要考虑到转录本长度，因为转录本越长，那么检测的片段也会更多。比较不同样本（across sample）同一个基因的表达情况，不必在意转录本长度，但要考虑到测序深度（sequence depth)，测序深度越高，检测到的概率越大。除这两因素外，还需考虑GC%导致的偏差及测序仪器的系统偏差。<br>目前对read count标准化的算法有RPKM（SE）, FPKM（PE），TPM, TMM等。但是，有一些下游分析的软件会要求是输入的count matrix是原始数据，未经标准化，比如说DESeq2，这个时候你需要注意你上一步所用软件会不会进行标准化。<br>计数分为三个水平： gene-level, transcript-level, exon-usage-level<br>标准化方法： FPKM RPKM TMM TPM</p>
</blockquote>
<h5 id="Gene-level"><a href="#Gene-level" class="headerlink" title="Gene-level"></a>Gene-level</h5><blockquote>
<p>HTSeq <a href="https://htseq.readthedocs.io/en/release_0.9.1/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.chenlianfu.com/?p=2438" target="_blank" rel="noopener">中文教程</a><br>featureCounts <a href="http://bioinf.wehi.edu.au/featureCounts/" target="_blank" rel="noopener">官方网址</a> Rsubread <a href="https://bioconductor.org/packages/release/bioc/html/Rsubread.html" target="_blank" rel="noopener">官方网址</a><br>bedtools multicov <a href="https://bedtools.readthedocs.io/en/latest/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.biotrainee.com/thread-153-1-1.html" target="_blank" rel="noopener">中文教程</a><br>Qualimap <a href="http://qualimap.bioinfo.cipf.es/" target="_blank" rel="noopener">官方网址</a> <a href="http://blog.sciencenet.cn/blog-2970729-1070459.html" target="_blank" rel="noopener">中文教程</a><br>GenomicRanges <a href="https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html" target="_blank" rel="noopener">官方网址</a></p>
</blockquote>
<h5 id="Transcript-level"><a href="#Transcript-level" class="headerlink" title="Transcript-level"></a>Transcript-level</h5><blockquote>
<p>cufflink <a href="https://cole-trapnell-lab.github.io/cufflinks/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.chenlianfu.com/?p=623" target="_blank" rel="noopener">中文教程</a><br>stringtie <a href="https://ccb.jhu.edu/software/stringtie/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.bio-info-trainee.com/2073.html" target="_blank" rel="noopener">中文教程</a><br>上述软件都是alignment-based<br>alignment-free软件，如kallisto, silfish, salmon，能够省去比对这一步，直接得到read count，在运行效率上更高。不过最近一篇文献Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis指出这类方法在估计丰度时存在样本特异性和读长偏差<br>salmon</p>
</blockquote>
<h5 id="Exon-usage-level"><a href="#Exon-usage-level" class="headerlink" title="Exon-usage-level"></a>Exon-usage-level</h5><blockquote>
<p>和基因水平的统计类似。但是值得注意的是为了更好的计数，我们需要提供无重叠的外显子区域的gtf文件Detecting differential usage of exons from RNA-seq data。用于分析差异外显子使用的DEXSeq提供了一个Python脚本（dexseq_prepare_annotation.py）执行这个任务。</p>
</blockquote>
<h5 id="基因表达量归一算法"><a href="#基因表达量归一算法" class="headerlink" title="基因表达量归一算法"></a>基因表达量归一算法</h5><p>待续 <a href="https://zhuanlan.zhihu.com/p/26750663" target="_blank" rel="noopener">参考</a></p>
<h5 id="htseq-count"><a href="#htseq-count" class="headerlink" title="htseq-count"></a>htseq-count</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">ls Homo*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $hg19_gff &gt; $&#123;id%_*&#125;.geneCounts 2&gt; $&#123;id%_*&#125;.HTseq.log);done</span><br><span class="line">ls Mus*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $mm10_gff &gt; $&#123;id%_*&#125;.geneCounts 2&gt; $&#123;id%_*&#125;.HTseq.log);done</span><br><span class="line">wc -l Homo*.geneCounts </span><br><span class="line">head -n 4 Homo*.geneCounts</span><br></pre></td></tr></table></figure>
<blockquote>
<p>-f bam/sam： 指定输入文件格式，默认SAM<br>-r name/pos: 你需要利用samtool sort对数据根据read name或者位置进行排序，默认是name<br>-s yes/no/reverse: 数据是否来自于strand-specific assay。DNA是双链的，所以需要判断到底来自于哪条链。如果选择了no， 那么每一条read都会跟正义链和反义链进行比较。默认的yes对于双端测序表示第一个read都在同一个链上，第二个read则在另一条链上。<br>-a 最低质量， 剔除低于阈值的read<br>-m 模式 union（默认）, intersection-strict and intersection-nonempty。一般而言就用默认的，作者也是这样认为的。<br>-i id attribute: 在GTF文件的最后一栏里，会有这个基因的多个命名方式（如下）， RNA-Seq数据分析常用的是gene_id， 当然你可以写一个脚本替换成其他命名方式。</p>
</blockquote>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-6aaf8956467c4fb6.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="homo.genecounts"></p>
<h5 id="编写脚本合并所有的样本为表达矩阵"><a href="#编写脚本合并所有的样本为表达矩阵" class="headerlink" title="编写脚本合并所有的样本为表达矩阵"></a>编写脚本合并所有的样本为表达矩阵</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">sed -i &apos;$d&apos; *geneCounts</span><br><span class="line">perl -lne &apos;if($ARGV=~/Homo_(.*).geneCounts/)&#123;print &quot;$1\t$_&quot;&#125;&apos; *|grep -v Homo_sapiens&gt;hg.count</span><br><span class="line">perl -lne &apos;if($ARGV=~/Mus_(.*).geneCounts/)&#123;print &quot;$1\t$_&quot;&#125;&apos; *|grep -v Mus &gt;mus.count</span><br><span class="line"></span><br><span class="line">hg &lt;- read.csv(file = &quot;hg.count&quot;,header = F,sep = &quot;\t&quot;)</span><br><span class="line">colnames(hg) &lt;- c(&apos;sample&apos;,&apos;gene&apos;,&apos;count&apos;)</span><br><span class="line">library(reshape2)</span><br><span class="line">hg_reads &lt;- dcast(hg,formula = gene ~ sample)</span><br><span class="line">write.table(hg_reads,file = &quot;hg_join.count&quot;,sep = &quot;\t&quot;,quote = FALSE,row.names = FALSE)</span><br><span class="line"></span><br><span class="line">mus &lt;- read.csv(file = &quot;mus.count&quot;,header = F,sep = &quot;\t&quot;)</span><br><span class="line">colnames(mus) &lt;- c(&apos;sample&apos;,&apos;gene&apos;,&apos;count&apos;)</span><br><span class="line">library(reshape2)</span><br><span class="line">mus_reads &lt;- dcast(mus,formula = gene ~ sample)</span><br><span class="line">write.table(mus_reads,file = &quot;mus_join.count&quot;,sep = &quot;\t&quot;,quote = FALSE,row.names = FALSE)</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-bf2396936c8bb516.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="homo_table"></p>
<h5 id="对表达矩阵可进行简单的探索"><a href="#对表达矩阵可进行简单的探索" class="headerlink" title="对表达矩阵可进行简单的探索"></a>对表达矩阵可进行简单的探索</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">summary(hg_reads) </span><br><span class="line">summary(mus_reads) </span><br><span class="line">row.names(hg_reads) &lt;- hg_reads$gene</span><br><span class="line">GAPDH &lt;- hg_reads[rownames(hg_reads)==&quot;GAPDH&quot;,]</span><br><span class="line"></span><br><span class="line">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">biocLite(&quot;dplyr&quot;)</span><br><span class="line">library(dplyr)</span><br><span class="line">sample_mean &lt;- group_by(hg,sample)%&gt;% summarize(mean=mean(count))</span><br><span class="line">gene_mean &lt;- group_by(hg,gene)%&gt;% summarize(mean=mean(count)) </span><br><span class="line">tmp &lt;- full_join(hg,gene_mean,by=&apos;gene&apos;)</span><br><span class="line">write.table(tmp,file = &quot;tmp.count&quot;,sep = &quot;\t&quot;,quote = FALSE,row.names = FALSE)</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-3a0f908f6e1b3a5d.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="summary"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-79ab4164d01877a0.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="image.png"></p>
<h5 id="另一种"><a href="#另一种" class="headerlink" title="另一种"></a>另一种</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line">options(stringsAsFactors = FALSE)</span><br><span class="line">#导入数据</span><br><span class="line">homo_control &lt;- read.table(&quot;Homo_SRR3589956.geneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;homo_control&quot;))</span><br><span class="line">homo_rep1 &lt;- read.table(&quot;Homo_SRR3589957.geneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;homo_rep1&quot;))</span><br><span class="line">homo_rep2 &lt;- read.table(&quot;Homo_SRR3589958.geneCounts&quot;,sep=&quot;\t&quot;,col.names = c(&quot;gene_symbol&quot;,&quot;rep2&quot;))</span><br><span class="line">mus_control1 &lt;- read.table(&quot;Mus_SRR3589959.geneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;mus_control1&quot;))</span><br><span class="line">mus_control2 &lt;- read.table(&quot;Mus_SRR3589960.geneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;mus_control2&quot;)) </span><br><span class="line">mus_rep1 &lt;- read.table(&quot;Mus_SRR3589961.geneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;mus_rep1&quot;)) </span><br><span class="line">mus_rep2 &lt;- read.table(&quot;Mus_SRR3589962.geneCounts&quot;, sep=&quot;\t&quot;,col.names = c(&quot;gene_symbol&quot;,&quot;mus_rep2&quot;))</span><br><span class="line">#去除后五行</span><br><span class="line">homo_control &lt;- homo_control[-58523:-58527,]</span><br><span class="line">homo_rep1 &lt;- homo_rep1[-58523:-58527,]</span><br><span class="line">homo_rep2 &lt;- homo_rep2[-58523:-58527,]</span><br><span class="line">mus_control1 &lt;- mus_control1[-48322:-48326,]</span><br><span class="line">mus_control2 &lt;- mus_control2[-48322:-48326,]</span><br><span class="line">mus_rep1 &lt;- mus_rep1[-48322:-48326,]</span><br><span class="line">mus_rep2 &lt;- mus_rep2[-48322:-48326,]</span><br><span class="line">#整合</span><br><span class="line">homo_raw_count &lt;- merge(merge(homo_control, homo_rep1, by=&quot;gene_symbol&quot;), homo_rep2, by=&quot;gene_symbol&quot;)</span><br><span class="line">mus_raw_count &lt;- merge(merge(mus_control1, mus_control2, by=&quot;gene_symbol&quot;), merge(mus_rep1,mus_rep2, by=&quot;gene_symbol&quot;))</span><br><span class="line">summary(homo_raw_count)</span><br><span class="line">summary(mus_raw_count)</span><br><span class="line">row.names(homo_raw_count) &lt;- homo_raw_count$gene_symbol</span><br><span class="line">row.names(mus_raw_count) &lt;- mus_raw_count$gene_symbol</span><br><span class="line">gapdh &lt;- homo_raw_count[rownames(homo_raw_count)==&quot;GAPDH&quot;,]</span><br><span class="line">gapdh &lt;- mus_raw_count[rownames(mus_raw_count)==&quot;GAPDH&quot;,]</span><br><span class="line"></span><br><span class="line">#因为我们无法在EBI数据库上直接搜索找到ENSMUSG00000024045.5这样的基因，只能是ENSMUSG00000024045的整数，没有小数点，所以需要进一步替换为整数的形式</span><br><span class="line">#第一步将匹配到的.以及后面的数字连续匹配并替换为空，并赋值给ENSEMBL</span><br><span class="line">#ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, raw_count_filt1$gene_id) </span><br><span class="line">#将ENSEMBL重新添加到raw_count_filt1矩阵</span><br><span class="line">#row.names(raw_count_filter) &lt;- ENSEMBL</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-8ee1f5c36ac2faa3.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="gapdh"><br>以上两种都有两个问题：AKAP95未找到 小鼠第一列不是gene name<br>和sort是否为reads name无关，htseq count -i 参数去掉则默认为gene id</p>
<h5 id="看生物学意义特殊的基因表现如何如GAPDH-β-ACTIN"><a href="#看生物学意义特殊的基因表现如何如GAPDH-β-ACTIN" class="headerlink" title="看生物学意义特殊的基因表现如何如GAPDH,β-ACTIN"></a>看生物学意义特殊的基因表现如何如GAPDH,β-ACTIN</h5><p><img src="http://upload-images.jianshu.io/upload_images/2116382-e29d08dec77d3ec7.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="gapdh"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">#! usr/bin/bash</span><br><span class="line">set -u</span><br><span class="line">set -e</span><br><span class="line">set -o pipefail</span><br><span class="line">hg19_ref=~/fastq/hisat2_index/hg19/genome</span><br><span class="line">mm10_ref=~/fastq/hisat2_index/mm10/genome</span><br><span class="line">hg19_gff=~/fastq/refdata/Genecode/gencode.v26lift37.annotation.gff3</span><br><span class="line">mm10_gff=~/fastq/refdata/Genecode/gencode.vM10.annotation.gtf</span><br><span class="line">ls Homo*.sorted.bam|while read id;do(htseq-count -f bam -r pos -s no $id $hg19_gff &gt; $&#123;id%%.*&#125;.sgeneCounts 2&gt; $&#123;id%%.*&#125;.sHTseq.log);done</span><br><span class="line">ls Mus*.sorted.bam|while read id;do(htseq-count -f bam -r pos -s no $id $mm10_gff &gt; $&#123;id%%.*&#125;.sgeneCounts 2&gt; $&#123;id%%.*&#125;.sHTseq.log);done</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">options(stringsAsFactors = FALSE)</span><br><span class="line">homo_control &lt;- read.table(&quot;Homo_SRR3589956.sgeneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;homo_control&quot;))</span><br><span class="line">homo_rep1 &lt;- read.table(&quot;Homo_SRR3589957.sgeneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;homo_rep1&quot;))</span><br><span class="line">homo_rep2 &lt;- read.table(&quot;Homo_SRR3589958.sgeneCounts&quot;,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;rep2&quot;))</span><br><span class="line">mus_control1 &lt;- read.table(&quot;Mus_SRR3589959.sgeneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;mus_control1&quot;))</span><br><span class="line">mus_control2 &lt;- read.table(&quot;Mus_SRR3589960.sgeneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;mus_control2&quot;)) </span><br><span class="line">mus_rep1 &lt;- read.table(&quot;Mus_SRR3589961.sgeneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;mus_rep1&quot;)) </span><br><span class="line">mus_rep2 &lt;- read.table(&quot;Mus_SRR3589962.sgeneCounts&quot;, sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;mus_rep2&quot;))</span><br><span class="line">homo_control &lt;- homo_control[-60472:-60476,]</span><br><span class="line">homo_rep1 &lt;- homo_rep1[-60472:-60476,]</span><br><span class="line">homo_rep2 &lt;- homo_rep2[-60472:-60476,]</span><br><span class="line">mus_control1 &lt;- mus_control1[-48441:-48445,]</span><br><span class="line">mus_control2 &lt;- mus_control2[-48441:-48445,]</span><br><span class="line">mus_rep1 &lt;- mus_rep1[-48441:-48445,]</span><br><span class="line">mus_rep2 &lt;- mus_rep2[-48441:-48445,]</span><br><span class="line">homo_raw_count &lt;- merge(merge(homo_control, homo_rep1, by=&quot;gene_id&quot;), homo_rep2, by=&quot;gene_id&quot;)</span><br><span class="line">mus_raw_count &lt;- merge(merge(mus_control1, mus_control2, by=&quot;gene_id&quot;), merge(mus_rep1,mus_rep2, by=&quot;gene_id&quot;))</span><br><span class="line">ENSEMBL &lt;- gsub(&quot;\\.\\d*_\\d*&quot;, &quot;&quot;, homo_raw_count$gene_id)</span><br><span class="line">row.names(homo_raw_count) &lt;- ENSEMBL # error</span><br><span class="line">ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, mus_raw_count$gene_id) </span><br><span class="line">row.names(mus_raw_count) &lt;- ENSEMBL</span><br><span class="line">summary(homo_raw_count)</span><br><span class="line">summary(mus_raw_count)</span><br><span class="line">#在UniProt数据库找到AKAP95的id</span><br><span class="line">MUS_AKAP95 &lt;- mus_raw_count[rownames(mus_raw_count)==&quot;ENSMUSG00000024045&quot;,] #reads很大</span><br></pre></td></tr></table></figure>
<p>最开始未加 -r pos 有问题 reads数目不同，后续计数没有大问题<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">count.matrix&lt;-function()</span><br><span class="line">&#123;</span><br><span class="line">  .libPaths(&quot;/home/tingm/R/x86_64-pc-linux-gnu-library/3.2&quot;)</span><br><span class="line">  library(limma)</span><br><span class="line">  library(edgeR)</span><br><span class="line">    #change</span><br><span class="line">    path&lt;-&quot;/home/tingm/fastq/homo_result_count/&quot;</span><br><span class="line">    #change</span><br><span class="line">      samples&lt;-read.table(&quot;samples.list.txt&quot;,header=F,sep=&quot;\t&quot;)</span><br><span class="line">      samples&lt;-samples[,1]</span><br><span class="line">      #change</span><br><span class="line">      data.raw&lt;-read.table(&quot;Homo_SRR3589956.sgeneCounts&quot;,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;Homo_SRR3589956.sgeneCounts&quot;))</span><br><span class="line">      for(sample in samples)</span><br><span class="line">      &#123;</span><br><span class="line">        print(sample)</span><br><span class="line">        file&lt;-paste(path,sample,sep=&quot;&quot;)</span><br><span class="line">        data.new&lt;-read.table(file,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,sample))</span><br><span class="line">        data.raw&lt;-merge(data.new,data.raw,by=&quot;gene_id&quot;)</span><br><span class="line">      &#125;</span><br><span class="line">     data.raw&lt;-data.raw[-1:-5,]</span><br><span class="line">     #ENSEMBL &lt;- gsub(&quot;\\.\\d*_\\d*&quot;, &quot;&quot;, data.raw$gene_id)</span><br><span class="line">     #row.names(data.raw) &lt;- ENSEMBL</span><br><span class="line">     # ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, mus_raw_count$gene_id)</span><br><span class="line">     write.table(data.raw,&quot;count.matix.csv&quot;,row.names=F,quote=F)</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure></p>
<blockquote>
<p>参考引用来自以下资料<br><a href="https://mp.weixin.qq.com/s/V32iruEVRKJ_1RubuS_lUQ" target="_blank" rel="noopener">PANDA姐的转录组入门(6)：reads计数</a><br><a href="https://mp.weixin.qq.com/s/RolngZkua6GJboHQMfP52w" target="_blank" rel="noopener">生信媛 转录组入门(6)： reads计数</a><br><a href="http://www.jianshu.com/p/24cf44b610a7" target="_blank" rel="noopener">lxmic转录组入门(6)： reads计数</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/14/转录组入门——5/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/14/转录组入门——5/" itemprop="url">转录组入门——5</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-14T18:21:01+08:00">2017-08-14</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/14/转录组入门——5/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/14/转录组入门——5/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="比对软件简介"><a href="#比对软件简介" class="headerlink" title="比对软件简介"></a>比对软件简介</h5><blockquote>
<p><a href="http://www.jianshu.com/p/681e02e7f9af" target="_blank" rel="noopener">比对软件比较</a><br>Bowtie1 <a href="http://bowtie-bio.sourceforge.net/index.shtml" target="_blank" rel="noopener">官方网址</a><br>Bowtie2 <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml" target="_blank" rel="noopener">官方网址</a> <a href="http://blog.leanote.com/post/songtaogui/Bowtie2%E5%AE%89%E8%A3%85%E4%BD%BF%E7%94%A8" target="_blank" rel="noopener">中文教程</a> <a href="http://blog.leanote.com/post/375952584@qq.com/Bowtie2%E7%9A%84%E5%AE%89%E8%A3%85%E4%B8%8E%E4%BD%BF%E7%94%A8" target="_blank" rel="noopener">中文教程</a><br><a href="https://cncbi.github.io/Bowtie2-Manual-CN/" target="_blank" rel="noopener">中文教程</a> <a href="http://fhqdddddd.blog.163.com/blog/static/18699154201422735644352/" target="_blank" rel="noopener">中文教程</a><br>Bowtie1 Bowtie2 <a href="http://blog.csdn.net/soyabean555999/article/details/62235577" target="_blank" rel="noopener">二者区别</a><br><a href="http://www.cnblogs.com/wq242424/p/7025946.html" target="_blank" rel="noopener">Bowtie 安装</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">#下载bowtie2-2.3.2-linux-x86_64.zip解压后有错误 未解决 tbb需要Python2.7</span><br><span class="line">find ./ -name libtbbmalloc_proxy.so.2</span><br><span class="line">echo $LD_LIBRARY_PATH</span><br><span class="line">export LD_LIBRARY_PATH=~/biosoft/tbb2017_20170604oss/lib/:$LD_LIBRARY_PATH</span><br></pre></td></tr></table></figure></p>
<p><a href="http://blog.sina.com.cn/s/blog_83f77c940102wp28.html" target="_blank" rel="noopener">TBB下载地址</a><br><a href="http://www.linuxidc.com/Linux/2014-01/94717.htm" target="_blank" rel="noopener">Linux LD_LIBRARY_PATH</a><br><a href="https://stackoverflow.com/questions/32592956/apt-get-command-not-found" target="_blank" rel="noopener">apt-get 和 yum install</a><br>Tophat <a href="https://ccb.jhu.edu/software/tophat/manual.shtml" target="_blank" rel="noopener">官方网址</a> <a href="http://www.chenlianfu.com/?p=910" target="_blank" rel="noopener">中文教程</a><br>Hisat2 <a href="https://ccb.jhu.edu/software/hisat2/index.shtml" target="_blank" rel="noopener">官方网址</a> <a href="http://blog.biochen.com/archives/337" target="_blank" rel="noopener">中文教程</a><br>STAR <a href="http://www.bio-info-trainee.com/727.html" target="_blank" rel="noopener">中文教程</a> <a href="http://www.cnblogs.com/daimakun/p/5081638.html" target="_blank" rel="noopener">中文教程</a><br>其他比对软件 <a href="http://www.jianshu.com/p/93f96e7538da" target="_blank" rel="noopener">参考</a></p>
</blockquote>
<h5 id="hisat2-下载index"><a href="#hisat2-下载index" class="headerlink" title="hisat2 下载index"></a>hisat2 下载index</h5><blockquote>
<p><a href="http://www.biotrainee.com/thread-26-1-1.html" target="_blank" rel="noopener">bowtie原理</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/hg19.tar.gz</span><br><span class="line">wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/data/mm10.tar.gz</span><br><span class="line">tar -zxvf *.tar.gz</span><br></pre></td></tr></table></figure></p>
</blockquote>
<h5 id="hiat2-Map，sam转bam，name-sort"><a href="#hiat2-Map，sam转bam，name-sort" class="headerlink" title="hiat2 Map，sam转bam，name-sort"></a>hiat2 Map，sam转bam，name-sort</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">#! usr/bin/bash</span><br><span class="line">set -u</span><br><span class="line">set -e</span><br><span class="line">set -o pipefail</span><br><span class="line">hg19_ref=~/fastq/hisat2_index/hg19/genome</span><br><span class="line">mm10_ref=~/fastq/hisat2_index/mm10/genome</span><br><span class="line">hg19_gff=~/fastq/refdata/Genecode/gencode.v26lift37.annotation.gff3</span><br><span class="line">mm10_gff=~/fastq/refdata/Genecode/gencode.vM10.annotation.gtf</span><br><span class="line">data_path=~/fastq</span><br><span class="line">ls Homo*.fastq.gz|while read id;do(hisat2 -t -p 5 -x $hg19_ref -1 $data_path/$&#123;id%_*&#125;_1.fastq.gz -2 $data_path/$&#123;id%_*&#125;_2.fastq.gz 2&gt;$&#123;id%_*&#125;_map.log | /home/tingm/software/samtools-1.3.1/samtools view -Sb  - &gt;$&#123;id%_*&#125;.bam);done</span><br><span class="line">ls Mus*.fastq.gz|while read id;do(hisat2 -t -p 5 -x $mm10_ref -1 $data_path/$&#123;id%_*&#125;_1.fastq.gz -2 $data_path/$&#123;id%_*&#125;_2.fastq.gz 2&gt;$&#123;id%_*&#125;_map.log | /home/tingm/software/samtools-1.3.1/samtools view -Sb  -  &gt;$&#123;id%_*&#125;.bam);done</span><br><span class="line">ls *.bam|while read id;do(/home/tingm/software/samtools-1.3.1/samtools sort -n -@ 5 $id  -o $&#123;id%%.*&#125;.Nsorted.bam);done</span><br><span class="line">#ls Homo*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $hg19_gff &gt; $&#123;id%%.*&#125;.geneCounts 2&gt; $&#123;id%%.*&#125;.HTseq.log);done</span><br><span class="line">#ls Mus*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $mm10_gff &gt; $&#123;id%%.*&#125;.geneCounts 2&gt; $&#123;id%%.*&#125;.HTseq.log);done</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-614c3461111f4783.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="bam"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-14e7421d4dbb7c5c.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="name-sorted bam"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-ffab566ae8f0b434.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="sorted bam"></p>
<h5 id="BWA"><a href="#BWA" class="headerlink" title="BWA"></a>BWA</h5><p><a href="https://starsyi.github.io/2016/05/24/BWA-%E5%91%BD%E4%BB%A4%E8%AF%A6%E8%A7%A3/" target="_blank" rel="noopener">中文参考</a></p>
<h5 id="Samtools"><a href="#Samtools" class="headerlink" title="Samtools"></a>Samtools</h5><p><a href="http://samtools.sourceforge.net/" target="_blank" rel="noopener">官方网址</a><br><a href="http://www.bio-info-trainee.com/518.html" target="_blank" rel="noopener">下载参考</a><br><a href="http://blog.csdn.net/sinat_38163598/article/details/72910115" target="_blank" rel="noopener">中文教程</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">wget https://github.com/samtools/samtools/releases/download/1.5/samtools-1.5.tar.bz2</span><br><span class="line">tar jxf samtools-1.5.tar.bz2</span><br><span class="line">cd samtools-1.5</span><br><span class="line">make</span><br><span class="line">echo &apos;PATH=$PATH:~/biosoft/samtools-1.5/&apos; &gt;&gt; ~/.bashrc</span><br><span class="line">source ~/.bashrc</span><br></pre></td></tr></table></figure></p>
<h5 id="两种排序方式及索引-默认排序是根据染色体位置-n-reads名称进行排序"><a href="#两种排序方式及索引-默认排序是根据染色体位置-n-reads名称进行排序" class="headerlink" title="两种排序方式及索引(默认排序是根据染色体位置 -n reads名称进行排序)"></a>两种排序方式及索引(默认排序是根据染色体位置 -n reads名称进行排序)</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">ls *.sra.bam|while read id;do(/home/tingm/software/samtools-1.3.1/samtools sort -@ 5 $id -o $&#123;id%%.*&#125;.sorted.bam);done</span><br><span class="line">ls *.sorted.bam|while read id;do(/home/tingm/software/samtools-1.3.1/samtools index $&#123;id%%.*&#125;.sorted.bam;done</span><br><span class="line">#head -1000 Homo_SRR3589956.sra.bam &gt; test.bam</span><br><span class="line">#samtools view test.bam | head</span><br><span class="line">#head -1000 Homo_SRR3589956.Nsorted.bam &gt; test.nsorted.bam</span><br><span class="line">#samtools view test.nsorted.bam | head</span><br><span class="line">#head -1000 Homo_SRR3589956.sorted.bam &gt; test.sorted.bam</span><br><span class="line">#samtools view test.sorted.bam | head</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-1126ea86c945546c.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="bam"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-65746a73521c7f0b.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="reads name"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-f50f20d61159b837.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="chr"></p>
<h5 id="IGV-载入bam文件及其index-如何查看，理解结果？"><a href="#IGV-载入bam文件及其index-如何查看，理解结果？" class="headerlink" title="IGV 载入bam文件及其index  如何查看，理解结果？"></a>IGV 载入bam文件及其index  如何查看，理解结果？</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">igv.sh</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-f0e2a6fa977e4068.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="sort.bam"></p>
<h5 id="质量控制"><a href="#质量控制" class="headerlink" title="质量控制"></a>质量控制</h5><blockquote>
<p>常用工具有<br>Picard <a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener">https://broadinstitute.github.io/picard/</a><br>RSeQC <a href="http://rseqc.sourceforge.net/" target="_blank" rel="noopener">http://rseqc.sourceforge.net/</a><br>Qualimap <a href="http://qualimap.bioinfo.cipf.es/" target="_blank" rel="noopener">http://qualimap.bioinfo.cipf.es/</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"># Python2.7环境下</span><br><span class="line">pip install RSeQC</span><br><span class="line"># 先对bam文件进行统计分析， 从结果上看是符合70~90的比对率要求</span><br><span class="line">bam_stat.py -i Homo_SRR3589956.sorted.bam</span><br><span class="line"># 基因组覆盖率的QC需要提供bed文件，可以直接RSeQC的网站下载，或者可以用gtf转换</span><br><span class="line">wget https://sourceforge.net/projects/rseqc/files/BED/Human_Homo_sapiens/hg19_RefSeq.bed.gz/download -O hg19_RefSeq.bed.gz</span><br><span class="line">gzip -d hg19_RefSeq.bed.gz</span><br><span class="line">read_distribution.py -i Homo_SRR3589956.sorted.bam -r ~/fastq/refdata/hg19_RefSeq.bed</span><br></pre></td></tr></table></figure></p>
</blockquote>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-43015f11229d4941.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="bam文件进行统计分析"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-0ae3b95e94e3037b.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="基因组覆盖率的QC"></p>
<blockquote>
<p>参考资料<br><a href="https://mp.weixin.qq.com/s/C66OKH08mjWvHOmq-s8V1A" target="_blank" rel="noopener">PANDA姐的转录组入门(5)：序列比对</a><br><a href="http://www.jianshu.com/p/681e02e7f9af" target="_blank" rel="noopener">hoptop 转录组入门（5）： 序列比对</a><br><a href="http://www.jianshu.com/p/93f96e7538da" target="_blank" rel="noopener">lxmic转录组入门（5）： 序列比对</a></p>
</blockquote>
<h5 id="Picard"><a href="#Picard" class="headerlink" title="Picard"></a>Picard</h5><p><a href="https://github.com/broadinstitute/picard/releases/tag/2.11.0" target="_blank" rel="noopener">下载参考2</a><br><a href="https://github.com/broadinstitute/picard/releases/tag/1.141" target="_blank" rel="noopener">下载版本较旧 因需java1.8 放弃旧版本</a><br><a href="https://github.com/broadinstitute/picard" target="_blank" rel="noopener">下载参考1 java需1.8</a><br><a href="https://java.com/zh_CN/download/help/linux_x64_install.xml#download" target="_blank" rel="noopener">java8下载</a><br><a href="http://www.cnblogs.com/OA-maque/p/4835040.html" target="_blank" rel="noopener">下载参考</a><br><a href="http://www.bioinfo-scrounger.com/archives/183" target="_blank" rel="noopener">终版下载参考</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">wget https://github.com/broadinstitute/picard/releases/download/2.11.0/picard.jar</span><br><span class="line">java -jar picard.jar -h</span><br></pre></td></tr></table></figure></p>
<p><a href="https://cncbi.github.io/Picard-Manual-CN/" target="_blank" rel="noopener">使用参考</a><br><a href="http://blog.sina.com.cn/s/blog_6721167201018jik.html" target="_blank" rel="noopener">使用参考</a></p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/11/转录组入门——4/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/11/转录组入门——4/" itemprop="url">转录组入门——4</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-11T22:08:50+08:00">2017-08-11</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/11/转录组入门——4/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/11/转录组入门——4/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h4 id="参考基因组-hg19-下载"><a href="#参考基因组-hg19-下载" class="headerlink" title="参考基因组(hg19)下载"></a>参考基因组(hg19)下载</h4><blockquote>
<p>NCBI     UCSC   ENSEMBL数据库 对应关系<br>GRCh36  hg18     ENSEMBL release_52<br>GRCh37  hg19     ENSEMBL release_59/61/64/68/69/75<br>GRCh38  hg38     ENSEMBL  release_76/77/78/80/81/82</p>
</blockquote>
<h5 id="NCBI"><a href="#NCBI" class="headerlink" title="NCBI"></a>NCBI</h5><p><a href="https://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/ARCHIVE/BUILD.37.3/Assembled_chromosomes/seq/" target="_blank" rel="noopener">https://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/ARCHIVE/BUILD.37.3/Assembled_chromosomes/seq/</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">#! /bin/bash</span><br><span class="line">file=$1</span><br><span class="line">chrs=`awk &apos;&#123;print $1&#125;&apos; $file`</span><br><span class="line">for i in $&#123;chrs[@]&#125;;</span><br><span class="line">do echo $i;</span><br><span class="line">wget https://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/ARCHIVE/BUILD.37.3/Assembled_chromosomes/seq/hs_ref_GRCh37.p5_$&#123;i&#125;.fa.gz;</span><br><span class="line">done</span><br><span class="line">gunzip *.gz</span><br><span class="line">for i in $&#123;chrs[@]&#125;;</span><br><span class="line">do cat hs_ref_GRCh37.p5_$&#123;i&#125;.fa &gt;&gt; hg19.fasta;</span><br><span class="line">done</span><br><span class="line">rm -fr *.fa</span><br></pre></td></tr></table></figure></p>
<h5 id="Emsembl-过于大-应该为3G大小？"><a href="#Emsembl-过于大-应该为3G大小？" class="headerlink" title="Emsembl 过于大 应该为3G大小？"></a>Emsembl 过于大 应该为3G大小？</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">nohup wget ftp://ftp.ensembl.org/pub/release-75/fasta/homo_sapiens/dna/Homo_sapiens.GRCh37.75.dna.toplevel.fa.gz &amp;</span><br></pre></td></tr></table></figure>
<h5 id="UCSC"><a href="#UCSC" class="headerlink" title="UCSC"></a>UCSC</h5><p><a href="https://www.jianshu.com/p/f101c9238fc5" target="_blank" rel="noopener">lxmic 转录组入门（4）：了解参考基因组及基因注释</a><br><img src="http://upload-images.jianshu.io/upload_images/2116382-88c008789cd1dd51.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 1 select Downloads —&gt; select genome data"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-2e1ade36ab2e236d.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 2-1 choose hg38"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-d94fcaa75393bf9c.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 2-2 download"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-f4d4898048b1b2a5.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 3-1 choose hg19"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-9ab848763bf29e4f.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Figure 3-2 download"><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">nohup wget http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/chromFa.tar.gz &amp;</span><br><span class="line">jobs</span><br><span class="line">fg %1</span><br><span class="line">bg %1</span><br><span class="line">kill %1</span><br><span class="line">tar -zxvf chromFa.tar.gz</span><br><span class="line">cat *.fa &gt; hg19.fa</span><br><span class="line">rm -rf chr*</span><br></pre></td></tr></table></figure></p>
<h4 id="参考基因组注释下载"><a href="#参考基因组注释下载" class="headerlink" title="参考基因组注释下载"></a>参考基因组注释下载</h4><h5 id="NCBI-1"><a href="#NCBI-1" class="headerlink" title="NCBI"></a>NCBI</h5><p><img src="http://upload-images.jianshu.io/upload_images/2116382-02d615eeeb235c1b.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="进入NCBI的Genome数据库"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-30b4ca74609e930f.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Download/FTP"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-75b1fba61ea288b8.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="选择物种：人"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-98122c5892131b89.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="进入下载hg38GTF注释文件"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-40eefc8f0afdd683.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="hg38GTF注释文件"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-eedc1feedef01e9a.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="其他版本"><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">nohup wget https://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/GFF/ref_GRCh38.p7_top_level.gff3.gz &amp;</span><br><span class="line">gzip -d ref_GRCh38.p7_top_level.gff3.gz</span><br></pre></td></tr></table></figure></p>
<h5 id="Ensembl"><a href="#Ensembl" class="headerlink" title="Ensembl"></a>Ensembl</h5><p><a href="http://www.bio-info-trainee.com/1469.html" target="_blank" rel="noopener">生信菜鸟团 基因组各种版本对应关系</a><br><img src="http://upload-images.jianshu.io/upload_images/2116382-78260f48c383e5d9.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Downloads"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-8e63dca8af3c1054.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="select"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-64ee6b0553666966.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="FTP site"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-45b8166f6b848095.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="release-75"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-7b9911c2b8cd58ba.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="GTF"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-27b6ef4681c23204.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="Homo sapiens"><br><img src="http://upload-images.jianshu.io/upload_images/2116382-6a2c9ddd309802a9.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="download"><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">nohup wget ftp://ftp.ensembl.org/pub/release-75/gtf/homo_sapiens/Homo_sapiens.GRCh37.75.gtf.gz &amp;</span><br><span class="line">gzip -d Homo_sapiens.GRCh37.75.gtf.gz</span><br></pre></td></tr></table></figure></p>
<h5 id="UCSC-1"><a href="#UCSC-1" class="headerlink" title="UCSC"></a>UCSC</h5><p><img src="http://upload-images.jianshu.io/upload_images/2116382-5cf747fa93b9d9b0.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="choose Table Browser"></p>
<blockquote>
<p>Navigate to <a href="http://genome.ucsc.edu/cgi-bin/hgTables" target="_blank" rel="noopener">http://genome.ucsc.edu/cgi-bin/hgTables</a><br>Select the following options:<br>clade: Mammal<br>genome: Human<br>assembly: Feb. 2009 (GRCh37/hg19)<br>group: Genes and Gene Predictions<br>track: UCSC Genes<br>table: knownGene<br>region: Select “genome” for the entire genome.<br>output format: GTF - gene transfer format<br>output file: enter a file name to save your results to a file, or leave blank to display results in the browser<br>Click ‘get output’.</p>
</blockquote>
<h5 id="Genecode"><a href="#Genecode" class="headerlink" title="Genecode"></a>Genecode</h5><p><a href="http://www.jianshu.com/p/f101c9238fc5" target="_blank" rel="noopener">转录组入门4-了解参考基因组及基因注释</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_26/GRCh37_mapping/gencode.v26lift37.annotation.gtf.gz</span><br><span class="line">wget ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_26/GRCh37_mapping/gencode.v26lift37.annotation.gff3.gz</span><br><span class="line">gunzip *.gz </span><br><span class="line">rm -rf *.gz</span><br></pre></td></tr></table></figure></p>
<h5 id="IGV"><a href="#IGV" class="headerlink" title="IGV"></a>IGV</h5><p><a href="http://www.biotrainee.com/thread-1998-1-1.html" target="_blank" rel="noopener">转录组作业（四）：来自零基础的小白</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">![软件自带基因组-TP53](http://upload-images.jianshu.io/upload_images/2116382-35124fbd81c8bdab.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240)</span><br><span class="line">![hg19.fa 蓝色带没有变化？？](http://upload-images.jianshu.io/upload_images/2116382-75961165acf84dd2.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240)</span><br><span class="line"></span><br><span class="line">***问题：linux服务器未安装上IGV；自行下载的hg19比对结果和软件自带不一致***</span><br></pre></td></tr></table></figure></p>
<p>sudo apt-get install glib-networking-common<br>```</p>
<blockquote>
<p>其他参考资料<br><a href="http://blog.sina.com.cn/s/blog_3d37a5690100wne7.html" target="_blank" rel="noopener">ncbi中的基因组genome全序列下载 </a><br><a href="http://www.bio-info-trainee.com/1985.html" target="_blank" rel="noopener">参考基因组下载及index建立</a><br><a href="http://www.bio-info-trainee.com/1469.html" target="_blank" rel="noopener">基因组各种版本对应关系</a><br><a href="https://www.yaolibio.com/2016/09/01/retrieve-genome-data-from-ucsc/" target="_blank" rel="noopener">从UCSC获取基因组序列及基因注释</a><br><a href="http://www.jianshu.com/p/f101c9238fc5" target="_blank" rel="noopener">转录组入门4-了解参考基因组及基因注释</a><br><a href="http://www.cnblogs.com/Demo1589/p/6950196.html" target="_blank" rel="noopener">关于基因组注释文件GTF的解释</a><br><a href="http://www.jianshu.com/p/48b5a0972301" target="_blank" rel="noopener">GTF/GFF文件的差异及其看相互转换</a><br><a href="http://mp.weixin.qq.com/s/mq--yW_vmm2C2ZpnYiPq-A" target="_blank" rel="noopener">ensembel及NCBI注释文件下载</a></p>
</blockquote>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  <article class="index-post card" itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/10/转录组入门——3/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
        <meta itemprop="name" content="mt">
        <meta itemprop="description" content="">
        <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
        <meta itemprop="name" content="Hexo">
      </span>
    </div>
    <header class="post-header">
      <h2 class="post-title" itemprop="name headline">
        <a class="post-title-link post-title-link-external" href="/2017/08/10/转录组入门——3/" itemprop="url">转录组入门——3</a>
      </h2>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-10T19:34:02+08:00">2017-08-10</time>
        </span>
        
        
        
        
        <span class="post-meta-divider"> | </span>
        
          
          <span class="post-comment-count">
            <i class="far fa-comments"></i> <a href="/2017/08/10/转录组入门——3/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/10/转录组入门——3/" itemprop="commentsCount"></span></a>
          </span>
          
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      
      <h5 id="SRA-Toolkit-软件"><a href="#SRA-Toolkit-软件" class="headerlink" title="SRA Toolkit 软件"></a>SRA Toolkit 软件</h5><p><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc" target="_blank" rel="noopener">官方网址</a></p>
<h5 id="软件下载（CentOS-6-8）"><a href="#软件下载（CentOS-6-8）" class="headerlink" title="软件下载（CentOS 6.8）"></a>软件下载（CentOS 6.8）</h5><p><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&amp;f=std#s-2" target="_blank" rel="noopener">官方参考</a><br><a href="http://www.jianshu.com/p/6a4855023330" target="_blank" rel="noopener">中文参考</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">#下载软件包</span><br><span class="line">wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.8.2-1/sratoolkit.2.8.2-1-centos_linux64.tar.gz</span><br><span class="line">#解压</span><br><span class="line">tar -zxf sratoolkit.2.8.2-1-centos_linux64.tar.gz </span><br><span class="line">#解压后文件夹移动到biosoft目录下</span><br><span class="line">mv sratoolkit.2.8.2-1-centos_linux64/ biosoft/</span><br><span class="line">#设置绝对路径</span><br><span class="line">echo &apos;PATH=$PATH:~/biosoft/sratoolkit.2.8.2-1-centos_linux64/bin&apos; &gt;&gt; ~/.bashrc</span><br><span class="line">source ~/.bashrc</span><br><span class="line">#测试安装</span><br><span class="line">fastq-dump -h</span><br></pre></td></tr></table></figure></p>
<h5 id="fastq-dump参数"><a href="#fastq-dump参数" class="headerlink" title="fastq-dump参数"></a>fastq-dump参数</h5><p><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc&amp;f=fastq-dump" target="_blank" rel="noopener">官方参考</a><br><a href="http://www.biotrainee.com/forum.php?mod=viewthread&amp;tid=1885&amp;fromuid=1685" target="_blank" rel="noopener">中文参考</a><br>–split-3 如果是双端测序数据，则输出两个文件，如果不是则只输出一个文件<br>-O 指定输出路径<br>–gzip 指定输出格式为gzip压缩格式<br>-A –accession 输入sra文件可以是绝对路径</p>
<h5 id="SRA文件转换为fastq文件-仅关注RNA-Seq数据"><a href="#SRA文件转换为fastq文件-仅关注RNA-Seq数据" class="headerlink" title="SRA文件转换为fastq文件(仅关注RNA-Seq数据)"></a>SRA文件转换为fastq文件(仅关注RNA-Seq数据)</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">for i in `seq 56 62`;do fastq-dump --gzip --split-3 -O ~/fastq/ -A ~/ncbi/public/sra/SRR35899$&#123;i&#125;.sra;done</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-f592e2a5c22a432f.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="gz"></p>
<h5 id="生信名词理解"><a href="#生信名词理解" class="headerlink" title="生信名词理解"></a>生信名词理解</h5><p>测序reads：高通量测序平台产生的序列标签就称为reads<br>GC含量：<a href="https://zh.wikipedia.org/wiki/GC%E5%90%AB%E9%87%8F" target="_blank" rel="noopener">中文参考</a><br>质量值：每测一个碱基会有相应的质量值，质量值用于衡量测序准确度<a href="http://www.cnblogs.com/daimakun/p/5092312.html" target="_blank" rel="noopener">中文参考</a><a href="https://en.wikipedia.org/wiki/FASTQ_format" target="_blank" rel="noopener">英文参考</a><br>adapter：接头是一段短的序列已知的核酸链，用于连接序列未知的目标测序片段<br>index：索引序列，是一段很短的寡居核酸链，用于在多个样品混合测序时，标记不同的样品，为区分不同样本而构建<a href="http://www.itwendao.com/article/detail/467548.html" target="_blank" rel="noopener">参考</a></p>
<h5 id="FastQC-软件"><a href="#FastQC-软件" class="headerlink" title="FastQC 软件"></a>FastQC 软件</h5><p><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank" rel="noopener">官方网址</a></p>
<h5 id="软件下载（CentOS-6-8）-1"><a href="#软件下载（CentOS-6-8）-1" class="headerlink" title="软件下载（CentOS 6.8）"></a>软件下载（CentOS 6.8）</h5><p><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/INSTALL.txt" target="_blank" rel="noopener">官方参考</a><br><a href="http://www.jianshu.com/p/6a4855023330" target="_blank" rel="noopener">中文参考</a><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">java -version</span><br><span class="line">wget http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.5.zip</span><br><span class="line">unzip fastqc_v0.11.5.zip</span><br><span class="line">mv FastQC/ biosoft/</span><br><span class="line">cd ~/biosoft/FastQC</span><br><span class="line">chmod 755 fastqc</span><br><span class="line">echo &apos;PATH=$PATH:~/biosoft/FastQC/&apos; &gt;&gt; ~/.bashrc</span><br><span class="line">source ~/.bashrc</span><br><span class="line">fastqc -h</span><br></pre></td></tr></table></figure></p>
<h5 id="fastqc"><a href="#fastqc" class="headerlink" title="fastqc"></a>fastqc</h5><p>-o 输出目录，需自己创建目录<br>–extract 是否解压输出文件，默认自动解压缩zip文件<br>-f 指定输入文件的类型，支持fastq|bam|sam三种格式的文件，默认自动识别<br>-t 同时处理的文件数目<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">fastqc -o fastqc_out --extract *.fastq.gz</span><br></pre></td></tr></table></figure></p>
<h5 id="fastqc质控结果"><a href="#fastqc质控结果" class="headerlink" title="fastqc质控结果"></a>fastqc质控结果</h5><p><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/" target="_blank" rel="noopener">官方参考</a><br><a href="http://www.jianshu.com/p/14fd4de54402" target="_blank" rel="noopener">中文参考</a><br><a href="http://fbb84b26.wiz03.com/share/s/3XK4IC0cm4CL22pU-r1HPcQQ2irG2836uQYm2iZAyh1Zwf3_" target="_blank" rel="noopener">中文参考</a></p>
<h5 id="批量查看MultiQC"><a href="#批量查看MultiQC" class="headerlink" title="批量查看MultiQC"></a>批量查看MultiQC</h5><p><a href="http://multiqc.info/docs/" target="_blank" rel="noopener">官方网址</a><br><a href="http://fbb84b26.wiz03.com/share/s/3XK4IC0cm4CL22pU-r1HPcQQ1iRTvV2GwkwL2AaxYi2fXHP7" target="_blank" rel="noopener">中文参考</a></p>
<h5 id="软件下载及使用"><a href="#软件下载及使用" class="headerlink" title="软件下载及使用"></a>软件下载及使用</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda install -c bioconda multiqc</span><br><span class="line">#进入对应文件，扫描zip文件，忽略html文件</span><br><span class="line">multiqc ~/fastq/fastqc_out/*fastqc.zip --ignore *.html</span><br></pre></td></tr></table></figure>
<h5 id="结果查看-重点关注的部分参考"><a href="#结果查看-重点关注的部分参考" class="headerlink" title="结果查看 重点关注的部分参考"></a>结果查看 重点关注的部分参考</h5><p><a href="https://zhuanlan.zhihu.com/p/27985781" target="_blank" rel="noopener">青山屋主 转录组入门3-测序数据质量检查</a></p>

      
    </main>
    <footer class="post-footer">
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  
  <nav class="pagination">
    <a class="extend prev" rel="prev" href="/"><i class="fas fa-angle-left"></i></a><a class="page-number" href="/">1</a><span class="page-number current">2</span><a class="page-number" href="/page/3/">3</a><a class="extend next" rel="next" href="/page/3/"><i class="fas fa-angle-right"></i></a>
  </nav>
  
  
  

<div class="comments" id="comments">
  
  
  <script type="text/javascript">
  _hcwp = window._hcwp || [];
  _hcwp.push({ "widget": "Bloggerstream", "widget_id": "tingm831@gmail.com", "selector": ".hc-comment-count", "label": "{\%COUNT%\}" });
  
  (function() {
    if ("HC_LOAD_INIT" in window) {
      return;
    }
    HC_LOAD_INIT = true;
    var lang = (navigator.language || navigator.systemLanguage || navigator.userLanguage || "en").substr(0, 2).toLowerCase();
    var hcc = document.createElement("script");
    hcc.type = "text/javascript";
    hcc.defer = true;
    hcc.src = ("https:" === document.location.protocol ? "https" : "http") + "://w.hypercomments.com/widget/hc/tingm831@gmail.com/" + lang + "/widget.js";
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(hcc, s.nextSibling);
  })();
  </script>
  
</div>



  
</section>

          </div>
          
          
          
<aside class="sidebar" id="sidebar">
  
  <div class="search">
    <div class="form-group">
      <i class="fas fa-search"></i> <input type="text" id="search-input" name="q" results="0" placeholder="Search" class="form-control"/>
    </div>
  </div>
  <div class="search-result-box" id="search-result"></div>
  
  
<div class="info sidebar-item" id="info">
  
  <img class="author-avatar" src="/images/avatar.png" alt="mt">
  
  <p class="author-name">mt</p>
  <p class="author-description"> </p>
  <div class="site-count">
    
    <div class="archives-count">
      <p>Archives</p>
      <p><a href="/archives/">22</a></p>
    </div>
    
    
    
    <span class="site-count-divider"> | </span>
    
    <div class="categories-count">
      <p>Categories</p>
      <p><a href="/categories/">0</a></p>
    </div>
    
    
    
    <span class="site-count-divider"> | </span>
    
    <div class="tags-count">
      <p>Tags</p>
      <p><a href="/tags/">1</a></p>
    </div>
    
  </div>
  
  <div class="rss">
    <a class="rss-link button sidebar-item" href="/atom.xml"><i class="fas fa-rss"></i> RSS</a>
  </div>
  
</div>


  <div class="sidebar-sticky">
    
    
    
<hr>
<div class="social-link sidebar-item">
  <p><i class="far fa-address-card"></i> Social Links</p>
  <ul>
    
    <li><i class="fab fa-github"></i> <a href="https://github.com/mt831" target="_blank">GitHub</a></li>
    
  </ul>
</div>


    
    
  </div>
</aside>


          
        </div>
      </div>
    </main>
    
<footer id="footer" class="footer">
  <div class="container">
    <div class="back-to-top">
      <a id="back-to-top"><i class="fas fa-angle-double-up"></i></a>
    </div>
    <div class="footer-container">
      <div class="footer-left">
        <div class="copyright">
          <span class="author">mt</span>
          <span class="year"><i class="far fa-copyright"></i> 2018-4-1 - 2018</span>
          
        </div>
        
        
<div class="busuanzi">
  <span id="busuanzi_container_site_pv"><i class="fas fa-eye"></i> <span id="busuanzi_value_site_pv"></span></span>
  <span id="busuanzi_container_site_uv"><i class="fas fa-user"></i> <span id="busuanzi_value_site_uv"></span></span>
  <span id="busuanzi_container_page_pv"><i class="far fa-file-alt"></i> <span id="busuanzi_value_page_pv"></span></span>
</div>


        
      </div>
      <div class="footer-right">
        <div class="custom-info">
          
        </div>
        <div class="powered-by">
          Proudly Powered by <a href="https://hexo.io/" target="_blank">Hexo</a> | Theme is <a href="https://github.com/AlynxZhou/hexo-theme-aria/" target="_blank">ARIA</a>
        </div>
      </div>
    </div>
  </div>
</footer>


  </body>
</html>
