<!DOCTYPE html><html lang="zh-CN"><head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"><meta content="yes" name="apple-mobile-web-app-capable"><meta content="black-translucent" name="apple-mobile-web-app-status-bar-style"><meta content="telephone=no" name="format-detection"><meta name="description"><title>转录组入门——6 | blog</title><link rel="stylesheet" type="text/css" href="/css/style.css?v=0.0.0"><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/normalize/8.0.0/normalize.min.css"><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/pure/1.0.0/pure-min.css"><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/pure/1.0.0/grids-responsive-min.css"><link rel="stylesheet" href="//cdn.bootcss.com/font-awesome/4.7.0/css/font-awesome.min.css"><script type="text/javascript" src="//cdn.bootcss.com/jquery/3.3.1/jquery.min.js"></script><link rel="Shortcut Icon" type="image/x-icon" href="/favicon.ico"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="apple-touch-icon-precomposed" href="/apple-touch-icon.png"><link rel="alternate" type="application/atom+xml" href="/atom.xml"></head><body><div class="body_container"><div id="header"><div class="site-name"><h1 class="hidden">转录组入门——6</h1><a id="logo" href="/.">blog</a><p class="description">学习记录</p></div><div id="nav-menu"><a class="current" href="/."><i class="fa fa-home"> 首页</i></a><a href="/archives/"><i class="fa fa-archive"> 归档</i></a><a href="/about/"><i class="fa fa-user"> 关于</i></a><a href="/atom.xml"><i class="fa fa-rss"> 订阅</i></a></div></div><div class="pure-g" id="layout"><div class="pure-u-1 pure-u-md-3-4"><div class="content_container"><div class="post"><h1 class="post-title">转录组入门——6</h1><div class="post-meta">Aug 15, 2017<span> | </span><span class="category"><a href="/categories/转录组入门/">转录组入门</a></span><script src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js" async></script><span id="busuanzi_container_page_pv"> | <span id="busuanzi_value_page_pv"></span><span> 阅读</span><span> |  本文共计 </span><span class="post-count">2,029</span><span> 字  |  阅读约需 </span><span class="post-count">10</span><span> 分钟 </span></span></div><div class="post-content"><h5 id="实现reads计数的软件教程"><a href="#实现reads计数的软件教程" class="headerlink" title="实现reads计数的软件教程"></a>实现reads计数的软件教程</h5><blockquote>
<p>map后获得了每条read比对到基因组上的位置，后面需要把基因组的位置对应到基因。这个提供基因位置的注释文件通常以GTF或GFF3格式呈现。基因表达量最直接的分析手段就是计算比对到每个基因的reads有多少条，在转录组测序中，我们通常称这个数字为count<br>对每个基因计数之后得到的count matrix再后续的分析中，要注意标准化的问题。<br>比较同一个样本(within-sample)不同基因之间的表达情况，需要考虑到转录本长度，因为转录本越长，那么检测的片段也会更多。比较不同样本（across sample）同一个基因的表达情况，不必在意转录本长度，但要考虑到测序深度（sequence depth)，测序深度越高，检测到的概率越大。除这两因素外，还需考虑GC%导致的偏差及测序仪器的系统偏差。<br>目前对read count标准化的算法有RPKM（SE）, FPKM（PE），TPM, TMM等。但是，有一些下游分析的软件会要求是输入的count matrix是原始数据，未经标准化，比如说DESeq2，这个时候你需要注意你上一步所用软件会不会进行标准化。<br>计数分为三个水平： gene-level, transcript-level, exon-usage-level<br>标准化方法： FPKM RPKM TMM TPM</p>
</blockquote>
<h5 id="Gene-level"><a href="#Gene-level" class="headerlink" title="Gene-level"></a>Gene-level</h5><blockquote>
<p>HTSeq <a href="https://htseq.readthedocs.io/en/release_0.9.1/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.chenlianfu.com/?p=2438" target="_blank" rel="noopener">中文教程</a><br>featureCounts <a href="http://bioinf.wehi.edu.au/featureCounts/" target="_blank" rel="noopener">官方网址</a> Rsubread <a href="https://bioconductor.org/packages/release/bioc/html/Rsubread.html" target="_blank" rel="noopener">官方网址</a><br>bedtools multicov <a href="https://bedtools.readthedocs.io/en/latest/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.biotrainee.com/thread-153-1-1.html" target="_blank" rel="noopener">中文教程</a><br>Qualimap <a href="http://qualimap.bioinfo.cipf.es/" target="_blank" rel="noopener">官方网址</a> <a href="http://blog.sciencenet.cn/blog-2970729-1070459.html" target="_blank" rel="noopener">中文教程</a><br>GenomicRanges <a href="https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html" target="_blank" rel="noopener">官方网址</a></p>
</blockquote>
<h5 id="Transcript-level"><a href="#Transcript-level" class="headerlink" title="Transcript-level"></a>Transcript-level</h5><blockquote>
<p>cufflink <a href="https://cole-trapnell-lab.github.io/cufflinks/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.chenlianfu.com/?p=623" target="_blank" rel="noopener">中文教程</a><br>stringtie <a href="https://ccb.jhu.edu/software/stringtie/" target="_blank" rel="noopener">官方网址</a> <a href="http://www.bio-info-trainee.com/2073.html" target="_blank" rel="noopener">中文教程</a><br>上述软件都是alignment-based<br>alignment-free软件，如kallisto, silfish, salmon，能够省去比对这一步，直接得到read count，在运行效率上更高。不过最近一篇文献Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis指出这类方法在估计丰度时存在样本特异性和读长偏差<br>salmon</p>
</blockquote>
<h5 id="Exon-usage-level"><a href="#Exon-usage-level" class="headerlink" title="Exon-usage-level"></a>Exon-usage-level</h5><blockquote>
<p>和基因水平的统计类似。但是值得注意的是为了更好的计数，我们需要提供无重叠的外显子区域的gtf文件Detecting differential usage of exons from RNA-seq data。用于分析差异外显子使用的DEXSeq提供了一个Python脚本（dexseq_prepare_annotation.py）执行这个任务。</p>
</blockquote>
<h5 id="基因表达量归一算法"><a href="#基因表达量归一算法" class="headerlink" title="基因表达量归一算法"></a>基因表达量归一算法</h5><p>待续 <a href="https://zhuanlan.zhihu.com/p/26750663" target="_blank" rel="noopener">参考</a></p>
<h5 id="htseq-count"><a href="#htseq-count" class="headerlink" title="htseq-count"></a>htseq-count</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">ls Homo*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $hg19_gff &gt; $&#123;id%_*&#125;.geneCounts 2&gt; $&#123;id%_*&#125;.HTseq.log);done</span><br><span class="line">ls Mus*.Nsorted.bam|while read id;do(htseq-count -f bam -i gene_name -s no $id $mm10_gff &gt; $&#123;id%_*&#125;.geneCounts 2&gt; $&#123;id%_*&#125;.HTseq.log);done</span><br><span class="line">wc -l Homo*.geneCounts </span><br><span class="line">head -n 4 Homo*.geneCounts</span><br></pre></td></tr></table></figure>
<blockquote>
<p>-f bam/sam： 指定输入文件格式，默认SAM<br>-r name/pos: 你需要利用samtool sort对数据根据read name或者位置进行排序，默认是name<br>-s yes/no/reverse: 数据是否来自于strand-specific assay。DNA是双链的，所以需要判断到底来自于哪条链。如果选择了no， 那么每一条read都会跟正义链和反义链进行比较。默认的yes对于双端测序表示第一个read都在同一个链上，第二个read则在另一条链上。<br>-a 最低质量， 剔除低于阈值的read<br>-m 模式 union（默认）, intersection-strict and intersection-nonempty。一般而言就用默认的，作者也是这样认为的。<br>-i id attribute: 在GTF文件的最后一栏里，会有这个基因的多个命名方式（如下）， RNA-Seq数据分析常用的是gene_id， 当然你可以写一个脚本替换成其他命名方式。</p>
</blockquote>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-6aaf8956467c4fb6.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="homo.genecounts"></p>
<h5 id="编写脚本合并所有的样本为表达矩阵"><a href="#编写脚本合并所有的样本为表达矩阵" class="headerlink" title="编写脚本合并所有的样本为表达矩阵"></a>编写脚本合并所有的样本为表达矩阵</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">sed -i &apos;$d&apos; *geneCounts</span><br><span class="line">perl -lne &apos;if($ARGV=~/Homo_(.*).geneCounts/)&#123;print &quot;$1\t$_&quot;&#125;&apos; *|grep -v Homo_sapiens&gt;hg.count</span><br><span class="line">perl -lne &apos;if($ARGV=~/Mus_(.*).geneCounts/)&#123;print &quot;$1\t$_&quot;&#125;&apos; *|grep -v Mus &gt;mus.count</span><br><span class="line"></span><br><span class="line">hg &lt;- read.csv(file = &quot;hg.count&quot;,header = F,sep = &quot;\t&quot;)</span><br><span class="line">colnames(hg) &lt;- c(&apos;sample&apos;,&apos;gene&apos;,&apos;count&apos;)</span><br><span class="line">library(reshape2)</span><br><span class="line">hg_reads &lt;- dcast(hg,formula = gene ~ sample)</span><br><span class="line">write.table(hg_reads,file = &quot;hg_join.count&quot;,sep = &quot;\t&quot;,quote = FALSE,row.names = FALSE)</span><br><span class="line"></span><br><span class="line">mus &lt;- read.csv(file = &quot;mus.count&quot;,header = F,sep = &quot;\t&quot;)</span><br><span class="line">colnames(mus) &lt;- c(&apos;sample&apos;,&apos;gene&apos;,&apos;count&apos;)</span><br><span class="line">library(reshape2)</span><br><span class="line">mus_reads &lt;- dcast(mus,formula = gene ~ sample)</span><br><span class="line">write.table(mus_reads,file = &quot;mus_join.count&quot;,sep = &quot;\t&quot;,quote = FALSE,row.names = FALSE)</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-bf2396936c8bb516.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="homo_table"></p>
<h5 id="对表达矩阵可进行简单的探索"><a href="#对表达矩阵可进行简单的探索" class="headerlink" title="对表达矩阵可进行简单的探索"></a>对表达矩阵可进行简单的探索</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">summary(hg_reads) </span><br><span class="line">summary(mus_reads) </span><br><span class="line">row.names(hg_reads) &lt;- hg_reads$gene</span><br><span class="line">GAPDH &lt;- hg_reads[rownames(hg_reads)==&quot;GAPDH&quot;,]</span><br><span class="line"></span><br><span class="line">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">biocLite(&quot;dplyr&quot;)</span><br><span class="line">library(dplyr)</span><br><span class="line">sample_mean &lt;- group_by(hg,sample)%&gt;% summarize(mean=mean(count))</span><br><span class="line">gene_mean &lt;- group_by(hg,gene)%&gt;% summarize(mean=mean(count)) </span><br><span class="line">tmp &lt;- full_join(hg,gene_mean,by=&apos;gene&apos;)</span><br><span class="line">write.table(tmp,file = &quot;tmp.count&quot;,sep = &quot;\t&quot;,quote = FALSE,row.names = FALSE)</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-3a0f908f6e1b3a5d.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="summary"></p>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-79ab4164d01877a0.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="image.png"></p>
<h5 id="另一种"><a href="#另一种" class="headerlink" title="另一种"></a>另一种</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line">options(stringsAsFactors = FALSE)</span><br><span class="line">#导入数据</span><br><span class="line">homo_control &lt;- read.table(&quot;Homo_SRR3589956.geneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;homo_control&quot;))</span><br><span class="line">homo_rep1 &lt;- read.table(&quot;Homo_SRR3589957.geneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;homo_rep1&quot;))</span><br><span class="line">homo_rep2 &lt;- read.table(&quot;Homo_SRR3589958.geneCounts&quot;,sep=&quot;\t&quot;,col.names = c(&quot;gene_symbol&quot;,&quot;rep2&quot;))</span><br><span class="line">mus_control1 &lt;- read.table(&quot;Mus_SRR3589959.geneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;mus_control1&quot;))</span><br><span class="line">mus_control2 &lt;- read.table(&quot;Mus_SRR3589960.geneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;mus_control2&quot;)) </span><br><span class="line">mus_rep1 &lt;- read.table(&quot;Mus_SRR3589961.geneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_symbol&quot;,&quot;mus_rep1&quot;)) </span><br><span class="line">mus_rep2 &lt;- read.table(&quot;Mus_SRR3589962.geneCounts&quot;, sep=&quot;\t&quot;,col.names = c(&quot;gene_symbol&quot;,&quot;mus_rep2&quot;))</span><br><span class="line">#去除后五行</span><br><span class="line">homo_control &lt;- homo_control[-58523:-58527,]</span><br><span class="line">homo_rep1 &lt;- homo_rep1[-58523:-58527,]</span><br><span class="line">homo_rep2 &lt;- homo_rep2[-58523:-58527,]</span><br><span class="line">mus_control1 &lt;- mus_control1[-48322:-48326,]</span><br><span class="line">mus_control2 &lt;- mus_control2[-48322:-48326,]</span><br><span class="line">mus_rep1 &lt;- mus_rep1[-48322:-48326,]</span><br><span class="line">mus_rep2 &lt;- mus_rep2[-48322:-48326,]</span><br><span class="line">#整合</span><br><span class="line">homo_raw_count &lt;- merge(merge(homo_control, homo_rep1, by=&quot;gene_symbol&quot;), homo_rep2, by=&quot;gene_symbol&quot;)</span><br><span class="line">mus_raw_count &lt;- merge(merge(mus_control1, mus_control2, by=&quot;gene_symbol&quot;), merge(mus_rep1,mus_rep2, by=&quot;gene_symbol&quot;))</span><br><span class="line">summary(homo_raw_count)</span><br><span class="line">summary(mus_raw_count)</span><br><span class="line">row.names(homo_raw_count) &lt;- homo_raw_count$gene_symbol</span><br><span class="line">row.names(mus_raw_count) &lt;- mus_raw_count$gene_symbol</span><br><span class="line">gapdh &lt;- homo_raw_count[rownames(homo_raw_count)==&quot;GAPDH&quot;,]</span><br><span class="line">gapdh &lt;- mus_raw_count[rownames(mus_raw_count)==&quot;GAPDH&quot;,]</span><br><span class="line"></span><br><span class="line">#因为我们无法在EBI数据库上直接搜索找到ENSMUSG00000024045.5这样的基因，只能是ENSMUSG00000024045的整数，没有小数点，所以需要进一步替换为整数的形式</span><br><span class="line">#第一步将匹配到的.以及后面的数字连续匹配并替换为空，并赋值给ENSEMBL</span><br><span class="line">#ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, raw_count_filt1$gene_id) </span><br><span class="line">#将ENSEMBL重新添加到raw_count_filt1矩阵</span><br><span class="line">#row.names(raw_count_filter) &lt;- ENSEMBL</span><br></pre></td></tr></table></figure>
<p><img src="http://upload-images.jianshu.io/upload_images/2116382-8ee1f5c36ac2faa3.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="gapdh"><br>以上两种都有两个问题：AKAP95未找到 小鼠第一列不是gene name<br>和sort是否为reads name无关，htseq count -i 参数去掉则默认为gene id</p>
<h5 id="看生物学意义特殊的基因表现如何如GAPDH-β-ACTIN"><a href="#看生物学意义特殊的基因表现如何如GAPDH-β-ACTIN" class="headerlink" title="看生物学意义特殊的基因表现如何如GAPDH,β-ACTIN"></a>看生物学意义特殊的基因表现如何如GAPDH,β-ACTIN</h5><p><img src="http://upload-images.jianshu.io/upload_images/2116382-e29d08dec77d3ec7.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240" alt="gapdh"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">#! usr/bin/bash</span><br><span class="line">set -u</span><br><span class="line">set -e</span><br><span class="line">set -o pipefail</span><br><span class="line">hg19_ref=~/fastq/hisat2_index/hg19/genome</span><br><span class="line">mm10_ref=~/fastq/hisat2_index/mm10/genome</span><br><span class="line">hg19_gff=~/fastq/refdata/Genecode/gencode.v26lift37.annotation.gff3</span><br><span class="line">mm10_gff=~/fastq/refdata/Genecode/gencode.vM10.annotation.gtf</span><br><span class="line">ls Homo*.sorted.bam|while read id;do(htseq-count -f bam -r pos -s no $id $hg19_gff &gt; $&#123;id%%.*&#125;.sgeneCounts 2&gt; $&#123;id%%.*&#125;.sHTseq.log);done</span><br><span class="line">ls Mus*.sorted.bam|while read id;do(htseq-count -f bam -r pos -s no $id $mm10_gff &gt; $&#123;id%%.*&#125;.sgeneCounts 2&gt; $&#123;id%%.*&#125;.sHTseq.log);done</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">options(stringsAsFactors = FALSE)</span><br><span class="line">homo_control &lt;- read.table(&quot;Homo_SRR3589956.sgeneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;homo_control&quot;))</span><br><span class="line">homo_rep1 &lt;- read.table(&quot;Homo_SRR3589957.sgeneCounts&quot;,sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;homo_rep1&quot;))</span><br><span class="line">homo_rep2 &lt;- read.table(&quot;Homo_SRR3589958.sgeneCounts&quot;,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;rep2&quot;))</span><br><span class="line">mus_control1 &lt;- read.table(&quot;Mus_SRR3589959.sgeneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;mus_control1&quot;))</span><br><span class="line">mus_control2 &lt;- read.table(&quot;Mus_SRR3589960.sgeneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;mus_control2&quot;)) </span><br><span class="line">mus_rep1 &lt;- read.table(&quot;Mus_SRR3589961.sgeneCounts&quot;, sep=&quot;\t&quot;, col.names = c(&quot;gene_id&quot;,&quot;mus_rep1&quot;)) </span><br><span class="line">mus_rep2 &lt;- read.table(&quot;Mus_SRR3589962.sgeneCounts&quot;, sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;mus_rep2&quot;))</span><br><span class="line">homo_control &lt;- homo_control[-60472:-60476,]</span><br><span class="line">homo_rep1 &lt;- homo_rep1[-60472:-60476,]</span><br><span class="line">homo_rep2 &lt;- homo_rep2[-60472:-60476,]</span><br><span class="line">mus_control1 &lt;- mus_control1[-48441:-48445,]</span><br><span class="line">mus_control2 &lt;- mus_control2[-48441:-48445,]</span><br><span class="line">mus_rep1 &lt;- mus_rep1[-48441:-48445,]</span><br><span class="line">mus_rep2 &lt;- mus_rep2[-48441:-48445,]</span><br><span class="line">homo_raw_count &lt;- merge(merge(homo_control, homo_rep1, by=&quot;gene_id&quot;), homo_rep2, by=&quot;gene_id&quot;)</span><br><span class="line">mus_raw_count &lt;- merge(merge(mus_control1, mus_control2, by=&quot;gene_id&quot;), merge(mus_rep1,mus_rep2, by=&quot;gene_id&quot;))</span><br><span class="line">ENSEMBL &lt;- gsub(&quot;\\.\\d*_\\d*&quot;, &quot;&quot;, homo_raw_count$gene_id)</span><br><span class="line">row.names(homo_raw_count) &lt;- ENSEMBL # error</span><br><span class="line">ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, mus_raw_count$gene_id) </span><br><span class="line">row.names(mus_raw_count) &lt;- ENSEMBL</span><br><span class="line">summary(homo_raw_count)</span><br><span class="line">summary(mus_raw_count)</span><br><span class="line">#在UniProt数据库找到AKAP95的id</span><br><span class="line">MUS_AKAP95 &lt;- mus_raw_count[rownames(mus_raw_count)==&quot;ENSMUSG00000024045&quot;,] #reads很大</span><br></pre></td></tr></table></figure>
<p>最开始未加 -r pos 有问题 reads数目不同，后续计数没有大问题<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">count.matrix&lt;-function()</span><br><span class="line">&#123;</span><br><span class="line">  .libPaths(&quot;/home/tingm/R/x86_64-pc-linux-gnu-library/3.2&quot;)</span><br><span class="line">  library(limma)</span><br><span class="line">  library(edgeR)</span><br><span class="line">    #change</span><br><span class="line">    path&lt;-&quot;/home/tingm/fastq/homo_result_count/&quot;</span><br><span class="line">    #change</span><br><span class="line">      samples&lt;-read.table(&quot;samples.list.txt&quot;,header=F,sep=&quot;\t&quot;)</span><br><span class="line">      samples&lt;-samples[,1]</span><br><span class="line">      #change</span><br><span class="line">      data.raw&lt;-read.table(&quot;Homo_SRR3589956.sgeneCounts&quot;,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;Homo_SRR3589956.sgeneCounts&quot;))</span><br><span class="line">      for(sample in samples)</span><br><span class="line">      &#123;</span><br><span class="line">        print(sample)</span><br><span class="line">        file&lt;-paste(path,sample,sep=&quot;&quot;)</span><br><span class="line">        data.new&lt;-read.table(file,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,sample))</span><br><span class="line">        data.raw&lt;-merge(data.new,data.raw,by=&quot;gene_id&quot;)</span><br><span class="line">      &#125;</span><br><span class="line">     data.raw&lt;-data.raw[-1:-5,]</span><br><span class="line">     #ENSEMBL &lt;- gsub(&quot;\\.\\d*_\\d*&quot;, &quot;&quot;, data.raw$gene_id)</span><br><span class="line">     #row.names(data.raw) &lt;- ENSEMBL</span><br><span class="line">     # ENSEMBL &lt;- gsub(&quot;\\.\\d*&quot;, &quot;&quot;, mus_raw_count$gene_id)</span><br><span class="line">     write.table(data.raw,&quot;count.matix.csv&quot;,row.names=F,quote=F)</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure></p>
<blockquote>
<p>参考引用来自以下资料<br><a href="https://mp.weixin.qq.com/s/V32iruEVRKJ_1RubuS_lUQ" target="_blank" rel="noopener">PANDA姐的转录组入门(6)：reads计数</a><br><a href="https://mp.weixin.qq.com/s/RolngZkua6GJboHQMfP52w" target="_blank" rel="noopener">生信媛 转录组入门(6)： reads计数</a><br><a href="http://www.jianshu.com/p/24cf44b610a7" target="_blank" rel="noopener">lxmic转录组入门(6)： reads计数</a></p>
</blockquote>
</div><div class="tags"><a href="/tags/bioinformatics/">bioinformatics</a></div><div class="post-nav"><a class="pre" href="/2017/08/21/转录组入门/转录组入门——7/">转录组入门——7</a><a class="next" href="/2017/08/14/转录组入门/转录组入门——5/">转录组入门——5</a></div><div id="container"></div><link rel="stylesheet" href="/css/default.css?v=0.0.0"><script src="/js/gitment.browser.js?v=0.0.0"></script><script>var gitment = new Gitment({
  owner: 'mt831',
  repo: 'mt831.github.io',
  oauth: {
    client_id: 'be43294b043c8abd1479',
    client_secret: '4addc0b15b35c8299f1a298458426e1aaee7c90a',
  },
})
gitment.render('container')
</script></div></div></div><div class="pure-u-1 pure-u-md-1-4"><div id="sidebar"><div class="widget"><div class="search-form"><input id="local-search-input" placeholder="Search" type="text" name="q" results="0"/><div id="local-search-result"></div></div></div><div class="widget"><div class="widget-title"><i class="fa fa-folder-o"> 分类</i></div><ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/Beginning-Perl-free-学习笔记/">Beginning Perl (free)学习笔记</a><span class="category-list-count">3</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/Linux基础/">Linux基础</a><span class="category-list-count">2</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/Test/">Test</a><span class="category-list-count">1</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/mysql学习记录/">mysql学习记录</a><span class="category-list-count">2</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/project/">project</a><span class="category-list-count">8</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/seq实战/">seq实战</a><span class="category-list-count">4</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/数据科学实训/">数据科学实训</a><span class="category-list-count">18</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/数据科学实训营/">数据科学实训营</a><span class="category-list-count">1</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/爬虫/">爬虫</a><span class="category-list-count">1</span></li><li class="category-list-item"><a class="category-list-link" href="/categories/转录组入门/">转录组入门</a><span class="category-list-count">8</span></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-star-o"> 标签</i></div><div class="tagcloud"><a href="/tags/hisat2/" style="font-size: 15px;">hisat2</a> <a href="/tags/bioinformatics/" style="font-size: 15px;">bioinformatics</a> <a href="/tags/Linux/" style="font-size: 15px;">Linux</a> <a href="/tags/sql/" style="font-size: 15px;">sql</a> <a href="/tags/mysql/" style="font-size: 15px;">mysql</a> <a href="/tags/php/" style="font-size: 15px;">php</a> <a href="/tags/bioinformaticsant/" style="font-size: 15px;">bioinformaticsant</a> <a href="/tags/RTG-tools/" style="font-size: 15px;">RTG-tools</a> <a href="/tags/nanopolish/" style="font-size: 15px;">nanopolish</a> <a href="/tags/samtools/" style="font-size: 15px;">samtools</a> <a href="/tags/trim-galore/" style="font-size: 15px;">trim_galore</a> <a href="/tags/perl/" style="font-size: 15px;">perl</a> <a href="/tags/HTSeq/" style="font-size: 15px;">HTSeq</a> <a href="/tags/FastViromeExplorer/" style="font-size: 15px;">FastViromeExplorer</a> <a href="/tags/giremi/" style="font-size: 15px;">giremi</a> <a href="/tags/gatk/" style="font-size: 15px;">gatk</a> <a href="/tags/star/" style="font-size: 15px;">star</a> <a href="/tags/python/" style="font-size: 15px;">python</a> <a href="/tags/爬虫，python/" style="font-size: 15px;">爬虫，python</a> <a href="/tags/picard/" style="font-size: 15px;">picard</a> <a href="/tags/git/" style="font-size: 15px;">git</a> <a href="/tags/SnpEff/" style="font-size: 15px;">SnpEff</a></div></div><div class="widget"><div class="widget-title"><i class="fa fa-file-o"> 最近文章</i></div><ul class="post-list"><li class="post-list-item"><a class="post-list-link" href="/2018/08/07/SQL/sql_2/">mysql学习记录--Part.2</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/08/06/SQL/sql_1/">mysql学习记录--Part.1</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/07/23/爬虫/拉勾网.Part1/">拉勾网.Part-1</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/28/project/nanopolish/">nanopolish</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/20/project/RTG-tools安装/">RTG-tools安装</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/14/project/ann.giremi/">ann.giremi.sh</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/14/project/star.gatk/">star.gatk.sh</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/14/project/trim.aln.count & find_virus_filt/">trim.aln.count.sh & find_virus_filt.sh</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/06/04/project/giremi及gatk下载/">giremi, gatk下载使用</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/05/16/project/samtools命令/">samtools</a></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-external-link"> 友情链接</i></div><ul></ul><a href="https://github.com/mt831" title="github" target="_blank">github</a></div></div></div><div class="pure-u-1 pure-u-md-3-4"><div id="footer">Copyright © 2018 <a href="/." rel="nofollow">blog.</a> Powered by<a rel="nofollow" target="_blank" href="https://hexo.io"> Hexo.</a><a rel="nofollow" target="_blank" href="https://github.com/tufu9441/maupassant-hexo"> Theme</a> by<a rel="nofollow" target="_blank" href="https://github.com/pagecho"> Cho.</a></div></div></div><a class="show" id="rocket" href="#top"></a><script type="text/javascript" src="/js/totop.js?v=0.0.0" async></script><script type="text/javascript" src="//cdn.bootcss.com/fancybox/3.2.5/jquery.fancybox.min.js" async></script><script type="text/javascript" src="/js/fancybox.js?v=0.0.0" async></script><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/fancybox/3.2.5/jquery.fancybox.min.css"><script type="text/javascript" src="/js/search.js?v=0.0.0"></script><script>var search_path = 'search.xml';
if (search_path.length == 0) {
   search_path = 'search.xml';
}
var path = '/' + search_path;
searchFunc(path, 'local-search-input', 'local-search-result');
</script><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = '//hm.baidu.com/hm.js?' + '3d768ac3c655ab53fbe8a4ac31a4528a';
  var s = document.getElementsByTagName("script")[0];
  s.parentNode.insertBefore(hm, s);
  })();
</script><script type="text/javascript" src="/js/codeblock-resizer.js?v=0.0.0"></script><script type="text/javascript" src="/js/smartresize.js?v=0.0.0"></script></div></body></html>