<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    
    <meta name="theme-color" content="#33363b">
    <meta name="msapplication-TileColor" content="#33363b">
    
    
    
    <meta name="keywords" content="python, R, math, bioinformatics">
    
    
    <link rel="apple-touch-icon" sizes="180x180" href="/favicons/apple-touch-icon.png">
    
    
    <link rel="icon" type="image/png" sizes="192x192" href="/favicons/android-chrome-192x192.png">
    
    
    <link rel="icon" type="image/png" sizes="32x32" href="/favicons/favicon-32x32.png">
    
    
    <link rel="icon" type="image/png" sizes="16x16" href="/favicons/favicon-16x16.png">
    
    
    <link rel="mask-icon" href="/favicons/safari-pinned-tab.svg" color="#33363b">
    
    
    <link rel="manifest" href="/favicons/site.webmanifest">
    
    
    <meta name="msapplication-config" content="/favicons/browserconfig.xml">
    
    
    <link rel="alternate" href="/atom.xml" title="mt's blog" type="application/atom+xml" />
    
    <link rel="stylesheet" type="text/css" href="/css/index.css">
    
    <link rel="stylesheet" type="text/css" href="/css/sidebar.css">
    
    
<link rel="stylesheet" type="text/css" href="/css/page.css">
<link rel="stylesheet" type="text/css" href="/css/post.css">

    <link rel="stylesheet" type="text/css" href="/css/custom.css">
    <link rel="stylesheet" type="text/css" href="/css/normalize.css">
    <link rel="stylesheet" type="text/css" href="/css/atom-one-dark.css">
    <link rel="stylesheet" type="text/css" href="/css/lightgallery.min.css">
    <script type="text/javascript" src="/js/jquery.min.js"></script>
    <script defer type="text/javascript" src="/js/util.js"></script>
    <script defer type="text/javascript" src="/js/scrollspy.js"></script>
    <script defer type="text/javascript" src="/js/fontawesome-all.min.js"></script>
    <script defer type="text/javascript" src="/js/lightgallery.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-fullscreen.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-hash.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-pager.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-thumbnail.min.js"></script>
    <script defer type="text/javascript" src="/js/lg-zoom.min.js"></script>
    
    <script defer src="/js/busuanzi.pure.mini.js"></script>
    
    
    <script defer type="text/javascript" src="/js/search.js"></script>
    <script type="text/javascript">
    $(document).ready(function () {
      var searchPath = "search.xml";
      if (searchPath.length === 0) {
        searchPath = "search.xml";
      }
      var path = "/" + searchPath;
      searchFunc(path, "search-input", "search-result");
    });
    </script>
    
    
    <script defer type="text/javascript" src="/js/index.js"></script>
    
    <script defer type="text/javascript" src="/js/custom.js"></script>
    <title>转录组入门——7 | mt's blog</title>
  </head>
  <body itemscope itemtype="http://schema.org/WebPage" lang=""  data-spy="scroll" data-target=".list-group">
    
<header id="header" class="header">
  <div class="container">
    <div class="header-container">
      <div class="header-title">
        <h1 class="title"><a href="/">mt's blog</a></h1>
        <h2 class="subtitle"></h2>
      </div>
      <div class="logo">
        <img src="/images/ARIA_logo.png" alt="logo">
      </div>
    </div>
    
<nav id="nav" class="nav">
  <a id="nav-toggle" class="nav-toggle"><i class="fas fa-bars"></i></a>
  <ul id="menu">
    
    <li><a href="/">首页</a></li>
    
    <li><a href="/archives/">归档</a></li>
    
    <li><a href="/categories/">分类</a></li>
    
    <li><a href="/tags/">标签</a></li>
    
  </ul>
</nav>


  </div>
</header>


    <main id="main" class="main">
      <div class="container">
        <div class="main-container">
          <div class="content">
            
<section id="post" class="post">
  
  <article class="post-container card " itemscope itemtype="http://schema.org/Article">
    <div class="post-block">
      <link itemprop="mainEntityOfPage" href="http://yoursite.com/2017/08/21/转录组入门/转录组入门——7/">
      <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
       <meta itemprop="name" content="mt">
       <meta itemprop="description" content="">
       <meta itemprop="image" content="/images/avatar.png">
      </span>
      <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
       <meta itemprop="name" content="mt's blog">
      </span>
    </div>
    <header class="post-header">
      <h1 class="post-title" itemprop="name headline">
       <span class="">转录组入门——7</span>
      </h1>
      <div class="post-meta">
        
        <span class="post-date">
          <i class="far fa-calendar-plus"></i> <time title="post-date" itemprop="dateCreated datePublished" datetime="2017-08-21T16:02:38+08:00">2017-08-21</time>
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        <span class="post-categories">
          <i class="far fa-folder-open"></i>
          
          
          <span itemprop="about" itemscope itemtype="http://schema.org/Thing"> <a href="/categories/转录组入门/" itemprop="url" rel="index"><span itemprop="name">转录组入门</span></a></span>
          
          
        </span>
        
        
        
        <span class="post-meta-divider"> | </span>
        
        
        <span class="post-comment-count">
          <i class="far fa-comments"></i> <a href="/2017/08/21/转录组入门/转录组入门——7/#comments" itemprop="discussionUrl"><span class="post-comment-count hc-comment-count" data-xid="2017/08/21/转录组入门/转录组入门——7/" itemprop="commentsCount"></span></a>
        </span>
        
        
      </div>
    </header>
    <main class="post-body" itemprop="articleBody">
      <figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">.libPaths(&quot;/home/tingm/R/x86_64-pc-linux-gnu-library/3.2&quot;)</span><br><span class="line">  library(limma)</span><br><span class="line">  library(edgeR)</span><br><span class="line">  library(DESeq2)</span><br><span class="line">    #change</span><br><span class="line">    path&lt;-&quot;/home/tingm/fastq/mus_result_count/&quot;</span><br><span class="line">    #change</span><br><span class="line">      samples&lt;-read.table(&quot;samples.list.txt&quot;,header=F,sep=&quot;\t&quot;)</span><br><span class="line">      samples&lt;-samples[,1]</span><br><span class="line">      #change</span><br><span class="line">      data.raw&lt;-read.table(&quot;Mus_SRR3589959.sgeneCounts&quot;,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,&quot;Mus_SRR3589959.sgeneCounts&quot;))</span><br><span class="line">      for(sample in samples)</span><br><span class="line">      &#123;</span><br><span class="line">        print(sample)</span><br><span class="line">        file&lt;-paste(path,sample,sep=&quot;&quot;)</span><br><span class="line">        data.new&lt;-read.table(file,header=F,sep=&quot;\t&quot;,col.names = c(&quot;gene_id&quot;,sample))</span><br><span class="line">        data.raw&lt;-merge(data.new,data.raw,by=&quot;gene_id&quot;)</span><br><span class="line">      &#125;</span><br><span class="line">     data.raw&lt;-data.raw[-1:-5,]</span><br><span class="line">        #change</span><br><span class="line">     names(data.raw) &lt;- c(&quot;gene_id&quot;,&quot;Akap95_2&quot;,&quot;control_2&quot;,&quot;Akap95_1&quot;,&quot;control_1&quot;)</span><br><span class="line">     write.table(data.raw,&quot;count.matix.csv&quot;,row.names=F,quote=F)</span><br></pre></td></tr></table></figure>
<h5 id="使用DESeq2进行差异基因分析-筛选出基因数目不一样"><a href="#使用DESeq2进行差异基因分析-筛选出基因数目不一样" class="headerlink" title="使用DESeq2进行差异基因分析 筛选出基因数目不一样"></a>使用DESeq2进行差异基因分析 筛选出基因数目不一样</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">##数据预处理</span><br><span class="line">database &lt;- read.table(file = &quot;count.matix.csv&quot;, sep = &quot; &quot;, header = T,row.names = 1)</span><br><span class="line">database &lt;- round(as.matrix(database))</span><br><span class="line">##设置分组信息并构建dds对象</span><br><span class="line">condition &lt;- factor(c(&quot;Akap95&quot;,&quot;control&quot;,&quot;Akap95&quot;,&quot;control&quot;), levels = c(&quot;control&quot;, &quot;Akap95&quot;))</span><br><span class="line">coldata &lt;- data.frame(row.names = colnames(database), condition)</span><br><span class="line">dds &lt;- DESeqDataSetFromMatrix(countData=database, colData=coldata, design=~condition)</span><br><span class="line">dds &lt;- dds[ rowSums(counts(dds)) &gt; 1, ]</span><br><span class="line">##使用DESeq函数估计离散度，然后差异分析获得res对象</span><br><span class="line">dds &lt;- DESeq(dds)</span><br><span class="line">res &lt;- results(dds)</span><br><span class="line">#summary(res)</span><br><span class="line">#最后设定阈值，筛选差异基因，导出数据(全部数据。包括标准化后的count数)</span><br><span class="line">res &lt;- res[order(res$padj),]</span><br><span class="line">diff_gene_DESeq2 &lt;- subset(res, padj &lt; 0.05 &amp; (log2FoldChange &gt; 1 | log2FoldChange &lt; -1))</span><br><span class="line">diff_gene_DESeq2 &lt;- row.names(diff_gene_DESeq2)</span><br><span class="line">resdata &lt;- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)),by=&quot;row.names&quot;,sort=FALSE)</span><br><span class="line">write.csv(resdata,file = &quot;control_vs_Akap95.csv&quot;,row.names = F)</span><br></pre></td></tr></table></figure>
<h5 id="使用edgeR进行差异基因分析"><a href="#使用edgeR进行差异基因分析" class="headerlink" title="使用edgeR进行差异基因分析"></a>使用edgeR进行差异基因分析</h5><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">##跟DESeq2一样，导入数据，预处理（用了cpm函数）</span><br><span class="line">exprSet&lt;- read.table(file = &quot;count.matix.csv&quot;, sep = &quot; &quot;, header = TRUE, row.names = 1, stringsAsFactors = FALSE)</span><br><span class="line">group_list &lt;- factor(c(&quot;Akap95&quot;,&quot;control&quot;,&quot;Akap95&quot;,&quot;control&quot;), levels = c(&quot;control&quot;, &quot;Akap95&quot;))</span><br><span class="line">exprSet &lt;- exprSet[rowSums(cpm(exprSet) &gt; 1) &gt;= 2,]</span><br><span class="line">##设置分组信息，并做TMM标准化</span><br><span class="line">exprSet &lt;- DGEList(counts = exprSet, group = group_list)</span><br><span class="line">exprSet &lt;- calcNormFactors(exprSet)</span><br><span class="line">##使用qCML（quantile-adjusted conditional maximum likelihood）估计离散度（只针对单因素实验设计）</span><br><span class="line">exprSet &lt;- estimateCommonDisp(exprSet)</span><br><span class="line">exprSet &lt;- estimateTagwiseDisp(exprSet)</span><br><span class="line">##寻找差异gene(这里的exactTest函数还是基于qCML并且只针对单因素实验设计)，然后按照阈值进行筛选即可</span><br><span class="line">et &lt;- exactTest(exprSet)</span><br><span class="line">tTag &lt;- topTags(et, n=nrow(exprSet))</span><br><span class="line">diff_gene_edgeR &lt;- subset(tTag$table, FDR &lt; 0.05 &amp; (logFC &gt; 1 | logFC &lt; -1))</span><br><span class="line">diff_gene_edgeR &lt;- row.names(diff_gene_edgeR)</span><br><span class="line">write.csv(tTag$table,file = &quot;control_vs_Akap95_edgeR.csv&quot;)</span><br></pre></td></tr></table></figure>
<h5 id="使用limma进行差异分析-limma的voom流程"><a href="#使用limma进行差异分析-limma的voom流程" class="headerlink" title="使用limma进行差异分析   limma的voom流程"></a>使用limma进行差异分析   limma的voom流程</h5><p>待续</p>
<h5 id="不同软件包分析结果比较"><a href="#不同软件包分析结果比较" class="headerlink" title="不同软件包分析结果比较"></a>不同软件包分析结果比较</h5><p>待续</p>
<blockquote>
<p><a href="http://www.jianshu.com/p/3bfb21d24b74" target="_blank" rel="noopener">转录组入门（7）：差异基因分析</a><br><a href="https://mp.weixin.qq.com/s/1mda7NpOxShbOxZKSJar7Q" target="_blank" rel="noopener">从零开始学转录组（7）：差异基因表达分析</a><br><a href="http://www.biotrainee.com/thread-2018-1-1.html" target="_blank" rel="noopener">转录组入门（7-8）从差异基因到富集分析</a><br><a href="http://www.agctu.org/question/179" target="_blank" rel="noopener">为什么RNA-seq的read数服从泊松分布？</a></p>
</blockquote>

    </main>
    <footer class="post-footer">
      
      
      <div class="post-tags">
        
        
        
        
        
        
        <a class="post-tag button" href="/tags/bioinformatics/" style="background: #fc6423" rel="tag"><i class="fas fa-tags"></i> bioinformatics</a>
        
        
      </div>
      
    </footer>
  </article>
  
  
  <div class="post-nav">
    
    <div class="post-nav-next post-nav-item">
      <a href="/2017/08/15/转录组入门/转录组入门——6/" rel="next" title="转录组入门——6">
        <i class="fas fa-angle-left"></i>
        <span class="nav-title">转录组入门——6</span>
      </a>
    </div>
    
    
    <div class="post-nav-prev post-nav-item">
      <a href="/2017/08/23/转录组入门/转录组入门——8/" rel="prev" title="转录组入门——8">
        <span class="nav-title">转录组入门——8</span>
        <i class="fas fa-angle-right"></i>
      </a>
    </div>
    
  </div>
  
  
  

<div class="comments" id="comments">
  
  
  <div id="hypercomments_widget"></div>
  
  <script type="text/javascript">
  _hcwp = window._hcwp || [];
  _hcwp.push({ "widget": "Bloggerstream", "widget_id": "tingm831@gmail.com", "selector": ".hc-comment-count", "label": "{\%COUNT%\}" });
  
  _hcwp.push({ "widget": "Stream", "widget_id": "tingm831@gmail.com", "xid": "2017/08/21/转录组入门/转录组入门——7/" });
  
  (function() {
    if ("HC_LOAD_INIT" in window) {
      return;
    }
    HC_LOAD_INIT = true;
    var lang = (navigator.language || navigator.systemLanguage || navigator.userLanguage || "en").substr(0, 2).toLowerCase();
    var hcc = document.createElement("script");
    hcc.type = "text/javascript";
    hcc.defer = true;
    hcc.src = ("https:" === document.location.protocol ? "https" : "http") + "://w.hypercomments.com/widget/hc/tingm831@gmail.com/" + lang + "/widget.js";
    var s = document.getElementsByTagName("script")[0];
    s.parentNode.insertBefore(hcc, s.nextSibling);
  })();
  </script>
  
</div>



  
</section>

          </div>
          
          
          
<aside class="sidebar" id="sidebar">
  
  <div class="search">
    <div class="form-group">
      <i class="fas fa-search"></i> <input type="text" id="search-input" name="q" results="0" placeholder="搜索" class="form-control"/>
    </div>
  </div>
  <div class="search-result-box" id="search-result"></div>
  
  
<div class="info sidebar-item" id="info">
  
  <img class="author-avatar" src="/images/avatar.png" alt="mt">
  
  <p class="author-name">mt</p>
  <p class="author-description"> </p>
  <div class="site-count">
    
    <div class="archives-count">
      <p>归档</p>
      <p><a href="/archives/">36</a></p>
    </div>
    
    
    
    <span class="site-count-divider"> | </span>
    
    <div class="categories-count">
      <p>分类</p>
      <p><a href="/categories/">5</a></p>
    </div>
    
    
    
    <span class="site-count-divider"> | </span>
    
    <div class="tags-count">
      <p>标签</p>
      <p><a href="/tags/">3</a></p>
    </div>
    
  </div>
  
  <div class="rss">
    <a class="rss-link button sidebar-item" href="/atom.xml"><i class="fas fa-rss"></i> RSS</a>
  </div>
  
</div>


  <div class="sidebar-sticky">
    
    




<hr>
<div class="post-toc sidebar-item" id="toc-div">
  <p><i class="fas fa-list-ol"></i> 文章目录</p>
  <div class="post-toc-content"><ol class="toc"><li class="toc-item toc-level-5"><a class="toc-link" href="#使用DESeq2进行差异基因分析-筛选出基因数目不一样"><span class="toc-text">使用DESeq2进行差异基因分析 筛选出基因数目不一样</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#使用edgeR进行差异基因分析"><span class="toc-text">使用edgeR进行差异基因分析</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#使用limma进行差异分析-limma的voom流程"><span class="toc-text">使用limma进行差异分析   limma的voom流程</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#不同软件包分析结果比较"><span class="toc-text">不同软件包分析结果比较</span></a></li></ol></div>
</div>



    
    
    
<hr>
<div class="social-link sidebar-item">
  <p><i class="far fa-address-card"></i> 社交链接</p>
  <ul>
    
    <li><i class="fab fa-github"></i> <a href="https://github.com/mt831" target="_blank">GitHub</a></li>
    
  </ul>
</div>


    
    
  </div>
</aside>


          
        </div>
      </div>
    </main>
    
<footer id="footer" class="footer">
  <div class="container">
    <div class="back-to-top">
      <a id="back-to-top"><i class="fas fa-angle-double-up"></i></a>
    </div>
    <div class="footer-container">
      <div class="footer-left">
        <div class="copyright">
          <span class="author">mt</span>
          <span class="year"><i class="far fa-copyright"></i> 2018-4-1 - 2018</span>
          
        </div>
        
        
<div class="busuanzi">
  <span id="busuanzi_container_site_pv"><i class="fas fa-eye"></i> <span id="busuanzi_value_site_pv"></span></span>
  <span id="busuanzi_container_site_uv"><i class="fas fa-user"></i> <span id="busuanzi_value_site_uv"></span></span>
  <span id="busuanzi_container_page_pv"><i class="far fa-file-alt"></i> <span id="busuanzi_value_page_pv"></span></span>
</div>


        
      </div>
      <div class="footer-right">
        <div class="custom-info">
          
        </div>
        <div class="powered-by">
          由 <a href="https://hexo.io/" target="_blank">Hexo</a> 强力驱动 | 主题 <a href="https://github.com/AlynxZhou/hexo-theme-aria/" target="_blank">ARIA</a>
        </div>
      </div>
    </div>
  </div>
</footer>


  </body>
</html>
